<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN""http://www.w3.org/TR/html4/loose.dtd"><html><!-- InstanceBegin template="/Templates/standard.dwt" codeOutsideHTMLIsLocked="false" --><head><meta http-equiv="Content-Type" content="text/html; charset="><!-- InstanceBeginEditable name="doctitle" --><title>Automated Alignment Tools</title><!-- InstanceEndEditable --><link rel="stylesheet" href="styles/mesquitePackage.css" type="text/css"><!-- InstanceBeginEditable name="head" --><!-- InstanceEndEditable --></head><body><div id="masthead">  <div id="banner"><img src="splash.gif" width="164" height="64"></div>  <div id="mainTitle"><font id="packageName"><b>Align</b></font><font id="version"><br>    Version 1.11<br>    20 November 2007</font><BR>  David R. Maddison, Travis J. Wheeler, and Wayne P. Maddison </div>  <div id="subtitle"><!-- InstanceBeginEditable name="subtitle" -->Automated Alignment <!-- InstanceEndEditable --></div> </div><div><!-- end masthead --><div id="content"><!-- InstanceBeginEditable name="content" -->  <h3><img src="images/alignToDropped.gif" alt="Pairwise Aligner" width="16" height="16">Pairwise Alignment tool  </h3>  <p>This tool allows one to drop one or a selected set of sequences onto a reference sequence; the sequences dropped will then be aligned to the reference sequence, preserving whatever gaps are present in the reference sequence. In the process, if gaps need to be inserted into the reference sequence, then they will be also inserted into other, non-selected sequences. </p>  <p>The dropdown menu for this tool allows you to change the gap opening cost (default    8) the gap extension cost (default 3) within bases of a sequences, as well    as the equivalent costs at the ends of the sequence (default 2 and 2, respectively).    </p>  <p>The default substitution costs      are:</p>  <ul>    <li>for DNA and RNA data, 5 for a transition and 10 for a transversion</li>    <li>for protein data, 5 for each substitution.</li>  </ul>   <p>These costs can be changed using the Substitution Costs dialog box available from the tool's dropdown menu.</p>     <h3>Aligning nucleotide sequences to match an amino acid alignment </h3>  <p>This feature allows one to take a matrix of nucleotides, and an existing alignment of their translated amino acids, and have the nucleotides realigned to match the amino acid alignment.  To do this, you will need to have in the same Mesquite file both the nucleotide matrix and the protein matrix.  For example, you could do the following: </p>  <ul> 	<li>Assign codon positions and genetic code to a nucleotide matrix (see the main Mesquite manual for details).</li>	<li>Adjust each sequence so that its reading frame is correct, by using the <a href="manualAlignment.html#shiftMinStops">Shift To Minimize Stops</a> feature.</li>	<li>Trim any incomplete codons from the ends of the sequences by selecting the	  entire matrix and choosing <font size="-1"><u>Matrix&gt;Alter/Transform&gt;Other	      Choices</u></font>... and selecting 	Trim Terminal Incomplete Codons.  "Terminal Incomplete Codons" are nucleotides	that are only part of a codon. For example, if one sequence starts at a third	position, then that third position nucleotide represents only one-third of a	codon, and it will be trimmed. Once this is done, only complete codons will	be left in the sequence </li>	<li>Translate the DNA matrix to amino acids by choosing <font size="-1"><u>Characters&gt;Make New Matrix From&gt;Translate      DNA To Protein</u></font>.  You will now have the protein matrix in your file.</li>    <li>Align the protein matrix. You could use, for example, the Clustal Align      feature described below. If instead you export the matrix (e.g., using      the <u><font size="-1">File&gt;Export</font></u> options), align the proteins in a separate program, you      will then need to choose <u><font size="-1">File&gt;Include</font></u> file to include the output of the      alignment program into your file.</li>    <li>Finally, go to your DNA matrix, and choose <font size="-1"><u>Matrix&gt;Alter/Transform&gt;Align          DNA to Protein</u></font>. </li>  </ul>  <h3>ClustalW Align </h3>  <p>This feature allows one to select a single block of sequences, and then have ClustalW align them. To do this, select the block, then choose Matrix&gt;Align Multiple Sequences&gt;Clustal Align... </p>  <p>You will be presented first with a query as to whether you want to do the ClustalW alignment on a separate thread, or on the same thread. Mesquite can do multiple things at once, because it can have one thing running on one computational &quot;thread&quot;, and another thing happening on a separate thread. There is a main thread of the program that is the thread the user deals with directly, and that allows you to give commands to Mesquite (via menus, etc.). If this main thread is busy with a calculation, then you will not be able to ask for new things to happen in Mesquite until the calculation is done. By choosing &quot;Separate&quot; in the query that appears, you are asking Mesquite to create a thread separate from the main thread, thus enabling you to do things in Mesquite while the Clustal alignment is proceeding. However, if you do this, you must remember not to edit the matrix or close the window showing the matrix; if you do, Mesquite will be very unhappy. The safest thing to do is choose &quot;No&quot; to that query. </p>  <p>Once you make that choice, you will see a dialog box in which you must enter the directory location of ClustalW; if you use the Browse button in the dialog box you can choose ClustalW and have the location filled in automatically. If you wish, you can also alter the options of clustalw. If you then press OK, Mesquite will send that section of the matrix to ClustalW and ask for it to be aligned; it will then harvest the results and reincorporate that piece into the matrix.</p>  <p>The version of ClustalW that is to be used by Mesquite <em>must</em> be a version of ClustalW that is executable from the command line of your operating system.  </p>  <p>&nbsp;  </p>  <h3>Muscle Align </h3>  <p>This tool is just like the ClustalW Align feature described above, except that it works for Robert C. Edgar's <a href="http://www.drive5.com/muscle/">MUSCLE</a> program.  </p>  <p>&nbsp;  </p><!-- InstanceEndEditable --></div><!--end content --><div id="navBar">  <div id="sectionLinks">    <ul>      <li><a href="index.html">Home</a></li>      <li><a href="manualAlignment.html">Manual Alignment</a></li>      <li><a href="automated.html">Automated Alignment</a></li>      <li><a href="visualization.html">Visualization Features</a></li>      <li><a href="statistics.html">Statistics</a></li>      <li><a href="citation.html">Citation</a></li>      <li><a href="faq.html">FAQs</a></li>      <li><a href="future.html">Future Directions</a></li>      <li><a href="acknowledgements.html">Acknowledgements</a></li>      <li style="margin-top: 20px;"></li>      <li><a href="sourceCode.html">Source Code</a></li>      <li style="margin-top: 20px;"></li>	  <li><protectedURL>a href="../../../docs/mesquite/manual.html"</protectedURL>Mesquite Manual</a></li>    </ul>	<p>&nbsp;</p>	<ul><li> <a href="http://mesquiteproject.org"  style="border: 0px;"><img src="images/mesquiteIcon.gif" alt="Mesquite Home" width="32" height="32"> Mesquite Home</a> </li>    </ul>  </div></div></div><div id="navBarExtension">&nbsp;</div><!--end navbar --><div id="siteInfo"> Copyright &copy; 2007 <a href="http://david.bembidion.org">DavidR. Maddison, </a><a href="http://traviswheeler.com/">Travis J. Wheeler</a>, and <a href="http://salticidae.org/wpm/home.html">WayneP. Maddison</a></div><br></body><!-- InstanceEnd --></html>