<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><!-- #BeginTemplate "/Templates/Template.dwt" --><!-- DW6 -->
<head>
<!--LINK REL="SHORTCUT ICON" HREF="images/mesq.ico"-->
<!-- #BeginEditable "doctitle" --> 
<title>Molecular Data</title>
<!-- #EndEditable -->
<meta http-equiv="Content-Type" content="text/html; charset=iso-8859-1">
</head>

<body bgcolor="#FFFFFF">
<div id="Page" style="position:absolute; left:0px; top:0px; z-index:1; visibility: visible; width: 100%; height: 100%"> 
  <table border="0" cellspacing="0" cellpadding="0" width="100%">
    <tr> 
	  <td align="left" valign="top" height="60"> 
        <table border="0" cellspacing="0" cellpadding="0" height="60" width="300">
          <tr> 
       		<td width="10" height="60" align="left" valign="top"> <font face="Trebuchet MS"><img src="../images/pixel.gif" width="1" height="1" alt=" " border="0"></font></td>
            <td bgcolor="#FFFFFF"> 
              <div align="left"><font face="Trebuchet MS"><a href="../manual.html"><img src="../images/mesquitesmall.gif" width="268" height="92" border="0" alt="Mesquite"></a></font></div>
            </td>
          </tr>
		  <tr>
			<td height="42" align="left" valign="top"></td>
			<td height="42" align="left" valign="top"><font face="Trebuchet MS"><a href="../manual.html"><img src="../images/mesquiteLowerManual.gif" border="0"></a></font></td>
		  </tr>
<!--	  <tr>
  		    <td colspan="2" height="10"></td>
		  </tr>
	      <tr>
		    <td colspan="2" height="15" align="center" valign="middle">
			<!-- Google CSE Search Box Begins
				<form id="searchbox_016202157087875381635:khljhzi-ps8" action="http://www.google.com/cse">
					<input type="hidden" name="cx" value="016202157087875381635:khljhzi-ps8" />
					<input type="hidden" name="cof" value="FORID:0" />
					<input name="q" type="text" size="20" />
					<input type="submit" name="sa" value="Search Mesquite" />
				<!--<img src="http://www.google.com/coop/images/google_custom_search_smnar.gif" alt="Google Custom Search" />
		  		</form>
			<!-- Google CSE Search Box Ends
        		    </td>
		  </tr> -->
        </table>
      </td>
	<td>
        <table border="0" cellspacing="0" cellpadding="0" height="134" width="550">
          <tr valign="top" height="92"> 
           	<td width="72" align="center"  height="92" valign="top" bgcolor="#FFFFFF"><font face="Trebuchet MS"><img src="../images/fadeRight.gif" width="72" height="160"></font></td>
			<td bgcolor="#99FF99"><img src="../images/introduction.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width = "160" bgcolor="#99FF99"> 
                <p><font face="Trebuchet MS"><span class="smallplain"><a href="../manual.html">
                  Introduction</a><br>
                  </span><a href="../whyMesquite.html">Why?</a><br>
                  <a href="../publication.html#howToCite">How to Cite</a>                <br>
                  <a href="../publication.html">Publication</a><br>
                  <a href="../support.html">Support</a><br>
                  <a href="../credits.html">Credits</a> </font><br>
                  <font face="Trebuchet MS"><span class="smallplain"><a href="../help.html">Help</a></span></font><br>
                  <a href="../Help/MesquiteFAQ.html"><font face="Trebuchet MS">FAQ</font></a><br> 
                  <a href="http://mesquiteproject.org"><font face="Trebuchet MS">Web Site</font></a><br>
			</td>
      		<td bgcolor="#99FF99"><img src="../images/basics.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="220"  bgcolor="#99FF99"> 
              <p><font face="Trebuchet MS"><span class="smallplain"><a href="../help.html"></a>
                <a href="../managingFiles.html">Files</a></span> <br>
                <span class="smallplain"><a href="../menus.html">Menus</a><br>
                <a href="../windows.html">Windows</a><br>
				<a href="../charts.html">Charts</a><br>
                <a href="../scripting.html">Scripts/Macros</a> <br>
                <a href="../modules.html">Modules</a><br>
				<a href="../howMesquiteWorks.html">How</a></span></font>
              </td>
      		<td bgcolor="#99FF99"><img src="../images/objects.gif" valign="bottom"></td>
			<td align="left" valign="top" height="92" width="260"  bgcolor="#99FF99">
				<p><font face="Trebuchet MS">
                <span class="smallplain"><a href="../characters.html">Characters</a></span></font><font face="Trebuchet MS"><span class="smallplain"><a href="../Taxa.html"><br>
                Taxa<br>
                </a></span></font><font face="Trebuchet MS"><a href="../trees.html">Trees</a><br>
                <a href="../glossary.html">Glossary</a> </font></p>
              <p><font face="Trebuchet MS"><em><font face="Trebuchet MS"><a href="../changes.html">New</a><br>
                &nbsp;&nbsp;<a href="../changes.html">Features</a></font></em></font></p>
              </td>
      		<td align="left" valign="top" height="92" width="92"><font face="Trebuchet MS"><img src="../images/fadeLeft.gif" width="92" height="160"></font></td>
		</tr>
    	<tr>
		    <td width="72" height="20"></td>
		<td></td>
		    <td>&nbsp;</td>
            <td>&nbsp;</td>
            <td>&nbsp;</td>
		</tr>
		</table>
		</td>
	</tr>
  </table>
  <table border="0" cellspacing="0" cellpadding="0" width="100%" height="100%">
    <tr> 
      <td valign="top" align="left" width="110" height="100%"> 
        <table border="0" cellspacing="0" cellpadding="0" width="142">
          <tr> 
			<td colspan="3" bgcolor="#99FF99"><p><font face="Trebuchet MS"><img src="../images/fadeDown.gif" width="100%" height="95" valign "top" border="0" alt=" "></font></p></td>
		  </tr>
          <tr> 
            <td width="6" bgcolor="#99FF99" height="35"></td>
			<td colspan="2" bgcolor="99FF99" valign="top" height="35"><p><font face="Trebuchet MS"><a href="../toc.html">Table of</a><br> 
			   &nbsp; &nbsp; <a href="../toc.html">Contents</a></font></p></td>
		  </tr>
		  <tr>
            <td width="6" bgcolor="#99FF99" height="35"></td>
            <td colspan="2" bgcolor="#99FF99" height="35"><p><font face="Trebuchet MS"><img src="../images/analyses.gif"></font></p>
              </td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle" bgcolor="#99FF99">&nbsp;</td>
            <td width="131" height="35" align="left" valign="middle"><p>
			  <font face="Trebuchet MS">
			  <a href="../CharacterEvolution/charEvolution.html">Character</a><br>
			    &nbsp; <a href="../CharacterEvolution/charEvolution.html">Evolution</a> <br> <br>
			  <a href="../Diversification/diversification.html">Diversification</a><br><br>
			  <a href="molecular.html">Molecular</a><br>
			    &nbsp; <a href="molecular.html">Data</a><br><br>
			  <a href="../popGen/popGen.html">Gene trees</a> <br>  <br>
			  <a href="../RandomSimul/RandomSimul.html">Simulations &amp;</a><br>
			    &nbsp; <a href="../RandomSimul/RandomSimul.html">Randomizations</a> <br> <br>
			  <a href="../continuous/continuous.html">Continuous</a><br>
			    &nbsp; <a href="../continuous/continuous.html">Characters</a><br> <br>
			  <a href="../otherPrograms.html">Use with</a><br>
			    &nbsp; <a href="../otherPrograms.html">Other Programs</a> <br><br>
			  <a href="../studies/index.html">Studies</a> </font></p></td>
            <td width="5" height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr bgcolor="#99FF99"> 
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
            <td height="35" align="left" valign="middle">&nbsp;</td>
          </tr>
          <tr> 
            <td height="35" align="left" valign="middle"><font face="Trebuchet MS"><img src="../images/fadeUp.gif" width="5" height="92"></font></td>
            <td height="35" align="left" valign="middle" colspan="2"><font face="Trebuchet MS"><img src="../images/fadeUp.gif" width="100%" height="92"></font></td>
          </tr>
        </table>
      </td>
      <td valign="top" width="100%" height="100%"> 
        <table width="700" border="0" cellpadding="10" cellspacing="0">
          <tr> 
            <td><font face="Trebuchet MS"><!-- #BeginEditable "body" --> 
              <h2>Molecular data</h2>
              <p>Molecular data (DNA or protein sequences) can be edited, manipulated, 
                simulated and analyzed in various ways in Mesquite. Most of the 
                features discussed elsewhere concerning editing and analysis of 
                general categorical data also apply to molecular data; here we 
                focus on features specifically designed for sequence data. </p>
              <h4>Contents</h4>
              <ul>
                <li><a href="#editing">Editing molecular data</a></li>
                <ul>
                  <li><a href="#altertransform">Alter/Transform Tools</a> </li>
                  <li><a href="#alignment">Alignment</a></li>
                  <li><a href="#finding">Finding sequences</a></li>
                  <li><a href="#sequencedisplay">Display of sequences</a> </li>
                  <li><a href="#consensus">Consensus sequences</a></li>
                </ul>
                <li><a href="#fusing">Combining molecular matrices and sequences</a> 
                  <ul>
                    <li><a href="#addingTaxa">Adding new taxa/sequences</a> 
                      <ul>
                        <li><a href="#includetaxafromfile">Include Taxa from File</a></li>
                        <li><a href="#dragAndDrop">Drag and Drop</a></li>
                      </ul>
                    </li>
                    <li><a href="#addingGenes">Adding genes</a> 
                      <ul>
                        <li><a href="#concatMatrices">Concatenate Matrices</a></li>
                        <li><a href="#fusedMatrices">Fused Matrix Export</a></li>
                      </ul>
                    </li>
                  </ul>
                </li>
                <li><a href="#managing">Other tools for managing molecular matrices</a> 
                  <ul>
                    <li><a href="#archivenames">Alternative Taxon Names</a> </li>
                    <li><a href="#translate">Translate to protein</a> </li>
                  </ul>
                </li>
                <li><a href="#codonpositions">Codon Positions</a> </li>
                <li><a href="#geneticcodes">Genetic Codes</a> </li>
                <li><a href="#genbank">Interactions with GenBank</a> </li>
                <li><a href="#simDNA">Simulating DNA sequence evolution</a></li>
                <li><a href="#statsDNA">Statistics for DNA sequences</a></li>
                <li><a href="#distances">DNA Distances</a></li>
                <li><a href="#statsProtein">Statistics for Protein sequences</a></li>
                <li><a href="#tertiary">Visualizing tertiary structure</a></li>
                <li><a href="#popGen">Sequence Data within populations</a></li>
                <li><a href="#rec">Reconstructing ancestral states</a></li>
              </ul>
              <p align="center"><img src="../images/DNAMatrix.gif" width="437" height="200"></p>
              <h2><a name="editing" id="editing"></a>Editing molecular data</h2>
              <p>Molecular data can be imported from files of NBRF, FASTA, GenBank/GenPept,
                PHYLIP, CLUSTAL, and simple table format. It can also be exported
                to
                some 
                of these formats.</p>
              <p>The <a href="../characters.html#editor">Character Matrix Editor</a> 
                can be used to edit a molecular sequence matrix. Standard ambiguity 
                codes are allowed.</p>
              <h3><a name="altertransform" id="altertransform"></a>Alter/Transform 
                Tools</h3>
              <p> The following can be applied to all or the selected portions 
                of a molecular sequence matrix in the Character Matrix Editor. 
                These are available under the Alter/Transform submenu of the Matrix 
                menu (some of these are available under &quot;Other Choices&quot;):</p>
              <ul>
                <li><strong>Reverse Complement</strong> (DNA matrix only) &#8212; 
                  reverses the order of contiguously selected blocks of sequence 
                  and complements the sequence</li>
                <li><strong>Collapse Gaps</strong> &#8212; collapses gaps in the 
                  selected block by pushing all sites to the left, to to yield 
                  unaligned sequences</li>
                <li><strong>Remove Gaps-Only Characters</strong> &#8212; Removes 
                  from the matrix all characters that consist of nothing but gaps.                </li>
                <li><strong>Shift Other to Match</strong> &#8212; Shifts other 
                  sequences to match a region in a selected sequences; described 
                  in the manual for the <protectedURL>a href="../../../mesquite/align/aAlignIntro/index.html"</protectedURL> 
                  Align package</a>. </li>
                <li><strong>Shift To Minimize Stop Codons</strong> &#8212; Shifts each 
                  sequence 0, 1, or 2 bases so as to minimize the number of stop codons.  The amount each sequence will be shifted will
				  vary from sequence to sequence.  This feature requires that <a href=#codonpositions>codon positions</a> be designated. </li>
                <li><strong>Convert Uncertainties to Polymorphisms</strong> &#8212; 
                  Converts all uncertainties in the selection to polymorphisms. 
                  For example, for a nucleotide sequence, a Y (C or T) will be 
                  converted into &quot;C <em>and</em> T&quot;. </li>
                <li><strong>Convert Polymorphisms to Uncertainties</strong> &#8212; 
                  Converts all polymorphisms in the selection to uncertainty. 
                  For example, for a nucleotide sequence, a cell that has both 
                  C and T will be converted into &quot;C or T&quot;, i.e., Y.</li>
                <li><strong>Arbitrarily Resolve State Ambiguity</strong> &#8212; 
                  Any cell that has partial uncertainty in the state (e.g., &quot;Y&quot;, 
                  which is C or T) will be resolved into one of its states, chosen 
                  randomly. Thus, a Y will be converted to a C with 50% probability, 
                  and to a T with 50%. </li>
                <li><strong>Nucleotide complement</strong> (DNA matrix only) &#8212; 
                  enters the complementary sequence into the selected cells</li>
                <li><strong>Trim Terminal Gap Characters</strong> &#8212; deletes 
                  characters at edges of matrix that are gaps-only.</li>
              </ul>
              <p>Other options may appear; see the page on <a href="../characters.html">characters</a> 
                for standard choices in this submenu. You can also apply the other 
                editing tools described for <a href="../characters.html">character 
                matrices</a>.</p>
              <p>The view of the matrix can be adjusted in various ways. Cells 
                can be colored according to the state at the site (<strong>Color 
                Cells</strong> submenu, <strong>Character State</strong>) or according 
                to a value like the GC bias (Color Cells submenu, <strong>Cell 
                Value</strong>; can request this coloring to use a moving window). 
                Examples of this are shown below. The Display submenu of the Matrix 
                menu contains other options such as a <strong>Bird's eye view</strong> 
                which makes the cells narrow to show more of the sequences.</p>
              <p><strong>Copy Sequence</strong> (at bottom of Edit menu) copies 
                the selected cells of the matrix into the computer's clipboard 
                as a sequence. That is, whereas the standard Copy would place 
                into the clipboard selected pieces of the matrix in tab-delimited 
                text format (e.g., if the sequence AATCA is selected, &quot;A-tab-A-tab-T-tab-C-tab-A&quot; 
                would be copied), this modified Copy Sequence command does not 
                include tabs (thus, &quot;AATCA&quot; would be copied). This style 
                of copying is useful when interacting with programs like <a href="http://www.genecodes.com">Sequencher 
                (TM)</a>. For instance, if you want to find a piece of sequence 
                in a matrix in Mesquite within a chromatogram viewer of Sequencher, 
                do the following: select the sequence in Mesquite, choose Copy 
                Sequence, then go to Sequencher, select Find Bases, and paste 
                the sequence as the search string.</p>
              <h3><a name="alignment"></a>Alignment</h3>
              <p>The <protectedURL></protectedURL> <a href="../../../mesquite/align/aAlignIntro/index.html">Align</a></a> 
                package contains utilities for sequence alignment. These include 
                manual alignment tools (for shifting blocks of sequence, for example), 
                and automated tools (e.g., sending a region of the matrix to ClustalW 
                to align, or a pairwise alignment tool in the editor that will 
                align one sequence to another). See the <protectedURL></protectedURL> 
                <a href="../../../mesquite/align/aAlignIntro/index.html">Align</a> 
                manual</a> for more details. </p>
              <h3><a name="finding"></a>Finding Sequences</h3>
              <p>You have several options to find sequences in a matrix. </p>
              <p>First, you can use the Search Data <img src="../images/searchData.jpg" width="206" height="16"> 
                entry at the top of the Character Data Editor to enter a sequence 
                and hit return. (If the entry indicates Search Features instead, 
                touch on the brown &amp; or / until it shows a violet <img src="../images/searchDicon.jpg" width="12" height="14"> 
                and is labeled Search Data.) The results of the search will be 
                displayed in the Search results window. </p>
              <p>Second, pieces of sequences can be found using the Find Sequence 
                and Find All Sequences submenus of the Edit menu. The current 
                options are:</p>
              <ul>
                <li> <strong>Matching Sequence</strong>: This finds sequences 
                  matching a target sequence the user enters. The search allows 
                  a certain number of mismatches. Optionally, it can search for 
                  the reverse, complement and reverse complement of the target 
                  sequence. </li>
                <li><font color="#000000"> <strong>Maintain Target Match</strong>: 
                  This highlights and maintains highlighted the first occurence 
                  of a given sequence in a given taxon. First, you are asked which 
                  taxon to search in. Then, it displays a panel like this:<br>
                  <img src="../images/maintainTargetMatch.jpg" width="350" height="27"><br>
                  underneath the matrix. The first button (red X) is to close 
                  the panel; the second pauses the search; the third allows you 
                  to select another taxon as your focus. If you type a sequence 
                  into the text area, the matching sequence (if any) will be highlighted 
                  in the matrix. Mesquite is constantly monitoring this text, 
                  and so you don't need to give any command to find again if you 
                  change the text. This is useful if working with a program like 
                  Sequencher. If you see a stretch of sequence while viewing chromatograms 
                  that you'd like to find in the matrix in Mesquite, type in the 
                  sequence into the text box and you will quickly be taken to 
                  it in the taxon.</font></li>
                <li><font color="#000000"><strong>Maintain Clipboard Match</strong>: 
                  This is similar to Maintain Target Match, except that it obtains 
                  the search string not from the text area but from the clipboard. 
                  If the clipboard changes, the function will automatically find 
                  the sequence again in the matrix. This is useful if working 
                  with a program like Sequencher. If you turn on Maintain Clipboard 
                  Match, then you can copy stretches of a sequence within Sequencher, 
                  and Mesquite will automatically highlight it, without your having 
                  to return to Mesquite or give any other command to it. (Mesquite 
                  is constantly monitoring the clipboard to see if it changes).</font></li>
              </ul>
              <h2><a name="fusing" id="fusing"></a>Combining molecular matrices
                or sequences</h2>
              <p>Often you have sequences in different matrices, and you need 
                to fuse them into a single matrix for analyses. You may be adding 
                new sequences from an existing gene, or you may be adding new 
                genes for existing taxa.</p>
              <h3><a name="addingTaxa"></a>Adding new taxa/sequences</h3>
              <p>If you want to take two matrices and concatenate them vertically, 
                i.e. add new taxa to an existing set of gene sequences, then you 
                can do it either from a menu or using Drag and Drop.</p>
              <table width="100%" border="0" cellspacing="0" cellpadding="0">
                <tr> 
                  <td><div align="center"><img src="images/gene1beforeConcat.jpg" width="72" height="58"></div></td>
                  <td><div align="center"><img src="images/genePQRBeforeInclude.jpg" width="70" height="58"></div></td>
                  <td><div align="center"><img src="images/becomesArrow.gif" width="44" height="14"></div></td>
                  <td><div align="center"><img src="images/genesAfterInclude.jpg" width="70" height="115"></div></td>
                </tr>
              </table>
              <p> 
              <ul>
                <li><a name="includetaxafromfile"></a><strong>Include Taxa from 
                  File</strong>: In the Taxa&amp;Trees menu, this function reads 
                  in an external file and attempts to concatenate taxa from the 
                  file onto taxa blocks of the current file, and also to add their 
                  data to existing matrices. This is useful to add new gene sequences 
                  to an existing file. File formats that can be incorporated are 
                  NEXUS, NBRF, and FASTA. This incorporation will NOT preserve 
                  the associated information of the external NEXUS file such as 
                  footnotes for the taxa and characters. </li>
                <li><a name="dragAndDrop" id="dragAndDrop"></a><strong>Drag and
                     Drop:</strong> If you drag FASTA, GenBank/GenPept, or NBRF
                     files onto a Character Matrix Editor showing sequence data,
                     the
                     sequences
                     in these
                  files will be added to the matrix as new taxa. All files that
                     you drag and drop in one operation must be of the the same
                     file format.</li>
              </ul>
              <h3><a name="addingGenes"></a>Adding genes</h3>
              <p>If you want to take two matrices and concatenate them horizontally, 
                i.e. add new genes to an existing set of taxa, then how you do 
                this depends on whether your taxa have the exact same names in 
                the two matrices, or not.</p>
              <table width="100%" border="0" cellspacing="0" cellpadding="0">
                <tr> 
                  <td><div align="center"><img src="images/gene1beforeConcat.jpg" width="72" height="58"></div></td>
                  <td><div align="center"><img src="images/gene2beforeConcat.jpg" width="69" height="58"></div></td>
                  <td><div align="center"><img src="images/becomesArrow.gif" width="44" height="14"></div></td>
                  <td><div align="center"><img src="images/genesAfterConcat.jpg" width="102" height="77"></div></td>
                </tr>
              </table>
              <p> 
              <ul>
                <li><a name="concatMatrices" id="concatMatrices"></a><strong>Concatenate 
                  Matrices</strong>: If the corresponding taxa have the exact 
                  same names in the two matrices (as shown in the example above 
                  &#8212; taxon B has the exact same name in both matrices, as 
                  does taxon C), then you can use the <font size="-1"><u>(Character 
                  Matrix Editor)Matrix&gt;Utilities&gt;Concatenate Matrices</u></font> 
                  command to fuse the matrices. If both matrices belong to the 
                  same taxon block, then simply use the command to concatenate. 
                  On the other hand, if the matrices have overlapping but not 
                  identical taxon blocks, then you must bring both matrices along 
                  with their respective taxon blocks into memory. Let's assume 
                  you have two separate files, one for Gene 1 and one for Gene 
                  2. Copy the first file (so as to save an original copy of gene 
                  1 on its one), and give the copy a name like &quot;Fused.nex&quot;. 
                  Then, open it in Mesquite. Next, use Link File in the edit menu 
                  to read in the second file. Now, go to the Character Matrix 
                  Editor for the Gene 1 Matrix, and select Concatenate Matrices 
                  from the Utilities submenu. Later you can choose Close File 
                  from the File menu to close the linked file for gene 2, and 
                  save your fused file.</li>
                <li><a name="fusedMatrices"></a><strong>Fused Matrix Export (NEXUS)</strong>: 
                  If the corresponding taxa have different names in the two matrices, 
                  then this is the easiest method to fuse, although it can also 
                  be useful when taxon names are the same. It is a special file 
                  exporter that can fuse matrices of different genes as it exports. 
                  It is available by choosing Export in the File menu. If you 
                  have a <strong>single taxa block</strong> with multiple matrices 
                  attached to it (e.g., one for each gene), then this exporter 
                  will write a single combined matrix with the sequences of different 
                  genes concatenated for each taxon. It will also write CHARPARTITION 
                  and CHARSETs to record which section of the fused matrix corresponds 
                  to which source matrix. It will also write a MrBayes block with 
                  that program's partitioning commands. If the different matrices 
                  are of mixed type (DNA, protein, non-molecular categorical) 
                  then you have a choice between reducing them to simple categorical 
                  data or maintaining a mixed matrix (suitable for MrBayes). Often 
                  however, you will have slightly different names for the sequences 
                  of different genes, and these may therefore exist in the file 
                  represented by <strong>different taxa blocks</strong>. For instance, 
                  you may have a taxa block with taxa &quot;A1&quot;, &quot;B1&quot;, 
                  and &quot;C1&quot; with corresponding matrix for COI sequences 
                  (gene 1) from species A, B and C, and another taxa block with 
                  taxa &quot;B2&quot;, &quot;C2&quot;, and &quot;D2&quot; for 
                  28S sequences (gene 2) from the same species. You want to fuse 
                  them as follows:<br>
                  <table width="100%" border="0" cellspacing="0" cellpadding="0">
                    <tr> 
                      <td><div align="center"><img src="images/gene1beforeConcat2.jpg" width="72" height="58"></div></td>
                      <td><div align="center"><img src="images/gene2beforeInclude.jpg" width="71" height="58"></div></td>
                      <td><div align="center"><img src="images/becomesArrow.gif" width="44" height="14"></div></td>
                      <td><div align="center"><img src="images/genesAfterConcat.jpg" width="102" height="77"></div></td>
                    </tr>
                  </table>
                  The Fused Matrix exporter permits you to export these into a 
                  single matrix as long as you have indicated how the sequences 
                  correspond to one another. To do this, we suggest you create 
                  a new taxa block representing the species or specimens. In this 
                  example, create a taxa block &quot;Species&quot; with taxa A, 
                  B and C. This will be the &quot;master block of taxa&quot; that 
                  will organize the export. (Alternatively, you could choose one 
                  of the genes' taxa blocks as the master block.) <a href="../popGen/popGen.html#newassociation">Set 
                  up a Taxa Association</a> between the master block of taxa and 
                  each of the other blocks of taxa. With the first Taxa Association 
                  between species and Gene 1 indicate that &quot;A1&quot; belongs 
                  with species A, &quot;B1&quot; belongs with species B, and &quot;C1&quot; 
                  belongs with species C. Set up the species-Gene 2 association 
                  similarly. The two taxa association will look like this in the 
                  List of Taxa window for the master taxa block:<br>
                  <img src="images/assocBeforeFusedExport.gif" width="730" height="134"> 
                  Then when you choose Fused Matrix Export, choose Species as 
                  your master taxa. The exporter will then find all of the data 
                  corresponding to each species, either under the species taxon 
                  itself or under one of the linked taxa indicated by the Taxa 
                  Association, and compose a fused matrix. If a single master 
                  taxon has more than one corresponding taxa in one of the other 
                  matrices, the data are merged using the same rules as for <a href="../characters.html#mergetaxa">Merge 
                  Taxa</a>.</li>
              </ul>
              <h2><a name="managing"></a>Other Tools for Managing Molecular matrices</h2>
              <p>Managing sequences in different matrices, especially if from 
                different genes, can be difficult. Several functions assist in 
                this. These features are not restricted to molecular data, but 
                we anticipate most of their use will be with sequences.</p>
              <p> 
              <ul>
                <li><strong><a name="archivenames"></a>Alternative Taxon Names</strong>: 
                  This feature allows you to have several alternative naming schemes 
                  for taxa, e.g. one set of names that includes your specimen 
                  codes, another set that uses formal taxonomic names (for publication), 
                  another set whose names are condensed for exporting the file 
                  to programs like CLUSTAL that can handle only short names, and 
                  so on. For details see <a href="../taxa.html#alternativeNames">here</a>.                </li>
                <li><strong><a name="translate"></a>Translation to protein</strong>: 
                  a new matrix of amino acids, derived by translation from existing 
                  protein-coding DNA matrix, can be formed by choosing <u><font size="-1">Characters&gt;Make 
                  New Matrix from&gt;Translate DNA to Protein</font></u>. This request that <a href="#codonpositions">codon positions</a>, and the correct <a href="#geneticcodes">genetic code</a>,
				  be assigned.</li>
              </ul>
              <h3><font face="Trebuchet MS"> </font> <a name="sequencedisplay"></a>Display 
                of Sequences</h3>
              <p>Protein-coding sequences can be colored by the amino acid into 
                which a triplet would be translated (under the genetic code for 
                that triplet) by choosing <u><font size="-1">Matrix&gt;Color Cells&gt;Color 
                Nucleotide by Amino Acid</font></u>. </p>
              <h3><font face="Trebuchet MS"> </font> <a name="consensus"></a>Consensus 
                Sequences</h3>
              <p>Consensus Sequences can be displayed above the character matrix 
                by choosing <u><font size="-1">Matrix&gt;Add Info Strip&gt;Consensus 
                Sequence Strip</font></u>, as indicated below by the arrows:</p>
              <p align="center"><br>
                <img src="images/consensus.gif" width="498" height="226"> </p>
              <p>In the above examples, two consensus sequences strips are displayed, 
                with slightly different options.</p>
              <p>Options are available by touching on the consensus sequence:</p>
              <p align="center"><img src="images/consensusAA.gif" width="386" height="285"></p>
              <p>&nbsp;</p>
               <h2><a name="codonpositions"></a>Codon Positions</h2>
              <p>You can assign codon positions to a portion of a sequence by the following steps:</p>
			  <ul>
			  <li>Open the character list window by choosing <u><font size="-1">Characters&gt;List of Characters</font></u>.</li>
			  <li>In the list window, select the characters whose position you wish to designate (choose Select All 
			  from the Edit menu if you wish to select all characters).</li>
			  <li>Touch on the title box ("Codon Position") of the Codon Position column, and a menu will drop down.  That menu will have a
			  submenu, called "Set Codon Position", that contains various options for setting codon positions.</li>
			  </ul>
			<h2><a name="geneticcodes"></a>Genetic Codes</h2>
              <p>The genetic code for sequence data can be specified in Mesquite's 
                List of Characters window (by choosing List of Characters, and 
                then <u><font size="-1">Columns&gt;Current Genetic Codes</font></u>, 
                or with a data matrix frontmost, <u><font size="-1">Matrix&gt;Genetic 
                Codes...</font></u>). Genetic codes are assigned to individual 
                characters (thus allowing one to have a mixed matrix of mitochondrial 
                and nuclear data, for example). To assign a genetic code, choose 
                select the characters, and use the popup menu of the title of 
                the &quot;Genetic Code&quot; column in the List of Characters 
                window.</p>
              <p>The genetic code affects, among other things, the Translate DNA 
                to Protein command, as well as the coloring of nucleotide sequences 
                if Color Nucleotide by Amino Acid is chosen. </p>
              <h2><a name="genbank" id="genbank"></a>Interactions with GenBank</h2>
              <p>The following features allow one to interact with GenBank. They 
                require that you have an Internet connection. </p>
              <ul>
                <li><strong>BLAST in Web Browser</strong>: Select a sequence or 
                  portion thereof in the data matrix. Choose <u><font size="-1">Matrix&gt;Search&gt;BLAST 
                  in Web Browser</font></u>, and Mesquite will send a BLAST request 
                  to GenBank to search for matching sequences. Your default web 
                  browser should open and take you to the BLAST page. </li>
                <li><strong>Top BLAST Matches</strong>: Select a sequence or portion 
                  thereof in the data matrix. Choose <u><font size="-1">Matrix&gt;Search&gt;Top 
                  BLAST Matches</font></u>, and you will be presented with a dialog 
                  box. In this you can choose: 
                  <ul>
                    <li> maximum number of matches to be sought by Mesquite </li>
                    <li>whether the report of results should be saved to a text 
                      file</li>
                    <li>whether details of the taxonomic lineage of the sequences 
                      should be reported</li>
                    <li>whether to import the top matches into the current matrix</li>
                    <li>the maximum time in seconds to wait for the BLAST search 
                      to complete. </li>
                  </ul>
                </li>
                <li><strong>Fetch &amp; Add GenBank Sequences</strong>: This option, 
                  in <u><font size="-1">Matrix&gt;Utilities&gt; Fetch &amp; Add 
                  GenBank Sequences</font></u>, allows one to enter a comma-delimited 
                  list of GenBank accession numbers, and Mesquite will acquire 
                  these from GenBank and import them into the current matrix.                </li>
              </ul>
              <h2><a name="simDNA" id="simDNA"></a>Simulating DNA sequence evolution</h2>
              <p>DNA sequence evolution can be simulated to build statistical 
                tests, for instance via parametric bootstrapping. See the page 
                on <a href="../RandomSimul/simDNA.html">simulating DNA sequences</a>.              </p>
              <h2><a name="statsDNA" id="statsDNA"></a>Statistics for DNA sequences</h2>
              <p>Calculations for categorical characters in general can be applied 
                to DNA sequences. For example, <a href="../CharacterEvolution/AncestralStates.html#parsimony">Parsimony 
                calculations</a> can be made for DNA sequences, as can basic descriptive 
                statistics such as the percent of a sequence or character that 
                is missing data or gaps. In addition, there are several modules 
                specifically designed for DNA data, illustrated by examples in 
                Mesquite_Folder/examples/Molecular. These calculate compositional 
                bias:</p>
              <ul>
                <li><a name="acgtBias"></a><strong>ACGT Compositional Bias</strong> 
                  &#8212; This module supplies the compositional bias of taxa, 
                  measured over the taxon's sequence. The bias is treated as a 
                  continuous character, and thus can be used wherever characters 
                  are used, as for instance in the reconstruction of the evolution 
                  of compositional bias as shown in the image below. It can return 
                  either the proportion G+C, or separately A, C, G, and T proportions.</li>
              </ul>
              <p align="center"><img src="images/gcBiasTaxa.jpg" width="446" height="350"></p>
              <ul>
                <li> <a name="charBias"></a><strong>Character Compositional Bias</strong> 
                  &#8212; This module supplies the compositional bias for characters. 
                  It calculates the percent of taxa with particular nucleotides 
                  (GC bias, or individual frequency of A, C, G or T) for a character. 
                  The image below shows a moving window analysis of compositional 
                  bias along a sequence; the instructions for generating the chart 
                  are given <a href="../charts.html#gcbias">here</a>.</li>
              </ul>
              <p align="center"><img src="../images/gcSequence.jpg" width="557" height="188"></p>
              <ul>
                <li> <a name="colorGCBias"></a><strong>GC bias coloring of matrices</strong> 
                  &#8212; The cells of the Character Matrix Editor may be colored 
                  according to a moving window of GC bias along the sequence, 
                  as shown below, by selecting <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Color 
                  Cells&gt;Color By Cell Value</font></u>, then once shown the 
                  colors can be smoothed by a moving window analysis by selecting 
                  <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Moving 
                  Window (for colors)</font></u>.</li>
              </ul>
              <p align="center"><img src="images/gcBiasColorMatrix.jpg" width="371" height="263"></p>
              <h2><a name="distances"></a>DNA Distances</h2>
              <p>Mesquite supports several distances for DNA data:</p>
              <p> 
              <ul>
                <li>Uncorrected (P) distances</li>
                <li>Jukes-Cantor</li>
                <li>Kimura 2-parameter</li>
                <li>F81 (Felsenstein, 1981)</li>
                <li>F84 (Felsenstein, 1984) </li>
              </ul>
              There are several options available (in the Distance Parameters 
              submenu) for dealing with ambiguous bases and gaps:</p> 
              <p> 
              <ul>
                <li><strong>Count Sites with Gap in Pair</strong>: If this is 
                  chosen, then when the distance between two sequences is calculated, 
                  then sites that contain a gap in one of the sequences but not 
                  the other will be by default included in the calculation. If 
                  this option is unchecked, then any site at which either sequence 
                  is a gap will be excluded. </li>
                <li><strong>Estimate Ambiguity Differences</strong>: If this is 
                  chosen, then sites in which one member of the pair has an ambiguity 
                  have their contribution to the distance estimated based upon 
                  the base patterns at non-ambiguous sites. </li>
                <li><strong>Base Frequence on Entire Matrix</strong>: If this 
                  is chosen, then those distances that use as parameters the frequencies 
                  of bases (A, C, G, and T) will calculate those frequencies based 
                  on all sequences; if it is not chosen, then the frequencies 
                  are calculated based upon just the two sequences being compared 
                  at any one time.</li>
              </ul></p>
              <h2><a name="statsProtein"></a>Statistics for Protein Data</h2>
              <p> 
              <ul>
                <li><a name="siteHydrophobicity"></a><strong>Site hydrophobicity</strong> 
                  &#8212; This module supplies the average amino acid hydrophobicity, 
                  averaged across taxa, for each site. It can be used in charts, 
                  for instance to see the relationship between a phylogenetic 
                  statistic for the site (character) and it average hydrophobicity. 
                  This <a href="../charts.html#varhydrophob">chart</a>, for example, 
                  shows parsimony character steps as a function of hydrophobicity:</li>
              </ul></p>
              <p align="center"><img src="../images/stepsVsHydro.jpg" width="260" height="217"></p>
              <p> 
              <ul>
                <li><a name="aaHydrophobicity"></a><strong>Amino Acid hydrophobicity</strong> 
                  &#8212; The cells of the Character Matrix Editor may be colored 
                  according to a moving window of hydrophobicity along the sequence, 
                  as shown below, by selecting <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Color 
                  Cells&gt;Color By Cell Value</font></u>, then once shown the 
                  colors can be smoothed by a moving window analysis by selecting 
                  <u><font size="-1" face="Arial, Helvetica, sans-serif">Matrix&gt;Moving 
                  Window (for colors)</font></u>.</li>
              </ul></p>
              <p align="center"><img src="images/aahydroColorMatrix.jpg" width="588" height="212"></p>
              <h2><a name="tertiary"></a>Visualizing tertiary structure</h2>
              <p>Although there are not yet dedicated windows for visualizing 
                phylogenetic statistics in the context of molecular structure, 
                features have been added to the Scattergram chart to allow it 
                to be adapted for this purpose. For instance, in this image cytochrome 
                B is shown, with the amino acids colored according to a simple 
                phylogenetic statistic: the number of parsimony steps on a phylogeny. 
                The colors are smoothed by a moving window, and show that several 
                coils of the molecule, a few at the left and one deep at the right, 
                evolve more rapidly than others. This example is illustrated in 
                the data file at Mesquite_Folder/examples/Molecular/06-cytochromeB.nex</p>
              <p align="center"><img src="images/cytStructure.jpg" width="360" height="292"></p>
              <p>To build such a chart, begin with a file with a matrix of protein 
                sequences. The procedure is also described in the example files 
                08-cytochromeBlinked.nex and 09-cytochromeBscatter.nex. </p>
              <p> 
              <ul>
                <li>Select <strong>New Linked Matrix</strong> from the Characters 
                  menu. When a matrix is made to be linked to a second matrix, 
                  the two matrices are constrained to have the same number of 
                  characters. </li>
                <li>Indicate that you want the linked matrix to be a Continuous 
                  matrix, and link it to your protein matrix. Then, turn it into 
                  a three dimensional matrix (Taxa X Characters X Coordinates 
                  [x, y and z]) by using <strong>Add Item</strong> and <strong>Rename 
                  Item</strong> in the Utilities submenu of the Matrix menu of 
                  the Character Matrix Editor. The x,y,z coordinates could be 
                  added for all taxa if known, but otherwise only one taxon needs 
                  to be filled out (because we will use the average x,y,z coordinates 
                  for the amino acids).</li>
                <li>Once the linked matrix of xyz amino acid positions is entered, 
                  select <u><font size="-1" face="Arial, Helvetica, sans-serif">Analysis&gt;New 
                  Scattergram for&gt; Characters</font></u>. Indicate you want 
                  the scattergram to be for <strong>Stored Characters.</strong> (If you have &quot;Use Stored Characters/Matrices by Default&quot; turned on in the Defaults submenu if the File menu, Mesquite won't ask you and will simply use Stored Characters.) Indicate <strong>Same</strong> value for the two axes. In 
                  the dialog box &quot;Values for axes&quot;, choose <strong>Mean 
                  Value of Character (Linked Matrix)</strong>. In response to 
                  &quot;Use characters from which matrix? (for Character Source)&quot; 
                  choose the protein sequence matrix as the matrix to be used. 
                  This will plot the sites (amino acids, characters) in their 
                  correct places, but as a series of round spots.</li>
                <li>To change the appearance of the plot, select <strong>Join 
                  the Dots</strong> in the Special Effects submenu of the Scattergram 
                  menu. Then select <strong>Thick Joints</strong>, deselect <strong>Show 
                  Dots</strong>, deselect <strong>Join First to Last</strong>, 
                  and set the marker size larger (e.g., 8). This will result in 
                  a plot as shown above, but without the colors.</li>
                <li>Next, choose <strong>Color by Third Value</strong> from the 
                  Colors menu and choose the value by which to color the amino 
                  acids. For parsimony steps, for instance, choose under <strong>Character 
                  Value with current tree</strong>.</li>
                <li>Finally, to use a moving window to smooth the colors, select 
                  <strong>Moving Window for Colors</strong> from the Colors menu 
                  and indicate the window size (e.g., 5).</li>
              </ul></p>
              <h2><a name="popGen"></a>Sequence data within populations</h2>
              <p>See the page on <a href="../popGen/popGen.html">population genetics</a>.</p>
              <h2><a name="rec" id="rec"></a>Reconstructing ancestral states</h2>
              <p>Ancestral states of molecular characters can be reconstructed 
                as described in the page on <a href="../CharacterEvolution/AncestralStates.html">reconstructing 
                ancestral states</a>. Likelihood methods are not yet available 
                for molecular characters.</p></li>
              </ul> <!-- #EndEditable --></font></td>
          </tr>
        </table>
        <table border="0" cellspacing="0" cellpadding="5" width="100%">
          <tr> 
            <td align="left" valign="top"> 
              <hr>
              <p align="center"><font face="Trebuchet MS"><span class="smallplain"><a href="http://mesquiteproject.org"><font size="2">Mesquite
              Home Page</font></a><font size="2"> | <a href="../manual.html">Mesquite
              Manual</a></font></span></font></p>
              <p><font face="Trebuchet MS"><span class="smallplain">Copyright 
                &copy; 2002-2007 by <a href="http://salticidae.org/wpm/home.html">Wayne 
                P. Maddison</a> and <a href="http://david.bembidion.org">David 
                R. Maddison</a>.<br>
&nbsp;All rights reserved.</span></font></p></td>
          </tr>
        </table>
      </td>
    </tr>
  </table>
</div>

<font face="Trebuchet MS"> 
</font>
</body>
<!-- #EndTemplate --></html>
