/* Mesquite.chromaseq source code.  Copyright 2005 D. Maddison, W. Maddison.  Disclaimer:  The Mesquite source code is lengthy and we are few.  There are no doubt inefficiencies and goofs in this code.  The commenting leaves much to be desired. Please approach this source code with the spirit of helping out. Perhaps with your help we can be more than a few, and make Mesquite better.  Mesquite is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY. Mesquite's web site is http://mesquiteproject.org  This source code and its compiled class files are free and modifiable under the terms of  GNU Lesser General Public License.  (http://www.gnu.org/copyleft/lesser.html) */package mesquite.chromaseq.PhredPhrap; import java.io.*;import java.util.*;import java.awt.*;import java.awt.event.*;import mesquite.lib.*;import mesquite.lib.characters.*;import mesquite.lib.duties.*;import mesquite.categ.lib.*;import mesquite.cont.lib.*;import mesquite.chromaseq.lib.*; /* ======================================================================== */public class PhredPhrap extends PhPhRunner implements ActionListener, ItemListener { 		NameParserManager nameParserManager;	String nameParsingRulesName="";	String editNameParserButtonString = "Edit Naming Rules...";	ChromFileNameParsing nameParsingRule;	Choice nameRulesChoice;		StringBuffer logBuffer;//	CommandRecord commandRec;	NameReference aceRef;	ProgressIndicator progIndicator = null;	boolean importing = true;	String[][] fileNameTranslation;		//for importing sequences	//MesquiteProject project = null;	FileCoordinator coord = null;	MesquiteFile file = null;	Taxa taxa  = null;	DNAData data = null;	DNAData originalData = null;	ContinuousData  qualityData = null;	MesquiteInteger maxChar = new MesquiteInteger(0);		static String previousDirectory = null;		int preDNANumberLength = 10;	final String processedFolder = "processed";	final String infoFileName = "info.xml";	final String originalFolder = "originals";//	final String fastaFolder = "fasta";	final String processedFastaFolder = "processedFasta";	final String processedACEFolder = "processedACE";	final String sequencesFolder = "sequences";	final String processedACESuffix = "m";	String fastaDirectory = null;	String processedFastaDirectory = null;	String processedACEDirectory = null;		boolean preferencesSet = false;				String phredParamPath; 	String phredPath;	String primerListPath;	String sampleCodeListPath;		String phrapOptions = "-qual_show 20 -vector_bound 0 ";	String phredOptions = "";	String fileExtension = ".ab1";	boolean requiresExtension=true;	boolean translateSampleCodes = true;	final static boolean makeAceAliases = false;	final static boolean runPhredPhrap = true;	boolean processPolymorphisms = true;	boolean verbose = true;	boolean singleTaxaBlock = false;	boolean addPhrapFailures = true;	boolean showBirdsEye = true;		double polyThreshold = 0.3;		ExtensibleDialog dialog = null;	SingleLineTextField primerListField = null;	SingleLineTextField dnaCodesField = null;	SingleLineTextField phredPathField =  null;	SingleLineTextField paramPathField = null;			int qualThresholdForTrim = 20;	int qualThresholdForLowerCase = 49;	boolean truncateMixedEnds = true;	int mixedEndWindow = 10;	int mixedEndThreshold = 5;	boolean renameContigsInAceFiles = true;	boolean addFragName = false;	boolean openFastaDirectory = true;//	boolean openAceDirectory=false;//	boolean unTrimAceReads = true;//	boolean importSequencesIntoMesquite = true;   //DEFUNCT	boolean backupOriginals = true;			/*.................................................................................................................*/	public boolean startJob(String arguments, Object condition, CommandRecord commandRec, boolean hiredByName){				loadPreferences();		aceRef = NameReference.getNameReference(AceFile.ACENAMEREF);				if (nameParserManager == null)			nameParserManager= (NameParserManager)MesquiteTrunk.mesquiteTrunk.findEmployeeWithName("#ChromFileNameParsManager");		if (nameParserManager == null)			return false;				logBuffer = new StringBuffer(200);		return true;	}	/*.................................................................................................................*/	public boolean doPhredPhrap(MesquiteProject project, boolean appendIfPossible, CommandRecord commandRec) {		importing = project !=null;		if (queryOptions()) {			MesquiteTrunk.mesquiteTrunk.incrementProjectBrowserRefreshSuppression();			boolean success = prepareAndRunPhredPhrap(project, appendIfPossible, commandRec);			MesquiteTrunk.mesquiteTrunk.decrementProjectBrowserRefreshSuppression();			return success;		}		return false;	}	/*.................................................................................................................*/	public void processSingleXMLPreference (String tag, String content) {		if ("requiresExtension".equalsIgnoreCase(tag))			requiresExtension = MesquiteBoolean.fromTrueFalseString(content);		else if ("phredParamPath".equalsIgnoreCase(tag))			phredParamPath = StringUtil.cleanXMLEscapeCharacters(content);		else if ("phredPath".equalsIgnoreCase(tag))			phredPath = StringUtil.cleanXMLEscapeCharacters(content);		else if ("primerListPath".equalsIgnoreCase(tag))			primerListPath = StringUtil.cleanXMLEscapeCharacters(content);		else if ("sampleCodeListPath".equalsIgnoreCase(tag))			sampleCodeListPath = StringUtil.cleanXMLEscapeCharacters(content);		else if ("previousDirectory".equalsIgnoreCase(tag))			previousDirectory = StringUtil.cleanXMLEscapeCharacters(content);		else if ("phrapOptions".equalsIgnoreCase(tag))			phrapOptions = StringUtil.cleanXMLEscapeCharacters(content);		else if ("fileExtension".equalsIgnoreCase(tag))			fileExtension = StringUtil.cleanXMLEscapeCharacters(content);		else if ("phredOptions".equalsIgnoreCase(tag))			phredOptions = StringUtil.cleanXMLEscapeCharacters(content);		else if ("nameParsingRulesName".equalsIgnoreCase(tag))			nameParsingRulesName = StringUtil.cleanXMLEscapeCharacters(content);		else if ("translateSampleCodes".equalsIgnoreCase(tag))			translateSampleCodes = MesquiteBoolean.fromTrueFalseString(content);		else if ("processPolymorphisms".equalsIgnoreCase(tag))			processPolymorphisms = MesquiteBoolean.fromTrueFalseString(content);		else if ("showBirdsEye".equalsIgnoreCase(tag))			showBirdsEye = MesquiteBoolean.fromTrueFalseString(content);		else if ("singleTaxaBlock".equalsIgnoreCase(tag))			singleTaxaBlock = MesquiteBoolean.fromTrueFalseString(content);		else if ("truncateMixedEnds".equalsIgnoreCase(tag))			truncateMixedEnds = MesquiteBoolean.fromTrueFalseString(content);		else if ("renameContigsInAceFiles".equalsIgnoreCase(tag))			renameContigsInAceFiles = MesquiteBoolean.fromTrueFalseString(content);		else if ("addFragName".equalsIgnoreCase(tag))			addFragName = MesquiteBoolean.fromTrueFalseString(content);		else if ("openFastaDirectory".equalsIgnoreCase(tag))			openFastaDirectory = MesquiteBoolean.fromTrueFalseString(content);		else if ("backupOriginals".equalsIgnoreCase(tag))			backupOriginals = MesquiteBoolean.fromTrueFalseString(content);		else if ("polyThreshold".equalsIgnoreCase(tag))			polyThreshold = MesquiteDouble.fromString(content);		else if ("qualThresholdForTrim".equalsIgnoreCase(tag))			qualThresholdForTrim = MesquiteInteger.fromString(content);		else if ("qualThresholdForLowerCase".equalsIgnoreCase(tag))			qualThresholdForLowerCase = MesquiteInteger.fromString(content);		else if ("mixedEndWindow".equalsIgnoreCase(tag))			mixedEndWindow = MesquiteInteger.fromString(content);		else if ("mixedEndThreshold".equalsIgnoreCase(tag))			mixedEndThreshold = MesquiteInteger.fromString(content);		preferencesSet = true;}/*.................................................................................................................*/public String preparePreferencesForXML () {		StringBuffer buffer = new StringBuffer(200);		StringUtil.appendXMLTag(buffer, 2, "requiresExtension", requiresExtension);  		StringUtil.appendXMLTag(buffer, 2, "translateSampleCodes", translateSampleCodes);  		StringUtil.appendXMLTag(buffer, 2, "phredParamPath", phredParamPath);  		StringUtil.appendXMLTag(buffer, 2, "phredPath", phredPath);  		StringUtil.appendXMLTag(buffer, 2, "primerListPath", primerListPath);  		StringUtil.appendXMLTag(buffer, 2, "sampleCodeListPath", sampleCodeListPath);  		StringUtil.appendXMLTag(buffer, 2, "processPolymorphisms", processPolymorphisms);  		StringUtil.appendXMLTag(buffer, 2, "polyThreshold", polyThreshold);  		StringUtil.appendXMLTag(buffer, 2, "previousDirectory", previousDirectory);  		StringUtil.appendXMLTag(buffer, 2, "singleTaxaBlock", singleTaxaBlock);  		StringUtil.appendXMLTag(buffer, 2, "qualThresholdForTrim", qualThresholdForTrim);  		StringUtil.appendXMLTag(buffer, 2, "qualThresholdForLowerCase", qualThresholdForLowerCase);  		StringUtil.appendXMLTag(buffer, 2, "mixedEndWindow", mixedEndWindow);  		StringUtil.appendXMLTag(buffer, 2, "mixedEndThreshold", mixedEndThreshold);  		StringUtil.appendXMLTag(buffer, 2, "truncateMixedEnds", truncateMixedEnds);  		StringUtil.appendXMLTag(buffer, 2, "renameContigsInAceFiles", renameContigsInAceFiles);  		StringUtil.appendXMLTag(buffer, 2, "addFragName", addFragName);  		StringUtil.appendXMLTag(buffer, 2, "openFastaDirectory", openFastaDirectory);  		StringUtil.appendXMLTag(buffer, 2, "backupOriginals", backupOriginals);  		StringUtil.appendXMLTag(buffer, 2, "phrapOptions", phrapOptions);  		StringUtil.appendXMLTag(buffer, 2, "fileExtension", fileExtension);  		StringUtil.appendXMLTag(buffer, 2, "phredOptions", phredOptions);  		StringUtil.appendXMLTag(buffer, 2, "nameParsingRulesName", nameParsingRulesName);  		StringUtil.appendXMLTag(buffer, 2, "showBirdsEye", showBirdsEye);  		preferencesSet = true;		return buffer.toString();	}	/*.................................................................................................................*/	/** returns whether this module is requesting to appear as a primary choice */	public boolean requestPrimaryChoice(){		return true;  	}	/*.................................................................................................................*/	public boolean isPrerelease(){		return true;	}	/*.................................................................................................................*/	public boolean isSubstantive(){		return true;	}	/*.................................................................................................................*/	public boolean queryFileLocations() {		MesquiteInteger buttonPressed = new MesquiteInteger(1);		ExtensibleDialog queryFilesDialog = new ExtensibleDialog(MesquiteTrunk.mesquiteTrunk.containerOfModule(), "Phred Phrap Locations & Options",buttonPressed);  //MesquiteTrunk.mesquiteTrunk.containerOfModule()		queryFilesDialog.addLabel("Phred Phrap - File Locations & Options");				phredPathField = queryFilesDialog.addTextField("Phred, Phrap, & Phd2Fasta path:", phredPath, 40);		Button phredBrowseButton = queryFilesDialog.addAListenedButton("Browse...",null, this);		phredBrowseButton.setActionCommand("phBrowse");				paramPathField = queryFilesDialog.addTextField("Phred parameter file:", phredParamPath, 40);		Button paramBrowseButton = queryFilesDialog.addAListenedButton("Browse...",null, this);		paramBrowseButton.setActionCommand("paramBrowse");				SingleLineTextField phredOptionsField = queryFilesDialog.addTextField("Phred options:", phredOptions, 26, true);		SingleLineTextField phrapOptionsField = queryFilesDialog.addTextField("Phrap options:", phrapOptions, 26, true);				queryFilesDialog.completeAndShowDialog(true);		if (buttonPressed.getValue()==0)  {			phredParamPath = paramPathField.getText();			phredPath = phredPathField.getText();			phrapOptions = phrapOptionsField.getText();			phredOptions = phredOptionsField.getText();		}		queryFilesDialog.dispose();		return (buttonPressed.getValue()==0);	}	/*.................................................................................................................*/	public boolean queryPostOptions() {		MesquiteInteger buttonPressed = new MesquiteInteger(1);		ExtensibleDialog queryPostDialog = new ExtensibleDialog(MesquiteTrunk.mesquiteTrunk.containerOfModule(), "Post-Phrap Options",buttonPressed);  //MesquiteTrunk.mesquiteTrunk.containerOfModule()		queryPostDialog.addLabel("Post-Phrap Processing by Mesquite");				IntegerField lowerCaseQualityField = queryPostDialog.addIntegerField("Quality threshold for lower case:", qualThresholdForLowerCase, 3);		Checkbox polyBox = queryPostDialog.addCheckBox("Convert multiple-peaks sites to ambiguity codes",processPolymorphisms);		DoubleField polyThresholdField = queryPostDialog.addDoubleField("Minimum secondary peak fraction for ambiguity:", polyThreshold, 6);		Checkbox truncateEndsBox = queryPostDialog.addCheckBox("Trim low quality regions from ends (FASTA & import)",truncateMixedEnds);		IntegerField trimQualityField = queryPostDialog.addIntegerField("Quality threshold for trim:", qualThresholdForTrim, 3);		IntegerField trimWindowField = queryPostDialog.addIntegerField("Trim window length:", mixedEndWindow, 3);		IntegerField trimThresholdField = queryPostDialog.addIntegerField("Trim window threshold:", mixedEndThreshold, 3);				//		Checkbox unTrimAceReadsField = queryPostDialog.addCheckBox("reset quality regions in reads in .ace files",unTrimAceReads);		Checkbox renameContigsField = queryPostDialog.addCheckBox("rename contigs in .ace files",renameContigsInAceFiles);		Checkbox addFragNameField = queryPostDialog.addCheckBox("add fragment name to contig name",addFragName);		queryPostDialog.addHorizontalLine(2);		Checkbox openFastaField = queryPostDialog.addCheckBox("open directory with fasta files",openFastaDirectory);//		Checkbox openAceField = queryPostDialog.addCheckBox("open directory with .ace file links",openAceDirectory);//		Checkbox importSequencesField = queryPostDialog.addCheckBox("import sequences into Mesquite",importSequencesIntoMesquite);		Checkbox showBirdsEyeBox=null;		if (importing) {			showBirdsEyeBox = queryPostDialog.addCheckBox("display matrix in bird's eye view",showBirdsEye);		}				String s = "After Phrap finishes assembling contigs, Mesquite will process the files produced.  It will change upper case nucleotides to lower case if the quality score falls below the specified quality threshold.\n\n";		s+= "Mesquite will also trim low quality ends of sequences, if you so choose.  This trimming is done by removing regions of leading or trailing nucleotides with quality values less than the specified quality threshold. ";		s+= "If the ends have a mixture of lower and higher quality nucleotides, then Mesquite will trim until if finds a window of nucleotides of specified length which have fewer than the \"Trim window threshold\" ";		s+= "of nucleotides below the quality score.  To put it another way, it will trim leading and trailing windows of nucleotides which have as many or more poor-quality nucleotides than specified; ";		s+= "it trims high-quality nucleotides only if they are external to low-quality nucleotides that are to be trimmed by this criterion.";		queryPostDialog.appendToHelpString(s);						queryPostDialog.completeAndShowDialog(true);		if (buttonPressed.getValue()==0)  {			qualThresholdForLowerCase = lowerCaseQualityField.getValue();			truncateMixedEnds = truncateEndsBox.getState();			qualThresholdForTrim = trimQualityField.getValue();						processPolymorphisms = polyBox.getState();			polyThreshold = polyThresholdField.getValue();						mixedEndWindow = trimWindowField.getValue();			mixedEndThreshold = trimThresholdField.getValue();			renameContigsInAceFiles = renameContigsField.getState();//			unTrimAceReads = unTrimAceReadsField.getState();			addFragName = addFragNameField.getState();			openFastaDirectory= openFastaField.getState();//			openAceDirectory = openAceField.getState();//			importSequencesIntoMesquite = importSequencesField.getState();			if (importing) {				showBirdsEye = showBirdsEyeBox.getState();			}		}		queryPostDialog.dispose();		return (buttonPressed.getValue()==0);	}	/*.................................................................................................................*/	public void getNameRuleFromChoice() {		nameParsingRule=null;		if (nameRulesChoice!=null) {			nameParsingRulesName = nameRulesChoice.getSelectedItem();			boolean noChoiceItems = (nameRulesChoice.getItemCount()<=0);			int sL = nameRulesChoice.getSelectedIndex();			if (sL <0) {				sL = 0;			}					if (!noChoiceItems) {				nameParsingRule = (ChromFileNameParsing)(nameParserManager.nameParsingRules.elementAt(sL));			}		}		if (nameParsingRule==null)			nameParsingRule = new ChromFileNameParsing();  //make default one	}	}	/*.................................................................................................................*/	public boolean queryOptions() {		if (nameParserManager == null)			return false;				MesquiteInteger buttonPressed = new MesquiteInteger(1);		dialog = new ExtensibleDialog(MesquiteTrunk.mesquiteTrunk.containerOfModule(), "Run Phred Phrap Options",buttonPressed);  //MesquiteTrunk.mesquiteTrunk.containerOfModule()		dialog.addLabel("Run Phred Phrap Options");		dialog.addHorizontalLine(2);				nameRulesChoice = dialog.addPopUpMenu ("File Naming Rules", nameParserManager.nameParsingRules.getElementArray(), 0);		nameParserManager.setChoice(nameRulesChoice);		nameRulesChoice.addItemListener(this);		if (nameRulesChoice!=null) {			boolean noChoiceItems = (nameRulesChoice.getItemCount()<=0);			int sL = nameParserManager.nameParsingRules.indexOfByName(nameParsingRulesName);			if (sL <0) {				sL = 0;			}					if (!noChoiceItems) {				nameRulesChoice.select(sL); 				nameParsingRule = (ChromFileNameParsing)(nameParserManager.nameParsingRules.elementAt(sL));			}		}				dialog.suppressNewPanel();		GridBagConstraints gridConstraints;		gridConstraints = dialog.getGridBagConstraints();		gridConstraints.fill = GridBagConstraints.NONE;		dialog.setGridBagConstraints(gridConstraints);		Panel panel = dialog.addNewDialogPanel(gridConstraints);		Button editNameParsersButton = dialog.addAButton(editNameParserButtonString, panel);		editNameParsersButton.addActionListener(this);		editNameParsersButton.setActionCommand("editNameParsersButton");		dialog.addBlankLine();						dnaCodesField = dialog.addTextField("Codes & names file:", sampleCodeListPath,26);		Button dnaCodesBrowseButton = dialog.addAListenedButton("Browse...",null, this);		dnaCodesBrowseButton.setActionCommand("DNANumbersBrowse");				Checkbox translateCodesBox = dialog.addCheckBox("translate sample codes using name file", translateSampleCodes);				primerListField = dialog.addTextField("Primer list file:", primerListPath,26);		Button primerBrowseButton = dialog.addAListenedButton("Browse...",null, this);		primerBrowseButton.setActionCommand("primerBrowse");				Checkbox requiresExtensionBox = dialog.addCheckBox("only process files with standard extensions (ab1,abi,ab,CRO,scf)", requiresExtension);		SingleLineTextField fileExtensionField = dialog.addTextField("file extension for chromatogram copies:", fileExtension, 8, true);		dialog.addHorizontalLine(2);						Panel buttonPanel = dialog.addNewDialogPanel();		Button fileLocationsButton = dialog.addAButton("Phred Phrap Locations & Options...",buttonPanel);		fileLocationsButton.addActionListener(this);		fileLocationsButton.setActionCommand("phLocations");				Checkbox backupOriginalsBox = dialog.addCheckBox("save backups of original chromatograms", backupOriginals);				Panel buttonPanel2 = dialog.addNewDialogPanel();		Button postProcessingButton = dialog.addAButton("Post-Phrap Options...",buttonPanel2);		postProcessingButton.addActionListener(this);		postProcessingButton.setActionCommand("postOptions");		Checkbox singleTaxaBlockBox=null;		if (importing) {			singleTaxaBlockBox = dialog.addCheckBox("import as single taxon block",singleTaxaBlock);		}				String s = "Mesquite searches within the name of each chromatogram file for both a code indicating the sample (e.g., a voucher number) and the primer name. ";		s+= "To allow this, you must first define an rule that defines how the chromatogram file names are structured.\n\n";		s+= "If you so choose, Mesquite will search for the sample code within a sample names file you select, on each line of which is:\n";		s+= "   <code><tab><short sample name><tab><long sample name>;\n";		s+= "where the code, short sample name, and long sample name are all single tokens (you can force a multi-word name to be a single token by surrounding the name with single quotes). ";		s+= "The short sample name is for the file names, and must be <27 characters; the long sample name is the name you wish to have within the FASTA file.\n\n";		dialog.appendToHelpString(s);				dialog.completeAndShowDialog(true);		boolean success=(buttonPressed.getValue()==0);		if (success)  {			fileExtension = fileExtensionField.getText();			requiresExtension = requiresExtensionBox.getState();			translateSampleCodes = translateCodesBox.getState();//			runPhredPhrap = runPhredPhrapBox.getState();			backupOriginals = backupOriginalsBox.getState();			primerListPath = primerListField.getText();			sampleCodeListPath = dnaCodesField.getText();			if (importing) {				singleTaxaBlock = singleTaxaBlockBox.getState();			}		}		storePreferences();  // do this here even if Cancel pressed as the File Locations subdialog box might have been used		dialog.dispose();		return success;	}	/*.................................................................................................................*/	boolean makeDirectoriesForFragment(String fragmentDirPath){				File newDir = new File(fragmentDirPath);		try { newDir.mkdir();    //make folder for this gene					processedFastaDirectory = fragmentDirPath  + MesquiteFile.fileSeparator + processedFastaFolder;		newDir = new File(processedFastaDirectory);		newDir.mkdir();	//make processed fastaFolder				String sequencesDirectory = fragmentDirPath  + MesquiteFile.fileSeparator + sequencesFolder;		newDir = new File(sequencesDirectory);		newDir.mkdir();	//make sequences folder for holding all of the sequences and files produces by phred and phrap				if (makeAceAliases) {			newDir = new File(fragmentDirPath  + MesquiteFile.fileSeparator + "ace");			newDir.mkdir();	//make folder for holding links (shortcuts, aliases) to unprocessed ACE files						processedACEDirectory = fragmentDirPath  + MesquiteFile.fileSeparator + processedACEFolder;			newDir = new File(processedACEDirectory);			newDir.mkdir();	//make folder for holding links (shortcuts, aliases) to processed ACE files		}		}		catch (SecurityException e) { 			logln("Couldn't make directory.");			return false;		}		return true;	}	/*.................................................................................................................*/	public void echoStringToFile(String s, StringBuffer sb){		logln(s);		sb.append(s +  StringUtil.lineEnding());   		}	/*.................................................................................................................*/	String getPhredCommand(){		if (MesquiteTrunk.isWindows())			return StringUtil.protectForWindows(phredPath + "phred.exe");		else			return StringUtil.protectForUnix(phredPath + "phred");	}	/*.................................................................................................................*/	String getPhrapCommand(){		if (MesquiteTrunk.isWindows())			return StringUtil.protectForWindows(phredPath + "phrap.exe");		else			return StringUtil.protectForUnix(phredPath + "phrap");	}	/*.................................................................................................................*/	String getPhd2FastaCommand(){		if (MesquiteTrunk.isWindows())			return StringUtil.protectForWindows(phredPath + "phd2fasta.exe");		else			return StringUtil.protectForUnix(phredPath + "phd2fasta");	}	int sequenceCount = 0;	String importedDirectoryPath, importedDirectoryName;				/*.................................................................................................................*/	public boolean prepareAndRunPhredPhrap(MesquiteProject project, boolean appendIfPossible, CommandRecord commandRec){		boolean pleaseStorePref = false;		if (!preferencesSet || StringUtil.blank(phredPath)) {			phredPath = MesquiteFile.chooseDirectory("Choose directory containing phred, phrap, and phd2fasta: ");			if (StringUtil.blank(phredPath))				return false;			if (!phredPath.endsWith(MesquiteFile.fileSeparator))				phredPath+=MesquiteFile.fileSeparator;			pleaseStorePref = true;		}		if (!preferencesSet || StringUtil.blank(phredParamPath)) {			MesquiteString paramDir = new MesquiteString();			MesquiteString paramFile = new MesquiteString();			phredParamPath = MesquiteFile.openFileDialog("Choose phred parameter file: ", paramDir, paramFile);			if (StringUtil.blank(phredParamPath))				return false;			pleaseStorePref = true;		}		String primerList = "";		PrimerList primers = null;		if (!preferencesSet || StringUtil.blank(primerListPath)) {			MesquiteString primerListDir = new MesquiteString();			MesquiteString primerListFile = new MesquiteString();			primerListPath = MesquiteFile.openFileDialog("Choose file containing primer list", primerListDir, primerListFile);			if (StringUtil.blank(primerListPath))				return false;			pleaseStorePref = true;		}				if ((!preferencesSet || StringUtil.blank(sampleCodeListPath)) && translateSampleCodes) {			MesquiteString dnaNumberListDir = new MesquiteString();			MesquiteString dnaNumberListFile = new MesquiteString();			sampleCodeListPath = MesquiteFile.openFileDialog("Choose file containing sample codes and names", dnaNumberListDir, dnaNumberListFile);			if (StringUtil.blank(sampleCodeListPath))				return false;			pleaseStorePref = true;		}						if (pleaseStorePref)			storePreferences();				boolean havePrimerList = false;		if (!StringUtil.blank(primerListPath)) {			primerList = MesquiteFile.getFileContentsAsString(primerListPath);			if (!StringUtil.blank(primerList) ) {				primers = new PrimerList(primerList);								havePrimerList = true;			}		}		else 			return false;				String directoryPath = MesquiteFile.chooseDirectory("Choose directory containing ABI files:", previousDirectory); //MesquiteFile.saveFileAsDialog("Base name for files (files will be named <name>1.nex, <name>2.nex, etc.)", baseName);				if (StringUtil.blank(directoryPath))			return false;				File directory = new File(directoryPath);		importedDirectoryPath = directoryPath + MesquiteFile.fileSeparator;		importedDirectoryName = directory.getName();		previousDirectory = directory.getParent();		storePreferences();		if (directory.exists() && directory.isDirectory()) {			logBuffer.setLength(0);			progIndicator = new ProgressIndicator(getProject(),"Preparing for Phred/Phrap");			progIndicator.setStopButtonName("Stop");			progIndicator.start();			boolean abort = false;			/*			Thread mt = Thread.currentThread();			 boolean piMine = false;			 if (mt instanceof MesquiteThread) 			 progIndicator = ((MesquiteThread)mt).getProgressIndicator();			 if (progIndicator ==null) {			 progIndicator = new ProgressIndicator(getProject(),"Running Phred & Phrap", 0);			 piMine = true;			 if (mt instanceof MesquiteThread)			 ((MesquiteThread)mt).setProgressIndicator(progIndicator);			 }			 progIndicator.setButtonMode(ProgressIndicator.FLAG_AND_HIDE);			 progIndicator.start();			 boolean abort = false;			 */						String sampleCodeList = "";			Parser sampleCodeListParser = null;			boolean haveNameList = false;			if (!StringUtil.blank(sampleCodeListPath) && translateSampleCodes) {				sampleCodeList = MesquiteFile.getFileContentsAsString(sampleCodeListPath);				if (!StringUtil.blank(sampleCodeList)) {					sampleCodeListParser = new Parser(sampleCodeList);					haveNameList = true;				}			}			String cPath;			String seqName;			String fullSeqName;			String fragName = "";			StringBuffer shellScript = new StringBuffer(6000);			StringBuffer postShellScript = new StringBuffer(6000);			sequenceCount = 0;						if (MesquiteTrunk.isWindows())  {				addWindowsHeader(shellScript);			} else {//				shellScript.append("setenv PHRED_PARAMETER_FILE "+StringUtil.protectForUnix(phredParamPath)+"\n");				shellScript.append("#!/bin/csh  -v\nsetenv PHRED_PARAMETER_FILE "+StringUtil.protectForUnix(phredParamPath)+"\n");			}			//	shellScript.append(getAliasCommand ("phred",  phredPath + "phred"));			//	shellScript.append(getAliasCommand ("phd2fasta",phredPath+"phd2fasta"));			//	shellScript.append(getAliasCommand ("phrap",phredPath+"phrap"));						//set path = ( $path /usr/bin /usr/sbin /phredPhrap/ph/)									int loc = 0;						String[] files = directory.list();			StringBuffer renameBuffer = new StringBuffer(1000);			String processedDirPath = directoryPath + MesquiteFile.fileSeparator + processedFolder;  // directory into which processed files go			String originalsDirPath = directoryPath + MesquiteFile.fileSeparator + originalFolder;  // directory for copies of original files			echoStringToFile("\nPhred Phrap processing of chromatograms as scripted by Mesquite", logBuffer);			echoStringToFile(" Processing directory: ", logBuffer);			echoStringToFile("  "+directoryPath+"\n", logBuffer);			echoStringToFile("Using names and codes file: " +sampleCodeListPath+"\n", logBuffer);			echoStringToFile("Using primers file: " + primerListPath+"\n", logBuffer);			String rootDir = processedDirPath;			String fragmentDirPath = "";			File newDir;			int numPrepared = 0;						shellScript.append(ShellScriptUtil.getChangeDirectoryCommand(rootDir));			newDir = new File(processedDirPath);						try { newDir.mkdir();			if (backupOriginals) {				newDir = new File(originalsDirPath);				newDir.mkdir(); 			}			}			catch (SecurityException e) {				logln("Couldn't make directory.");				if (progIndicator!=null) progIndicator.goAway();				return false;			}						String runningFilePath = rootDir + MesquiteFile.fileSeparator + "running";			String statusFilePath = rootDir + MesquiteFile.fileSeparator + "status";			if (runPhredPhrap)				MesquiteFile.putFileContents(runningFilePath, "Phred Phrap are running...", true);			fileNameTranslation= new String[2][files.length];						for (int i=0; i<files.length; i++) {				progIndicator.spin();				if (progIndicator.isAborted())					abort = true;				if (abort)					break;				fragName = "";				if (files[i]==null )					;				else {					cPath = directoryPath + MesquiteFile.fileSeparator + files[i];					File cFile = new File(cPath);					if (cFile.exists() && !cFile.isDirectory() && (!files[i].startsWith(".")) && (!requiresExtension || (files[i].endsWith("ab1") ||  files[i].endsWith(".abi")  || files[i].endsWith(".ab")  ||  files[i].endsWith(".CRO") || files[i].endsWith(".scf")))) {												String chromFileName = cFile.getName();						if (StringUtil.blank(chromFileName)) {							echoStringToFile("Bad file name; it is blank.", logBuffer);							// remove "running"							if (progIndicator!=null) progIndicator.goAway();							return false;						}												MesquiteString sampleCodeSuffix = new MesquiteString();						MesquiteString sampleCode = new MesquiteString();						MesquiteString primerName = new MesquiteString();												//here's where the names parser processes the name												if (nameParsingRule!=null) {							if (!nameParsingRule.parseFileName(this, chromFileName, sampleCode, sampleCodeSuffix, primerName, logBuffer))								continue;						}						else {							echoStringToFile("Naming parsing rule is absent.", logBuffer);							return false;						}						String startToken = "Sample";						if (!StringUtil.blank(nameParsingRule.getStartToken()))							startToken = nameParsingRule.getStartToken();												MesquiteString stLouisString = new MesquiteString("");						if (primers != null)							fragName = primers.getFragmentName(primerName.getValue(),stLouisString);																		if (!StringUtil.blank(sampleCode.getValue())) {							/* Translate code number to sample name if requested  */							if (haveNameList && translateSampleCodes) {								loc = sampleCodeList.indexOf(sampleCode.getValue());   //еее problem:  if have 551A and 551, this will pick up 551																if (loc<0 && !(sampleCode.getValue().equalsIgnoreCase("0000")||sampleCode.getValue().equalsIgnoreCase("000"))) {									seqName = sampleCode.getValue();									fullSeqName = sampleCode.getValue();								}								else {									sampleCodeListParser.setPosition(loc+sampleCode.getValue().length()+1);									seqName = StringUtil.removeNewLines(sampleCodeListParser.getNextToken());									fullSeqName = StringUtil.removeNewLines(sampleCodeListParser.getNextToken());									if (!";".equals(sampleCodeListParser.getNextToken()))										fullSeqName = seqName;								}							}							else {								seqName = sampleCode.getValue();								fullSeqName = sampleCode.getValue();							}						}						else {							seqName = chromFileName.substring(1, 10); // change!							fullSeqName = seqName;						}												seqName = StringUtil.cleanseStringOfFancyChars(seqName + sampleCodeSuffix.getValue());  // tack on suffix						fullSeqName = StringUtil.cleanseStringOfFancyChars(fullSeqName + sampleCodeSuffix.getValue());																		progIndicator.spin();												fragmentDirPath = processedDirPath;						if (StringUtil.blank(fragName)) {							echoStringToFile("  Primer not found in primer file: " + primerName, logBuffer);  // so stop processing this one							echoStringToFile("   " + chromFileName + " not processed", logBuffer);						}						else {								fragmentDirPath = StringUtil.cleansePath(directoryPath + MesquiteFile.fileSeparator + processedFolder + MesquiteFile.fileSeparator + fragName);							if (!makeDirectoriesForFragment(fragmentDirPath)){   //make directories for fragment in case they don't already exist								if (progIndicator!=null) progIndicator.goAway();								return false;							}							String sequenceDirPath = StringUtil.cleansePath(fragmentDirPath + MesquiteFile.fileSeparator + sequencesFolder  + MesquiteFile.fileSeparator + seqName); 							String infoFilePath = StringUtil.cleansePath(sequenceDirPath  + MesquiteFile.fileSeparator + infoFileName); 							newDir = new File(sequenceDirPath);     							try {								if (!newDir.exists()) {  // then this is our first time encountering this sequence									sequenceCount++;									shellScript.append(ShellScriptUtil.getChangeDirectoryCommand(sequenceDirPath));									shellScript.append(getPhredCommand() + " -id . -pd . -d " + StringUtil.blankIfNull(phredOptions) + " \n");									if (runPhredPhrap)										shellScript.append(ShellScriptUtil.getWriteStringAsFile(statusFilePath, "" + sequenceCount + ". " + seqName + " (Phred)"));									shellScript.append(getPhd2FastaCommand() + " -id . -os '" + seqName + "' -oq '" + seqName + ".qual'\n");									shellScript.append(getPhrapCommand() + " '" + seqName  + "' -new_ace " + StringUtil.blankIfNull(phrapOptions) + " \n");									if (runPhredPhrap)										shellScript.append(ShellScriptUtil.getWriteStringAsFile(statusFilePath, "" + sequenceCount + ". " + seqName + " (Phrap)"));									shellScript.append(ShellScriptUtil.getSetFileTypeCommand(sequenceDirPath + MesquiteFile.fileSeparator + "" + seqName + ".ace"));									//postShellScript.append(getSetFileTypeCommand(sequenceDirPath + MesquiteFile.fileSeparator + "" + seqName + processedACESuffix + ".ace"));									if (makeAceAliases) {										postShellScript.append(ShellScriptUtil.getLinkCommand(sequenceDirPath + MesquiteFile.fileSeparator + seqName + ".ace",  fragmentDirPath  + MesquiteFile.fileSeparator + "ace/" + seqName +".ace\n"));										postShellScript.append(ShellScriptUtil.getLinkCommand(sequenceDirPath + MesquiteFile.fileSeparator + seqName + processedACESuffix + ".ace",processedACEDirectory + "/" + seqName + processedACESuffix +".ace\n"));										//postShellScript.append("/Developer/Tools/setFile -t TEXT " +protectForUnix(fragmentDirPath)  + MesquiteFile.fileSeparator + "ace/'" + seqName +".ace'\n");									}									if (!StringUtil.blank(fragName))										echoStringToFile("Preparing " + seqName + "  ("+fragName+")", logBuffer);									else 										echoStringToFile("Preparing " + seqName, logBuffer);									numPrepared++;								}																newDir.mkdir(); //make new directory for this sequence																//now copy file to "originals" folder								if (backupOriginals) {									try { 										String pathInOriginalFolder = originalsDirPath + MesquiteFile.fileSeparator + chromFileName;  // path to where original will be stored										File originalFile = new File(pathInOriginalFolder); //										MesquiteFile.copy(cFile, originalFile);									}									catch (IOException e) {										logln( "Can't copy: " + seqName); 									}								}																//now move and rename the original to this sequence's directory								writeInfoFile(infoFilePath, fullSeqName);								try {									String newFileName = startToken+sampleCode.getValue()+"." + stLouisString.getValue()+primerName+fileExtension;									String newFilePath = sequenceDirPath + MesquiteFile.fileSeparator + newFileName;														File newFile = new File(newFilePath); //									int count=1;									while (newFile.exists()) {										newFileName = startToken+sampleCode.getValue()+"." + stLouisString.getValue()+primerName+count+fileExtension;										newFilePath = sequenceDirPath + MesquiteFile.fileSeparator + newFileName;										newFile = new File(newFilePath);										count++;									}									if (verbose) 										renameBuffer.append("  " + chromFileName + "  renamed to  " + newFileName + "\n");									fileNameTranslation[0][i] = newFileName;									fileNameTranslation[1][i] = chromFileName;									cFile.renameTo(newFile); 																	}								catch (SecurityException e) {									logln( "Can't rename: " + seqName);								}																															}							catch (SecurityException e) {								logln( "Can't make directory: " + seqName);							}						}					}				}			}						echoStringToFile("Number of files prepared: " + numPrepared, logBuffer);									if (verbose)				echoStringToFile("\n" + renameBuffer.toString()+"\n", logBuffer);									if (!abort) {				if (runPhredPhrap) {					shellScript.append(ShellScriptUtil.getRemoveCommand(runningFilePath));					shellScript.append(ShellScriptUtil.getRemoveCommand(statusFilePath));					if (openFastaDirectory)						postShellScript.append(ShellScriptUtil.getOpenDirectoryCommand(fragmentDirPath+ MesquiteFile.fileSeparator + processedFastaFolder + MesquiteFile.fileSeparator));  // so folder is opened up at the end//					if (openAceDirectory)//					postShellScript.append(getOpenDirectoryCommand(fragmentDirPath+ MesquiteFile.fileSeparator + "ace" + MesquiteFile.fileSeparator));  // so folder is opened up at the end				}								progIndicator.spin();				if (shellScript.length()>0){					//note name of file should end in .command for Terminal in OS X to run it; on other OS's we may be able to get the script to run directly from Runtime.exec					String scriptPath = rootDir + MesquiteFile.fileSeparator + "ppscript.bat";					MesquiteFile.putFileContents(scriptPath, shellScript.toString(), true);					if (runPhredPhrap) {						try {							//String scriptPath = StringUtil.protectForUnix(rootDir + MesquiteFile.fileSeparator + "ppscript.command");							ShellScriptUtil.setScriptFileToBeExecutable(scriptPath);							//logln("Request to change permissions of file \"ppscript.bat\" given");							Process proc = ShellScriptUtil.executeScript(scriptPath);							if (proc==null) {								if (progIndicator!=null) progIndicator.goAway();								return false;							}							if (progIndicator!=null)								progIndicator.setTitle("Running Phred & Phrap");							logln("\nExecution of Phred-Phrap command file begun.");							/*On OS X at this point the script operates asyncronously, so if we wanted to immediately go into the post processing we'd have to listen for some file showing up or such							 When runtime.exec runs the script directly, proc.waitFor() can be used to force this thread to hang until the script finishes*/														/* the following does nothing on OS X, but will be useful on other OS's if previous statement executes script directly so as to show processing in log window  							 InputStream stream = proc.getInputStream();							 int c = 0;							 while ((c=stream.read())!=-1){							 log("" + (char)c);							 }							 /**/						}						catch (IOException e){							logln("IOE error: " +  e);							abort =true;						}					}				}			}						if (runPhredPhrap && !abort) {				try {					String oldStatus = "";					int equalStatusCount = 0;					int count=-3;					logln("");					//if (progIndicator!=null) progIndicator.setTotalValue(sequenceCount);					while (MesquiteFile.fileExists(runningFilePath) && !abort){						//if (count>2) {						String status = "";						if (MesquiteFile.fileExists(statusFilePath)) {							status = MesquiteFile.getFileContentsAsString(statusFilePath);						}						if (!StringUtil.blank(status)) {							if (progIndicator!=null) {								if (progIndicator.getTotalValue()==0) {									progIndicator.setTotalValue(sequenceCount);									//progIndicator.repaintBar();								}								progIndicator.setText(status);								String s = StringUtil.getFirstItem(status,".");								int num = MesquiteInteger.fromString(s);//								Debugg.println("   s: " + s+ ",  num: " + num);								progIndicator.setCurrentValue(num);							}							if (status.equalsIgnoreCase(oldStatus))								equalStatusCount++;							else {								equalStatusCount = 0;								log(status);							}							if (equalStatusCount>25)								logln("There may be a problem: Phred and Phrap do not seem to be actively analyzing files.");							oldStatus = status;						}						else {							log(".");							if (progIndicator!=null) progIndicator.spin();						}						count = 1;												//}						if (progIndicator.isAborted())							abort = true;						Thread.sleep(200);						count++;					}					logln("");					if (abort)						echoStringToFile("Mesquite processing stopped.", logBuffer);					else						echoStringToFile("Phred & Phrap analyses completed.", logBuffer);				}				catch (InterruptedException e){					Thread.currentThread().interrupt();				}				//PostPhrap pPhrap = (PostPhrap)hireEmployee(commandRec, PostPhrap.class, "Post Phrap Processing");				if (!abort) {					echoStringToFile("Post-Phrap processing begins.", logBuffer);					postPhrapOnDirectory(commandRec, project,appendIfPossible, rootDir);					if (postShellScript.length()>0 && makeAceAliases){						//note name of file should end in .command for Terminal in OS X to run it; on other OS's we may be able to get the script to run directly from Runtime.exec						MesquiteFile.putFileContents(rootDir + MesquiteFile.fileSeparator + "postpp.bat", postShellScript.toString(), true);						if (runPhredPhrap) {							try {								String scriptPath = rootDir + MesquiteFile.fileSeparator + "postpp.bat";								ShellScriptUtil.setScriptFileToBeExecutable(scriptPath);   								//logln("Request to change permissions of file \"postpp.command\" given");								Process proc = ShellScriptUtil.executeScript(scriptPath);								if (proc==null) {									if (progIndicator!=null) progIndicator.goAway();									return false;								}								logln("Post-Phrap command file executed.");																/* the following does nothing on OS X, but will be useful on other OS's if previous statement executes script directly so as to show processing in log window  								 InputStream stream = proc.getInputStream();								 int c = 0;								 while ((c=stream.read())!=-1){								 log("" + (char)c);								 }								 /**/							}							catch (IOException e){								logln("IOE error: " +  e);							}						}					}				}			}						if (progIndicator!=null)				progIndicator.goAway();			MesquiteFile.putFileContents(rootDir + MesquiteFile.fileSeparator + "phredPhrapLog.txt", logBuffer.toString(), true);			if (project != null) {				//project.getCoordinatorModule().saveFile(project.getHomeFile());				if (originalData != null)					originalData.setDirty(true); //so that user will prompted; seems to be needed for annotations to be saved			}	}		return true;	}		private void addWindowsHeader(StringBuffer shellScript) {		shellScript.append("REGEDIT /E %Temp%.\\tmp-cygwin.reg \"HKEY_LOCAL_MACHINE\\SOFTWARE\\Cygnus Solutions\\Cygwin\\mounts v2\\/\"> nul\n");		shellScript.append("FOR /F \"tokens=1* delims==\" %%A IN ('TYPE %Temp%.\\tmp-cygwin.reg ^| FIND \"native\"') DO SET CYGWIN_ROOT=%%B> nul\n");		shellScript.append("SET CYGWIN_ROOT=%CYGWIN_ROOT:\"=%> nul\n");		shellScript.append("if exist %Temp%.\\tmp-cygwin.reg del %Temp%.\\tmp-cygwin.reg\n");		shellScript.append("SET PATH=.;%CYGWIN_ROOT%\\bin;%CYGWIN_ROOT%\\opt\\elinos\\bin;%PATH%\n");		shellScript.append("set PHRED_PARAMETER_FILE="+phredParamPath+"\n");	}		int aceFileCount = 0;	boolean processFolderAborted = false;	Vector goodNews = new Vector();	Vector badNews = new Vector();	/*.................................................................................................................*/	public void writeInfoFile(String infoFilePath, String fullName){		StringBuffer infoFileBuffer = new StringBuffer();		infoFileBuffer.append(fullName);  //here write the XML info file 		MesquiteFile.putFileContents(infoFilePath, infoFileBuffer.toString(), true);	}	/*.................................................................................................................*/	public void processInfoFile(String infoFilePath, MesquiteString fullName){		String s = MesquiteFile.getFileContentsAsString(infoFilePath);  //convert this so it is doing as XML file!		fullName.setValue(s);	}		/*.................................................................................................................*/	public int numFilesEndingWith(String directoryPath, String[] files, String ending){		int count=0;		for (int i=0; i<files.length; i++) { // going through the folders and finding the ace files			if (files[i]!=null ) {				String filePath = directoryPath + MesquiteFile.fileSeparator + files[i];				File cFile = new File(filePath);				if (cFile.exists()) {					if (!cFile.isDirectory()) {						if (files[i].endsWith(ending)) {							count++;						}					}				}			}		}		return count;	}			/*.................................................................................................................*/	public boolean processFolderAfterPhrap(MesquiteProject project, boolean appendIfPossible, File directory, String directoryPath, String enclosingDirName, String geneName, String dataFilePath, int level, CommandRecord commandRec){		boolean addingPhrapFailures = false;		if (processFolderAborted)			return false;		String[] files = directory.list();		String sequenceName = geneName;		if (level == 1)			sequenceName = directory.getName();		if (level == 1 && project!=null) {			taxa = null;  //making sure nothing old is remembered accidentally						//Wayne: getting ready to import; making matrices to be filled			if (appendIfPossible && project != null){				data = (DNAData)project.getCharacterData(directory.getName() + " (from Phred/Phrap)");				originalData = (DNAData)project.getCharacterData(directory.getName() + " (ORIGINAL from Phred/Phrap)");				qualityData = (ContinuousData)project.getCharacterData("Quality Scores for " + directory.getName()  + " from Phred/Phrap");				if (originalData != null)					originalData.saveChangeHistory = false;				if (qualityData != null)					 qualityData.saveChangeHistory = false;				if (data != null){					taxa = data.getTaxa();					data.saveChangeHistory = false;				}							}			if (taxa == null){				if (singleTaxaBlock) {					if (project.getNumberTaxas()>0)						taxa = project.getTaxa(0);					if (taxa == null) {						taxa = project.createTaxaBlock(0);						taxa.setName("Sequences");					}				}				else {					taxa = project.createTaxaBlock(0);					taxa.setName(directory.getName() + " Sequences");				}			}			CharactersManager manageCharacters = (CharactersManager)coord.findElementManager(CharacterData.class);			MesquiteString uid = new MesquiteString("phphImportID", MesquiteTrunk.getUniqueIDBase());			MesquiteString gN = new MesquiteString("geneName", directory.getName());						if (data == null){				data =  (DNAData)manageCharacters.newCharacterData(taxa, 0, "DNA Data");  //				data.saveChangeHistory = false;				data.addToFile(file, project, manageCharacters);  				data.setName(directory.getName() + " (from Phred/Phrap)");				data.attachIfUniqueName(uid);				data.attachIfUniqueName(gN);				data.attachIfUniqueName(new MesquiteString("phphImportMatrixType", "edited"));			}			if (originalData == null){				originalData =  (DNAData)manageCharacters.newCharacterData(taxa, 0, "DNA Data");  //				originalData.saveChangeHistory = false;				originalData.setEditorInhibition(true);				originalData.addToFile(file, project, manageCharacters);  				originalData.setName(directory.getName() + " (ORIGINAL from Phred/Phrap)");				data.addToLinkageGroup(originalData); //link matrices!				originalData.attachIfUniqueName(uid);				originalData.attachIfUniqueName(gN);				originalData.attachIfUniqueName(new MesquiteString("phphImportMatrixType", "original"));			}			if (qualityData == null){				qualityData =  (ContinuousData)manageCharacters.newCharacterData(taxa, 0, "Continuous Data");  //				qualityData.saveChangeHistory = false;				qualityData.addToFile(file, project, manageCharacters);  				data.addToLinkageGroup(qualityData); //link matrices!				qualityData.setName("Quality Scores for " + directory.getName()  + " from Phred/Phrap");				qualityData.attachIfUniqueName(uid);				qualityData.attachIfUniqueName(gN);				qualityData.attachIfUniqueName(new MesquiteString("phphImportMatrixType", "quality"));			}			maxChar.setValue(0);					}		String processedAceFilePath = null;		AceFile ace = null;		int currentRead = -1;		for (int i=0; i<files.length && !processFolderAborted; i++) { // going through the folders and finding the ace files			if (files[i]!=null ) {				String filePath = directoryPath + MesquiteFile.fileSeparator + files[i];				String infoFilePath = directoryPath + MesquiteFile.fileSeparator + infoFileName;				File cFile = new File(filePath);				if (cFile.exists()) {					if (cFile.isDirectory()) {						processFolderAfterPhrap(project, appendIfPossible, cFile, filePath, directory.getName(), sequenceName, dataFilePath, level+1, commandRec);						//log(".");					}					else {						if (files[i].endsWith(".ace") && !addingPhrapFailures) {							aceFileCount++;							if (progIndicator!=null) {								if (progIndicator.getTotalValue()==0) {									progIndicator.setTotalValue(sequenceCount);									//progIndicator.repaintBar();								}								progIndicator.setText("Processing: " + files[i] + " ("+ geneName+")");								progIndicator.setCurrentValue(aceFileCount);								if (progIndicator.isAborted()) { 									processFolderAborted = true;									return false;								}							}							echoStringToFile("Processing ACE file: " + files[i], logBuffer);     							String baseName = files[i].substring(0,files[i].length()-4);  //this is the name of the sequence							processedAceFilePath = directoryPath + MesquiteFile.fileSeparator + baseName+processedACESuffix+".ace";							ace = new AceFile(filePath,processedAceFilePath, dataFilePath, this, commandRec,processPolymorphisms, polyThreshold);							ace.setBaseName(baseName);							MesquiteString fullName = new MesquiteString(baseName);							processInfoFile(infoFilePath, fullName);							String fragmentDirPath = StringUtil.getAllButLastItem(StringUtil.getAllButLastItem(directoryPath,MesquiteFile.fileSeparator ),MesquiteFile.fileSeparator );															ace.setLongSequenceName(fullName.toString());							if (ace.getNumContigs()>=1) {								log(ace.contigListForLog()+StringUtil.lineEnding());								if (processPolymorphisms)									ace.processPolys(commandRec);  //creates an additional CO that has polys in it								if (renameContigsInAceFiles)									ace.renameContigs(fullName.toString(), addFragName, geneName);								ace.setLowQualityToLowerCase(commandRec,qualThresholdForLowerCase); 								if (truncateMixedEnds)									ace.trimMixedEnds(mixedEndThreshold, mixedEndWindow, qualThresholdForTrim);//								if (unTrimAceReads)//								ace.unTrimQA();																MesquiteFile.putFileContents(fragmentDirPath  + MesquiteFile.fileSeparator + processedFastaFolder + MesquiteFile.fileSeparator + baseName+".fas", ace.toFASTAString(commandRec,processPolymorphisms, qualThresholdForTrim), true);								MesquiteFile.putFileContents(processedAceFilePath, ace.toString(commandRec,processPolymorphisms), true);																ace.setNameTranslation(fileNameTranslation);								if (project != null) {									ace.importSequence(taxa, data, originalData, qualityData, singleTaxaBlock, aceRef, processPolymorphisms, maxChar);								}								goodNews.addElement("    (" + geneName + ") -- " + fullName);							}							else {								echoStringToFile("   ACE file contains no contigs!\n", logBuffer); 								badNews.addElement("    (" + geneName + ") -- " + baseName);								if (addPhrapFailures && project !=null) {									addingPhrapFailures = true;									i=0;									ace.createEmptyContig(numFilesEndingWith(directoryPath,files,".phd.1"));  //create an empty contig									if (renameContigsInAceFiles)										ace.renameContigs(fullName.toString(), addFragName, geneName);								}							}														if (!addingPhrapFailures)								ace.dispose();						}						else if (files[i].endsWith(".phd.1") && addingPhrapFailures) {							currentRead++;							ace.addPhdFileAsRead(currentRead, directoryPath, files[i], processPolymorphisms, polyThreshold);						}					}				}			}		}		if (addingPhrapFailures && ace !=null ) {  // have to process AceFile that we have manually made			MesquiteFile.putFileContents(processedAceFilePath, ace.toString(commandRec,processPolymorphisms), true);			ace.setNameTranslation(fileNameTranslation);			if (project != null)				ace.importSequence(taxa, data, originalData, qualityData, singleTaxaBlock, aceRef, processPolymorphisms, maxChar);			ace.dispose();		}		if (level ==1 && project !=null){ 			//Wayne: cleaning up and showing matrix			if (maxChar.getValue()<data.getNumChars()-1) {//				Debugg.println("maxChar: " + maxChar.getValue() + ", data.getNumChars(): " + data.getNumChars());				data.deleteCharacters(maxChar.getValue()+1, data.getNumChars()-maxChar.getValue()+1, false);				originalData.deleteCharacters(maxChar.getValue()+1, originalData.getNumChars()-maxChar.getValue()+1, false);				qualityData.deleteCharacters(maxChar.getValue()+1, qualityData.getNumChars()-maxChar.getValue()+1, false);			}			/*for (int ic = 0; ic<data.getNumChars(); ic++){			 for (int it=0; it<data.getNumTaxa(); it++){			 originalData.setState(ic, it, data.getState(ic, it));			 }			 }*/			data.showMe();  			DataWindowMaker mb = coord.findCharacterDataEditor(data);						if (mb != null) {				String commands;				if (showBirdsEye)					commands = "getWindow; tell It; setSize 700 400; colorCells  #mesquite.chromaseq.ColorQuality.ColorQuality;  toggleBirdsEye on;  showWindow; endTell;";  				else					commands = "getWindow; tell It; setSize 700 400; colorCells  #mesquite.chromaseq.ColorQuality.ColorQuality;  showWindow; endTell;";  				Puppeteer p = new Puppeteer(this);				MesquiteInteger pos = new MesquiteInteger(0);				p.execute(mb, commands, pos, "", false);			}			data = null;			originalData = null;			qualityData = null;					}		return false;	}	/*.................................................................................................................*/	public boolean postPhrapOnDirectory(CommandRecord commandRec, MesquiteProject project, boolean appendIfPossible, String directoryPath){		if (StringUtil.blank(directoryPath))			return false;		File directory = new File(directoryPath);		if (directory.exists() && directory.isDirectory()) {			logln(" ");			String dataFilePath = null;			if (project != null) {				coord = project.getCoordinatorModule();				file = project.getHomeFile();				file.changeLocation(importedDirectoryPath, importedDirectoryName + ".nex");				dataFilePath = importedDirectoryPath;			}			aceFileCount = 0;			processFolderAborted = false;			if (progIndicator!=null)				progIndicator.setTitle("Processing Phred/Phrap output");						goodNews.removeAllElements();			badNews.removeAllElements();						processFolderAfterPhrap(project, appendIfPossible, directory, directoryPath,"", null, dataFilePath, 0, commandRec);						logln("");			echoStringToFile("", logBuffer);			echoStringToFile("========================", logBuffer);			echoStringToFile("Phred/Phrap processing report", logBuffer);			echoStringToFile("", logBuffer);			if (goodNews.size()>0){				echoStringToFile("Sequences obtained from the following genes and samples:", logBuffer);				for (int i=0; i<goodNews.size(); i++)					echoStringToFile("   " + (String)goodNews.elementAt(i), logBuffer);			}			if (badNews.size()>0){				echoStringToFile("Sequences FAILED to be obtained from the following genes and samples:", logBuffer);				for (int i=0; i<badNews.size(); i++)					echoStringToFile("   " + (String)badNews.elementAt(i), logBuffer);			}			echoStringToFile("========================", logBuffer);						logln("");			logln("Completed.");		}		return false;	}	/*.................................................................................................................*/	public boolean postPhrap(MesquiteProject project, CommandRecord commandRec){		String directoryPath = MesquiteFile.chooseDirectory("Choose directory containing Phrap files:"); 		logBuffer.setLength(0);		return postPhrapOnDirectory(commandRec,project, false, directoryPath);	}	/*.................................................................................................................*/	public String getName() {		return "Prepare and Run Phred and Phrap";	}	/*.................................................................................................................*/	public boolean showCitation() {		return false;	}		/*.................................................................................................................*/	public String getExplanation() {		return "Prepares a folder of abi files for Phred and Phrap, and makes a shell script to invoke Phred and Phrap within each folder, and runs the script.";	}	/*.................................................................................................................*/	public  void actionPerformed(ActionEvent e) {		if (e.getActionCommand().equalsIgnoreCase("phLocations")) {			queryFileLocations();		}		else if (e.getActionCommand().equalsIgnoreCase("postOptions")) {			queryPostOptions();		}		else if (e.getActionCommand().equalsIgnoreCase("editNameParsersButton")) {			if (nameParserManager!=null)				nameParsingRule = nameParserManager.chooseNameParsingRules(nameParsingRule);		}		else if (e.getActionCommand().equalsIgnoreCase("phBrowse")) {			phredPath = MesquiteFile.chooseDirectory("Choose directory containing phred, phrap, and phd2fasta: ");			if (!StringUtil.blank(phredPath)) {				if (!phredPath.endsWith(MesquiteFile.fileSeparator))					phredPath+=MesquiteFile.fileSeparator;				phredPathField.setText(phredPath);			}		}		else if (e.getActionCommand().equalsIgnoreCase("primerBrowse")) {			MesquiteString primerListDir = new MesquiteString();			MesquiteString primerListFile = new MesquiteString();			String s = MesquiteFile.openFileDialog("Choose file containing primer list", primerListDir, primerListFile);			if (!StringUtil.blank(s)) {				primerListPath = s;				if (primerListField!=null) 					primerListField.setText(primerListPath);			}		}		else if (e.getActionCommand().equalsIgnoreCase("DNANumbersBrowse")) {			MesquiteString dnaNumberListDir = new MesquiteString();			MesquiteString dnaNumberListFile = new MesquiteString();			String s = MesquiteFile.openFileDialog("Choose file containing sample codes and names", dnaNumberListDir, dnaNumberListFile);			if (!StringUtil.blank(s)) {				sampleCodeListPath = s;				if (dnaCodesField!=null) 					dnaCodesField.setText(sampleCodeListPath);			}		}		else if (e.getActionCommand().equalsIgnoreCase("paramBrowse")) {			MesquiteString paramDir = new MesquiteString();			MesquiteString paramFile = new MesquiteString();			phredParamPath = MesquiteFile.openFileDialog("Choose phred parameter file: ", paramDir, paramFile);			if (!StringUtil.blank(phredParamPath)) {				paramPathField.setText(phredParamPath);			}		}	}			public void itemStateChanged(ItemEvent e) {		if (e.getItemSelectable() == nameRulesChoice){			getNameRuleFromChoice();		}	}}