<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><!-- InstanceBegin template="/Templates/MesquiteTemplate.dwt" codeOutsideHTMLIsLocked="false" -->
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />


<!-- InstanceBeginEditable name="doctitle" -->
<title>Ancestral States</title>
<!-- InstanceEndEditable -->

<![endif]--><!--[if IE]>
<style type="text/css"> 
/* place css fixes for all versions of IE in this conditional comment */
.thrColFixHdr #sidebar2, .thrColFixHdr #sidebar1 { padding-top: 30px; }
.thrColFixHdr #mainContent { zoom: 1; }
/* the above proprietary zoom property gives IE the hasLayout it needs to avoid several bugs */
</style>
<![endif]-->

<style type="text/css">
<!--
-->
</style>
<link href="CSS/mesquitePackage.css" rel="stylesheet" type="text/css" />
<link href="SpryAssets/SpryCollapsiblePanel.css" rel="stylesheet" type="text/css" />

<script src="SpryAssets/SpryCollapsiblePanel.js" type="text/javascript"></script>
<script type="text/javascript" src="SpryAssets/SpryURLUtils.js"></script>
<script type="text/javascript" src="javascript/mesquite.js"></script>

<!-- InstanceBeginEditable name="head" -->
<!-- InstanceEndEditable -->
</head>

<body bgcolor="#ffffff" class="thrColFixHdr" >

<div id="container">
  <div id="header">
    <table width="100%" border="0" cellpadding="4">
  <tr>
    <td width="51%"><strong id="PackageTitle">Mesquite</strong></td>
    <td width="33%" align="right"><div id="subtitle"><!-- InstanceBeginEditable name="subtitle" -->Ancestral States<!-- InstanceEndEditable --></div></td>
    <td width="2%">&nbsp;</td>
    <td width="14%">Version 3.6<br />
      27 December 2018<br /></td>
  </tr>
</table>

</div>
  
  
<div id="sidebar1">
  
  
  
  
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('http://www.mesquiteproject.org')">Mesquite Website Home</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('home.html')">Introduction</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('Installation.html')">Download & Installation</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('History%20of%20New%20Features.html')">Recent Changes</a>
  </div>

  <div id="OverviewOfMesquitePanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Overview of Mesquite ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('What%20Mesquite%20does.html');">What Mesquite Does</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Mesquites%20Features.html');">List of Features</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Development%20Team.html');">Development Team</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Contact%20Us.html');">Contact Us</a></td></tr>
		</table>
    </div>
  </div>
  <div id="GettingStartedPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Getting Started ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Installation.html');">Download and Installation</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Additional%20Mesquite%20Packages.html');">Additional Packages</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Beginning%20Mesquite.html');">Beginning Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Example%20Files.html');">Example Files</a></td></tr>
		</table>
    </div>
  </div>
  <div id="BasicMesquiteOperationPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Basic  Operations ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Files.html');">Files</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Menus.html');">Menus</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Windows.html');">Windows</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Charts.html');">Charts</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Selection.html');">Selection</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Modules.html');">Modules</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simplifying%20the%20Interface.html');">Simplifying Mesquite</a></td></tr>
		</table>
    </div>
  </div>
  <div id="TaxaTreesCharactersPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Taxa, Trees, Characters ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Characters%20&%20Matrices.html');">Characters &amp; Matrices</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Taxa.html');">Taxa</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Trees.html');">Trees</a></td></tr>
		</table>
    </div>
  </div>
  <div id="AnalysesPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Analyses ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Character%20Evolution.html');">Character Evolution</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Align.html');">Sequence Alignment</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Analyzing%20Molecular%20Data.html');">Molecular Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Diversification.html');">Diversification</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Gene%20Trees.html');">Gene Trees</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simulations%20&%20Randomizations.html');">Simulations &amp; Randomizations</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Analyzing%20Continuous%20Characters.html');">Continuous Characters</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Studies%20of%20Tree%20Topology.html');">Studies of Tree Topology</a></td></tr>
		</table>
    </div>
  </div>
  <div id="WorkflowsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Workflows ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Managing%20Molecular%20Data.html');">Managing Molecular Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Managing%20Continuous%20Data.html');">Managing Continuous Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Interactions%20with%20Other%20Programs.html');">Interactions with other programs</a></td></tr>
		</table>
    </div>
  </div>
  <div id="GettingHelpPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Getting Help ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Learning%20how%20to%20use%20Mesquite.html');">Learning Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simplifying%20the%20Interface.html');">Simplifying Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Bugs%20and%20Troubleshooting.html');">Bugs &amp; Troubleshooting</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Getting%20Help.html');">Getting Help</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Suggesting%20New%20Features.html');">Your Suggestions</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Glossary.html');">Glossary</a></td></tr>
  			<tr><td><a href="javascript:pageLink('FAQ.html');">FAQ</a></td></tr>
		</table>
    </div>
  </div>
  <div id="TechnicalDetailsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Technical Details ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('How%20Mesquite%20Works.html');">How Mesquite Works</a></td></tr>
  			<tr><td><a href="javascript:pageLink('MesquiteLogSupport.html');">Log & Support Files</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Scripts%20&%20Macros.html');">Scripts &amp; Macros</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Source%20Code.html');">Source Code</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Developers%20Documentation.html');">Developer's Documentation</a></td></tr>
		</table>
    </div>
  </div>
  <div id="HistoryPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">History ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('History%20of%20Mesquite.html');">History of Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('whyMesquite.html');">Why Mesquite?</a></td></tr>
  			<tr><td><a href="javascript:pageLink('History%20of%20New%20Features.html');">New Features & Bug List</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Release%20Dates.html');">Release Dates</a></td></tr>
		</table>
    </div>
  </div>
  <div id="PublishingResultsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Publishing Results ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Documenting%20Your%20Work.html');">Documenting your work</a></td></tr>
  			<tr><td><a href="javascript:pageLink('How%20to%20Cite%20Mesquite.html');">Citing Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Publishing%20Results.html');">Publishing results</a></td></tr>
		</table>
    </div>
  </div>

  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('Acknowledgements.html')">Acknowledgments</a>
  </div>

<h3>&nbsp;</h3>
  <!-- end #sidebar1 --></div>

  <div id="mainContent">
    
<!-- InstanceBeginEditable name="content" -->
    
    
<h1 id="toc0"><a name="Studying the History of Character Evolution"></a>Studying the History of Character Evolution</h1>
 With a phylogenetic tree and a distribution of character states in the observed (terminal) taxa, Mesquite can attempt to reconstruct the character states at ancestral nodes. Two separate issues to consider are the method by which the reconstruction is done, and how its results are displayed to the user. Mesquite currently can use either parsimony, likelihood or Bayesian methods to reconstruct ancestral states, and has several display methods, including &quot;Trace Character History&quot; which paints the branches of the tree to show the reconstruction.<br />
We recommend highly that you examine the example files provided in the folder &quot;Ancestral State Examples&quot;. The minimal configuration to use with these examples is &quot;Ancestral States&quot; (indicate this configuration under File&gt;Activate/Deactivate Packages&gt;Choose Configuration), but you can also leave Mesquite in its default All Installed Modules mode.<br />
<div style="text-align: center"><img src="files/ancStates.gif" alt="ancStates.gif" title="ancStates.gif" /></div><br />
<h3 id="toc1"><a name="Studying the History of Character Evolution--Trace Character History"></a><a name="tch"></a>Trace Character History</h3>
 The Trace Character History facility graphically represents a history of character evolution on the tree. It is available under the Analysis:Tree menu of a tree window (e.g., the basic Tree Window, Dependent Tree Window, Mirror Tree Window, Multitree Window). If you select this you will probably be asked for a source of characters (e.g., stored characters) and areconstruction method (e.g., <a  href="Ancestral%20States.html#parsimony">parsimony</a>, <a  href="Ancestral%20States.html#likelihood">likelihood</a>, <a  href="Ancestral%20States.html#scm">stochastic character mapping</a>). (If you have &quot;Use Stored Characters/Matrices by Default&quot; turned on in the Defaults submenu if the File menu, Mesquite won't ask you and will simply use Stored Characters.) The tree will be painted to show ancestral states, and a trace legend will appear. The Trace Legend contains an important text area that gives details of the current ancestral state tracing. You can also see details of the reconstruction by switching the window to Text mode using the menu item Window&gt;View Modes&gt;Text.<br />
<br />
For categorical and molecular data, you can change the colors used in Trace Character by double clicking on the color rectangle in the Trace Legend. Revert to Default Colors is available in the Trace menu.<br />
<br />
For parsimony reconstructions, any tree drawing style will suffice. For likelihood reconstructions, we recommend the Balls&amp;Sticks style (Display menu, Tree Form) with Line Style &quot;Square&quot;. This permits you to see the relative likelihoods and branch lengths. For stochastic character mapping, we recommend the Square Tree style in order to display the changes within a branch.<br />
<br />
The Trace menu gives menu items to control the character history and its display. Some important ones are:<br />
<strong>Character history source</strong>: Typically you may use Trace Character History to reconstruct the ancestral states of an observed character. Alternatively, you can trace a simulated history (&quot;Simulate Ancestral States&quot;). The reconstructed states need not be based on actual data, but could be based on simulated data. &quot;Simulate Ancestral States&quot; shows the &quot;actual&quot; history of character evolution branch by branch as it occured in the <a  href="Simulations%20%26%20Randomizations.html">simulation</a>, not as it was reconstructed, and thus may show ancestral states that would be unreconstructable, obliterated by subsequent changes.<br />
<br />
<strong>Next, Previous, Choose Character History</strong>: Usually, this will allow you to choose which character to view. You can also scroll through characters using the blue arrows in the trace legend.<br />
<br />
<strong>Trace Display Mode</strong>: With Trace&gt;Trace Display Mode&gt;Shade States and Trace&gt;Trace Display Mode&gt;Label States you choose whether to see the branches painted, or have the states indicated in labels. If painted, you can also ask that states be indicated by labels by choosing Trace&gt;Label States.<br />
<br />
You can also see details of the reconstruction at a node by holding the cursor over the branch. A description of the reconstructed states will appear at the bottom part of the Trace Character Legend. Another method is to use the Text view of the window (using the Window&gt;View Modes&gt;Text menu item) and scroll down — a text version of the trace should appear.<br />
<br />
Reconstruction Method: For more details on reconstruction methods, see the sections on <a  href="Ancestral%20States.html#parsimony">parsimony</a>, <a  href="Ancestral%20States.html#likelihood">likelihood</a>, and <a  href="Ancestral%20States.html#scm">stochastic character mapping</a>.<br />
<h3 id="toc2"><a name="Studying the History of Character Evolution--Trace All Characters"></a><a name="tac"></a>Trace All Characters</h3>
 Trace All Characters summarizes ancestral state reconstructions of many characters simultaneously. To request it, choose Choose (Tree Window)Analysis:Tree&gt;Trace All Characters. A text window like that shown below will appear, listing the ancestral states reconstructed at each node for each character. Node numbers show up in red on the tree. (Alternatively, spots showing node numbers in the figure below can be turned on in the Tree Window's Display menu by selecting Show Node Numbers.)<br />
<div style="text-align: center"><img src="files/traceAll.gif" alt="traceAll.gif" title="traceAll.gif" /></div><br />
<span style="display: block; text-align: left;">By default only the selected nodes are listed. (Nodes can be selected using tools in the Tree Window.) You can request to show all nodes by turning off Show Selected Nodes Only in the Trace_All menu. By default all characters are listed; this can be changed using the Show Selected Characters Only menu item.</span><br />
<span style="display: block; text-align: left;">The ancestral state reconstruction can be controlled in the Trace_All menu of the tree window.</span><br />
<span style="display: block; text-align: left;">The table is either listed by characters or by nodes; you can switch from one to the other using the Rows are Characters menu item</span><br />
<span style="display: block; text-align: left;">Columns in the table in the text window may not appear perfectly aligned, but it is presented as a tab-delimited table, so you should be able to copy the text and paste it in to a text file to read in to your favorite spreadsheet program.</span><br />
<h3 id="toc3"><a name="Studying the History of Character Evolution--Trace Character Over Trees"></a><a name="tcot"></a>Trace Character Over Trees</h3>
 The Trace Character Over Trees facility summarizes ancestral state reconstructions over a series of trees. This is useful to understand how ancestral state reconstructions vary over a series of trees, for instance if there is uncertainty in the tree. It works for categorical characters only. Also, Trace Character Over Trees does NOT calculate a consensus tree for you. As with all other analyses in the Tree Window, it works with the tree that is given to it by the Tree Window. If you want to make your summary on a consensus tree, then you need to put the consensus tree into the Tree Window first and then request Trace Character Over Trees.<br />
<br />
Choose (Tree Window)Analysis:Tree&gt;Trace Character Over Trees. This examines a series of trees, and for each examines a character's ancestral states on that tree. For each node in the tree in the tree window, it attempts to summarize what ancestral states are reconstructed for that same clade in the series of trees (as long as the same clade exists in the other trees). For example, imagine the tree in the tree window includes the clade Tetrapoda. Each of the series of trees is examined, and if that tree includes the clade Tetrapoda, then its reconstructed ancestral states are examined. If the tree doesn't include Tetrapoda, then it is ignored for the sake of summarizing the tetrapod ancestral states. The tree in the tree window is then decorated to summarize what ancestral states are reconstructed for each of the clades.<br />
Here is an example of Trace Character Over Trees in action:<br />
<div style="text-align: center"><img src="files/tcot.gif" alt="tcot.gif" title="tcot.gif" /></div><br />
The cursor is over a node (the most recent common ancestor of carinatum and coxendix), and thus the legend shows a summary. The node (i.e., the clade it represents) is present in only 445 of the 545 trees examined. For this reason, 100/545 or 18.3% of the pie chart for that node is shown in red, as the node is not present in that fraction of the trees. In addition, 100 of the trees with that node have an equivocal reconstruction at that node; those trees are shown in gray in the pie chart. Of the 345 trees with the node and an unequivocal reconstruction, 321 trees have &quot;isodiametric&quot; reconstructed at the node (shown in white), and in 24 trees have state &quot;slightly trans.&quot; at that node (shown as a very thin sliver of green). In this example, &quot;Count Trees with Uniquely Best States&quot; is selected in the Calculate submenu of the Trace_Over_Trees menu. With this option a tree is counted as having a state at a node only if the state is the only optimal state. What is considered optimal depends on the reconstruction method. With parsimony, states are considered equally optimal if they are equally parsimonious. With likelihood, the Decision Threshold is used to decide whether states are good enough to be considered within the optimal set.<br />
<br />
There are two alternatives to &quot;Count Trees with Uniquely Best States&quot;. One is &quot;Count All Trees with State&quot;. With this option for counting, a tree is counted as having a state at a node if the state is within the optimal set, whether or not there are other states within the optimal set. When the &quot;Count All Trees with State&quot; option is used, the sum of tree counts for the states at a node can more than the total number of trees with the clade, for a tree can get counted multiply at a node, under each state in an equivocal assignment. The other calculation option is &quot;Average Frequencies Across Trees&quot;. This option is only available for reconstruction methods such as likelihood that yield a frequency or probability for each state at each node. The value presented for a node for a state is then the average frequency of that state across all of the trees possessing that node. All trees with the node are included in this calculation, even if the frequencies are very low.<br />
An important option is what trees to examine. If the tree in the tree window is a consensus tree, then the trees examined might be the original set of most parsimonious trees that built the consensus. Trace Character Over Trees could then show how the ancestral state reconstruction varies among the most parsimonious trees. The trees examined might also be derived from a Bayesian analysis (perhaps subsampled using the Sample Trees from Separate File tree source), and the ancestral states obtained by likelihood, to do an analysis in Lutzoni's style. The trees might be random resolutions of an unresolved tree, or trees with random noise added to branch lengths, and so on. This would allow you to see how the results would vary if the tree changed. See the submenu Trace_Over_Trees&gt;Tree Source for options.<br />
<h3 id="toc4"><a name="Studying the History of Character Evolution--Summarize State Changes Over Trees"></a><a name="SCOT"></a>Summarize State Changes Over Trees</h3>
 The Summarize State Changes Over Trees facility summarizes ancestral state changes over a series of trees. To examine the changes over entire trees, choose Summarize State Changes Over Trees in the Taxa&amp;Trees menu. To examine the changes just within one clade, select the single ancestral branch of that clade (not the entire clade) in a tree window, and choose Summarize Changes in Selected Clade in the Analysis:Tree menu. You will be asked to choose a reconstruction method, as well as the trees to examine. The State Changes window will appear, that displays a summary of the minimum, maximum, and average number of changes of each sort, across the trees. It also displays the fraction of trees that is reconstructed to have a particular number of each sort of change.<br />
<div style="text-align: center"><img src="files/SCOT.jpg" alt="SCOT.jpg" title="SCOT.jpg" style="height: 441px; width: 800px;" /></div><span style="display: block; text-align: center;"><br />
</span><br />
For reconstruction methods that allow for multiple mappings of character reconstruction for a character on a tree (multiple MPRs for parsimony reconstruction of unordered and ordered characters; stochastic character mapping), then for each tree, a number of mappings will be examined for each tree, with each mapping contributing 1/(number of mappings examined) to the frequency calculations. For MPRs, all MPRs will be examined up to a limit you set; if the limit you set is less than the number of MPRs, the MPRs will be randomly sampled from among the total. If the total number of MPRs for the whole tree exceeds 2 to the power 63, then Mesquite will not make the calculation, and will give you a warning. For stochastic character mapping, the number of mappings examined will need to be specified.<br />
If you ask Mesquite to sample MPRs, then it will consider MPRs distinct if they differ anywhere in the entire tree. For example, imagine you ask Meqsuite to Summarize Changes in Selected Clade, and that there are only two MPRs within that clade, but there are other regions of the tree outside of that clade that show ambiguity in the most parsimonious reconstructions, yielding a total of not two but 1000 MPRs over the entire tree. Then, when Mesquite summarizes changes within the selected clade, it will consider the changes within that clade across all 1000 MPRs. Future versions of Mesquite will provide the option to only consider MPRs distinct if they differ within the clade of interest.<br />
For reconstruction methods that do not allow for multiple mappings of character reconstructions, and instead use equivocal (e.g, states 0 or 1) to an ancestral node, for example parsimony for parsimony models other than unordered and ordered, or standard likelihood, then Mesquite examines the single (potentially ambiguous) reconstruction, and counts only unambiguous changes.<br />
You can constrain the mappings examined by specifying changes to avoid, using the Allow Changes menu in the Summarize_Changes menu. If you deselect 0 to 1 changes in the dialog box, then any mapping that contain a 0 to 1 change will not be counted.<br />
<h3 id="toc5"><a name="Studying the History of Character Evolution--Parsimony Reconstruction Methods"></a><a name="parsimony"></a>Parsimony Reconstruction Methods</h3>
 Parsimony reconstruction methods find the ancestral states that minimize the number of steps of character change given the tree and observed character distribution. They can use different assumptions (models of evolution). For categoricalcharacters, the unordered states assumption is that one step is counted for any change. The ordered states assumption is that the number of steps from state i to state j is |i-j|. Thus, the number of steps from state 2 to state 5 is 3 steps. Astepmatrix explicitly specifies the number of steps from state to state by a matrix. Mesquite does not yet do parsimony calculations for irreversible, Dollo and character state tree assumptions, although these models are listed in menus and you can assign them to particular characters. For continuous characters, the linear cost assumption is that the cost of a change from state x to state y is |x-y|. The squared change assumption is that the cost of a change from state x to state y is (x-y) squared.<br />
Mesquite's parsimony calculations attempt to match MacClade's. Some differences remain in special cases of polymorphic terminal taxa with stepmatrices. Mesquite allows hard polytomies in the tree when stepmatrices are used.<br />
<a name="parsModel"></a><strong>Assigning a parsimony model</strong>: The parsimony model used for a character's calculations is the model assigned to it, if the character is one stored in in a matrix in a file. A parsimony model can be assigned in the List of Characters window. Select the row(s) corresponding to the desired character(s), and then touch on the column heading &quot;Parsimony Model&quot;. A drop-down menu contains a submenu that allows you to select the models to apply. You can also change the parsimony model assigned to the character being traced in Trace Character History using the Parsimony Model submenu of the Trace menu. (Recall that Mesquite cannot yet do calculations with irreversible and Dollo models.)<br />
If the characters used in parsimony reconstruction are not stored in a matrix but rather come directly from another source of characters such as simulations, a single parsimony model can be chosen to be applied to all of the characters coming from this source. Thus, for instance, when using Trace Character History, the Parsimony Model submenu of the Trace menu can be used to assign the model to be used.<br />
<a name="stepmatrices"></a><strong>Creating and editing stepmatrices</strong>: To create a stepmatrix, select Characters&gt;New Character Model&gt;Stepmatrix. A window will appear in which you can edit the cost of i to j transitions. The number of states allowed is initially 10 (0 through 9), but you can change the number of states under (Edit Stepmatrix)&gt;Step_matrix&gt;Set maximum state. The maximum number of states for a categorical character is 56; the maximum state value is therefore 55. This stepmatrix editor does not do triangle inequality checking (see discussion in manual of MacClade, which does check the triangle inequality).<br />
<a name="parsimonyStatistics"></a><strong>Parsimony statistics</strong>: The parsimony calculations are used also for Treelength, Character Steps, Consistency Index, andRetention Index. These statistics are available in many places. For example, for the current tree in the tree window, go to Analysis:Tree&gt;Values for Current Tree. To see the statistics for individual characters, you can prepare in the tree window the tree on which you want the statistics calculated, then go to the List of Characters window and choose Column&gt;Number for Character&gt;Character Value with current tree.... In the dialog box that appears choose the statistic you want. You can also access these calculations from the charts.<br />
<a name="mprs"></a><strong>Most Parsimonious Reconstructions (MPRs):</strong> There are often multiple reconstructions of character evolution that entail the same number of steps, and are most parsimonious. These Most Parsimonious Reconstructions (MPRs) can now be individually examined for discretely valued characters (categorical, DNA, RNA, protein) if the parsimony model is unordered or ordered. With character history traced, then choosing &quot;MPRs Mode&quot; in the Trace menu will change the tracing to show only one MPR at a time. You can then examine other MPRs using the blue triangles next to &quot;MPRs&quot; in the Trace Legend. The number of MPRs can also be calculated using the &quot;Number of MPRs&quot; item in the Trace menu.<br />
<h3 id="toc6"><a name="Studying the History of Character Evolution--Likelihood Reconstruction Methods"></a><a name="likelihood"></a>Likelihood Reconstruction Methods</h3>
 Likelhood reconstruction methods find the ancestral states that maximize the probability the observed states would evolve under a stochastic model of evolution (Schluter et al., 1997; Pagel, 1999). The likelihood reconstruction finds, for each node, the state assignment that maximizes the probability of arriving at the observed states in the terminal taxa, given the model of evolution, and allowing the states at all other nodes to vary. (In fact, this considers all possible assignments to the other ancestral states.) This is equivalent to the marginal reconstruction of Swofford's PAUP*, or the Fossil Likelihood reconstruction of Pagel's Discrete.<br />
<br />
You can use likelihood for the reconstruction by selecting &quot;Likelihood Ancestral States&quot; when first requesting Trace Character History, or from the Method submenu of the Trace menu after Trace Character History is already active. When using Likelihood Ancestral States in Trace Character History, it is recommended that you use a Tree Form for the drawing that uses spots at the nodes (for example, (Tree Window)Display&gt;Tree Form&gt;Balls &amp; Sticks). These spots at the nodes will indicate relative likelihoods with pie diagrams as in Schluter et al. 1997.<br />
<br />
<a name="likeModels"></a>At present only categorical characters are supported by the likelihood calculations. Models for DNA and protein evolution are not yet available for use by likelihood. Two models of evolution are currently supported, the Mk1 model and the AsymmMk model.<br />
<ul><li><a name="mk1"></a>Mk1 model (&quot;Markov k-state 1 parameter model&quot;) is a k-state generalization of the Jukes-Cantor model, and corresponds to Lewis's (2001) Mk model. The single parameter is the rate of change. Any particular change (from state 0 to 1 or state 3 to 2, for example) is equally probable. Mesquite's rate of change parameter is equivalent to the q values of Pagel's Multistate program when the q's are constrained to be equal. Thus for a character with three states 0, 1 and 2, the Mk1 model would have an instantaneous rate matrix of the following form:</li></ul><br />
<div style="text-align: center"><img src="files/markovTable1.png" alt="markovTable1.png" title="markovTable1.png" style="height: 107px; width: 258px;" /></div><ul class="quotelist"><li><br />
</li></ul><ul><li><a name="asymmmk"></a>AsymmMk model (&quot;Asymmetrical Markov k-state 2 parameter model&quot;) has two parameters: one for the rate of change from state from 0 to 1 (the &quot;forward&quot; rate) and one for the rate of change from 1 to 0 (the &quot;backward&quot; rate). Thus, this is a simple model that allows a bias in gains versus loses. As of version 1.1 of Mesquite, this model supports only binary (0,1) characters. Mesquite supports two alternative ways to describe the model. The two parameters can be forward rate and backward rate, or overall rate and bias of gains versus losses. Thus, if forward and backward rates are both 0.5, then this can alternatively be described as a rate of 0.5 and a bias of 1.0 (i.e., unbiased). The conversion between the two representations is done by the following formulas: forward rate = overall rate * square root(bias); backward rate = overall rate / square root(bias). This conversion means that forward rate / backward rate = bias. Thus an AsymmMk model would have an instantaneous rate matrix of the following form:</li></ul><br />
<div style="text-align: center"><img src="files/markovTable2.png" alt="markovTable2.png" title="markovTable2.png" style="height: 82px; width: 308px;" /></div><br />
<br />
<ul class="quotelist"><li>As of version 1.1 of Mesquite, the AsymmMk model has two options for the handling of the root. (1) &quot;Root State Frequencies Equal&quot;: With this option, the root is permitted (or required, depending on your point of view) to have expected state frequencies different from those implied by the model. In estimating the likelihood of the model, and in calculating marginal likelihoods for states at internal nodes, probabilities can be summed over all possible ancestral state reconstructions. This effectively treats the expected state frequencies of the root as equal (0.5/0.5). This is the approach of Schluter et al. (1997), Pagel (1999) and Mesquite versions 1.0 through 1.06. When rates of gains and losses are different, this can yield perplexingly ambiguous states at the root of the tree (Schluter et al, 1997), which can be viewed positively as conservative, or negatively as a consequence of a contradiction between an implicit assumption of equal state frequencies at the root and biased equilibrium state frequencies implied by the model. (2) &quot;Root State Frequencies Same as Equilibrium&quot;: With this option, the expected frequencies at the root are assumed to be consistent with the model's rates. A difference between rates of gains and losses in the model implies biased equilibrium frequencies. These implicit equilibrium frequencies are used as priors for calculating the likelihood of the model and for calculating likelihoods of ancestral states. This approach is now the default in Mesquite. It can be viewed positively as applying the model of evolution consistently throughout the tree, or negatively as imposing the assumption of a prior (albeit one derived from the data) at the root. These two options can give remarkably different reconstructions when one state is rare and the forward and backward rates are estimated from the data. The options can be chosen in the model's <a  href="Ancestral%20States.html#editingModels">editor</a>.</li></ul><br />
Many programs bundle the rate of evolution into the branch lengths of the tree itself. Thus, to change the rate of evolution, the tree needs to be stretched or shrunk; there is no separate rate parameter that belongs to the stochastic model of evolution. This works well as long as the branch lengths are understood in the same way by the model and the tree, i.e., the tree's time units (calibration of time scale) are the same as that of the model. However, in Mesquite different calculations might make different assumptions about the time scale: coalescence calculations might need the tree's branches measured in generations, while a Jukes Cantor model might assume they are in expected nucleotide substitutions. Thus, many stochastic models in Mesquite have an extra parameter compared to other programs: the scaling of the model to the tree. For this reason Mk1 has a rate parameter to scale the rate against the tree.<br />
<br />
If parameters of a model are unspecified, Mesquite currently <a  href="Processes%20of%20Character%20Evolution.html#param">estimates</a> them based on the data. Note: Mesquite currently estimates parameters on each character separately, not on the entire data matrix. In addition Mesquite's likelihood calculations do NOT estimate branch lengths. They use pre-existing branch lengths (if a branch length is unassigned, it is treated as 1.0).<br />
<br />
Mesquite cannot do likelihood calculations in trees with soft polytomies, or if some taxa have polymorphisms or ambiguous in the character. Missing data and gaps (inapplicable) are permitted; the calculations are then done as if taxa so coded are absent from the tree. The calculations also require that the states of a character are contiguous from zero; i.e., the character cannot have only states 0,1 and 3.<br />
Other programs that reconstruct ancestral states using likelihood are Pagel's Discrete and Swofford's PAUP*.<br />
<br />
<a name="editingModels"></a><strong>Making, editing and applying probability models</strong>: To use the likelihood calculations, stochastic (probabilistic) models of evolution must be defined. Two models are predefined: a general Mk1 model and a general AsymmMk model. Both of these have their parameters unspecified.<br />
<br />
You can also create your own models and specify their parameters by selecting Characters&gt;New Character Model&gt;Markov k-state 1-parameter model (to make an Mk1 model) or Characters&gt;New Character Model&gt;Asymmetrical 2-param. Markov-k model (to make anAsymmMk model). In either case a window will appear in which you can specify the parameters. The Mk1 model allows you to change the rate. Also, you can change the maximum state allowed using a menu item in the Mk1_model menu (e.g., to restrict it to binary characters, choose 1 as the maximum state). The AsymmMk model allows you to change the forward and backward rates, and the assumption about root state frequencies. You can also choose to express the two parameters in the AsymmMk model as a rate (which controls both forward and backward rates) and a bias (which controls the ratio of forward to backward rates). A bias of greater than 1 means forward changes are more probable; a bias of less than 1 means that backward changes are more probable.<br />
<br />
After creating a model, you can edit it by selecting it under Characters&gt;Edit Character Model. You can rename or delete a model by going to the List of Character Models window available under Characters.<br />
<br />
Once models are defined they can be applied and used. When setting up a likelihood calculation, if you indicate to use &quot;Stored Probability Model&quot;, the calculation will use the selected model for all characters. Alternatively, if the characters used are stored in a matrix (instead of generated temporarily such as by simulations), then each character can be assigned a model in advance of the calculation. This can be done by going to the List of Characters window, selecting the row(s) corresponding to the desired character(s), and then touching on the column heading &quot;Probability Model&quot;. A drop-down menu contains a submenu that allows you to select the models to apply. These models will remain assigned to the characters if you save and reopen the file. You can also change the current probability model applied to a character by selecting a module in the Probability Model submenu of the Trace menu. Once models are assigned to the characters, then the these are treated as the &quot;Current&quot; models applied to the characters. To indicate that the likelihood calculations use these assigned models, indicate &quot;Current Probability Model&quot; when asked for the source of models.<br />
<br />
<a name="optimize"></a><strong>Optimization Settings</strong>: Models used for likelihood calculations may have adjustable settings for the optimization routines used to estimate parameter values. These settings can be changed under Characters&gt;Model Settings; once changed the settings are universal, applying to all calculations with that category of model. They are stored in the preferences directory and used again the next time you start Mesquite. The Mk1 Model has one setting: The coarseness of the intervals surveyed in optimizing the rate parameter. Wider intervals may result in finding the optimum more quickly when rates are high, but may make less accurate when rates are low. Mesquite by default tries the optimization twice, first with width 1.0 then again with width 10.0, and then chooses the best results. You can request that Mesquite try a single fixed width; suggested values are 1.0 to 20.0. To request Mesquite that use its default strategy, enter a width of 0. The AsymmMk model has one setting, concerning the values of parameters used as starting points in the search for optimal parameter values. This setting is used when both parameters of the model are unspecified and need to be estimated. There are three options: (1) The rate is first estimated using the Mk1 model, and then the optimization routine is given that rate plus a bias of 1.0 as starting values. (2) The forward and backward rates of 1.0 and 1.0 are used as starting values, and then 1.0 and 0.1, and then 0.1 and 1.0. Of these three attempts, the parameter values from the attempt yielding the highest likelihood is chosen. (3) Options 1 and 2 are combined, resulting in four attempts to estimate parameters, first using the Mk1 rate then the three alternative forward and backward rates as starting values. Of the four attempts, the parameter values from the attempt yielding the highest likelihood is chosen. The default option is (1). Because the AsymmMk sometimes uses the Mk1 model, changing the setting of the Mk1 model may affect results from the AsymmMk model.<br />
<br />
<a name="report"></a><strong>Reporting of results</strong>: Likelihood ancestral state reconstructions can be reported in various ways. A first issue is whether only the best estimates are shown at a node or instead the support for each state is shown at a node, regardless of how strong or weak. This is controlled in Trace Character History by the Display Proportional to Weights menu item. If it is selected, then support for each state is shown; otherwise, only the states judged best are shown. The judgment of what are the best states is made according to a decision threshold T, such that if the log likelihoods of two states differ by T or more, the one with lower likelihood (higher negative log likelihood) is rejected. This is set using the Likelihood Decision Threshold menu item.<br />
<br />
What states are judged best can be viewed using Trace Character History in several ways: (1) When the cursor is held over a branch and the list of states appears at the bottom of the Trace Legend, the states judged best according to the threshold are marked with an asterisk; (2) In the Text view of the window, the list of reconstructions shows an asterisk by each state judged best at the mode; and (3) When Display Proportional to Weights is turned off, only the best states are shaded on the branches. The threshold and best states are also used in Trace Character Over Trees.<br />
<br />
Another issue is whether the likelihoods of alternative states are reported as is (Raw Likelihoods) or not. The other two options are to report proportional likelihoods (likelihoods of states are scaled so that they add up to 1, and thus for each state is shown its proportion of total likelihood) or negative log likelihoods. The former are convenient for interpretation and visualization; the latter may be more easily used in statistical tests.<br />
<br />
<h3 id="toc7"><a name="Studying the History of Character Evolution--Stochastic character mapping"></a><a name="scm"></a>Stochastic character mapping</h3>
 Stochastic character mapping (Neilsen, 2002; Huelsenbeck, et al. 2003) simulates realizations of precise histories of character evolution in a way consistent with likelihoods of the model and ancestral states. By &quot;precise&quot; we mean that the realizations depict not just the states at nodes of the tree, but the states at all points along a branch between the nodes. Thus, with stochastic character mapping you often see changes within branches that are in principle unobservable but at the same time predicted, indeed demanded, by the rates of character change. For instance, there is stochastic character mapping used with Trace Character History:<br />
<div style="text-align: center"><img src="files/scm.jpg" alt="scm.jpg" title="scm.jpg" /></div><br />
Stochastic character mapping is chosen as a Reconstruction Method when you start Trace Character History, or from the Trace menu. It can be done currently only with non-molecular categorical data. Stochastic character mapping in Mesquite is primarily a visualization tool at present. While there are many interesting calculations that can be derived from these realizations, as done by the program SIMMAP (Bollback, 2006), none of them is yet done by Mesquite.<br />
<br />
We recommend that the Square Tree drawing style be used to visualize stochastic character mapping. The default Square Tree corners type, &quot;Right Angle&quot;, is best (available under Display&gt;Corners). With Rounded or Diagonal corners, the basal part of each branch is shown in gray and the reconstruction shown only in the distal portion, because of the complexity of trying to draw intermediate changes on a slanted or rounded corner.<br />
<br />
To obtain an account of the detailed location of changes, go to the text view of the window. You can also use the scripting command getAsTable to the tree window itself to obtain a coded output that can be parsed programmatically.<br />
<h3 id="toc8"><a name="Studying the History of Character Evolution--Comparison and Interaction with other Programs"></a><a name="compare"></a>Comparison and Interaction with other Programs</h3>
 <a name="compareNexus"></a><strong>NEXUS-based programs</strong>: Mesquite currently saves the models of evolution for likelihood in the private MESQUITECHARMODELS block in NEXUS files. A private block is used because there is as yet no standard for designating such models. Thus, PAUP*, MrBayes and other programs doing likelihood calculations will not be able to access these character models.<br />
<br />
<a name="compareMacClade"></a><strong>Comparison with MacClade</strong>: Users familiar with MacClade (<a  href="http://macclade.org/" rel="nofollow">macclade.org</a>) will notice some of its features missing from Mesquite, and vice versa. MacClade is restricted to parsimony reconstructions, but has the following features that Mesquite currently lacks. MacClade's Trace Character facility has the ability to fix states at a node (the paintbrush tool) and to show individual MPR's (MPRs mode, formerly Equivocal Cycling). MacClade's Trace All Changes mode and Changes &amp; Stasis chart summarize reconstructed changes in all characters. Parsimony models include Dollo, irreversible and character state trees. Mesquite, on the other hand, includes likelihood reconstructions, reconstructions for continuous characters better integrated with Trace Character, branch-length sensitive calculations and other features such as Trace Over Trees.<br />
<br />
<a name="comparePagel"></a><strong>Pagel's Discrete and Multistate programs</strong>: Pagel's Discrete and Multistate programs also do likelihood reconstructions of ancestral states. Discrete's Fossil Likelihoods with the Global option corresponds to Mesquite's Likelihood Ancestral States. To obtain reconstructions in Multistate equivalent to Mesquite's, define the restricted model equivalent to Mk1 or AsymmMk, using a series of &quot;restrict&quot; commands to set the q parameters equal as appropriate. Use the &quot;test&quot; command to estimate the q rates. Then, restrict the rates to those estimated. Next, use the &quot;fossil&quot; command to fix the node of interest to each of the states in turn, after each using the &quot;test&quot; command to ask Multistate to evaluate the likelihood. These likelihoods are the global likelihoods for the states at the node.<br />
<br />
Discrete and Multistate have several features not available currently in Mesquite, including the Local option for parameter estimation, more complete reporting of statistics for the reconstructions, and calculations to test correlation among characters using likelihood ratio tests.<br />
Mesquite can import and export data files for Discrete and Multistate (ppy files). To import, select the file with Mesquite and choose Pagel format in the import dialog box. To export, select File&gt;Export....<br />
<h3 id="toc9"><a name="Studying the History of Character Evolution--Acknowledgments"></a><a name="acknow"></a>Acknowledgments</h3>
 David Swofford assisted by providing code in C, translated by us to Java, for the optimization routines used in the likelihood reconstruction. Peter Midford wrote much of the module that modified the likelihood calculations to perform stochastic character mapping. Paul Lewis, David Swofford and Mark Holder helped us with useful discussions on likelihood reconstructions.<br />
<h3 id="toc10"><a name="Studying the History of Character Evolution--References"></a><a name="refs"></a>References</h3>
 Bollback, J. P. 2006 SIMMAP: Stochastic character mapping of discrete traits on phylogenies. BMC Bioinformatics. 7:88.<br />
Huelsenbeck, J.P., R. Nielsen, and J.P. Bollback. 2003. Stochastic mapping of morphological characters. Systematic Biology 52:131-158.<br />
Lewis, P.O. 2001. A likelihood approach to estimating phylogeny from discrete morphological character data. Systematic Biology 50:913-925.<br />
Lutzoni F, M. Pagel &amp; V. Reeb. 2002. Major fungal lineages are derived from lichen symbiotic ancestors. Nature 411: 937-940.<br />
Maddison, D.R. and W.P. Maddison. 2000. MacClade version 4: Analysis of phylogeny and character evolution. Sinauer Associates, Sunderland Massachusetts.<br />
Nielsen, R. 2002. Mapping mutations on phylogenies. Systematic Biology. 51:729-739.<br />
Pagel, M. 1999. The maximum likelihood approach to reconstructing ancestral character states of discrete characters on phylogenies. Systematic Biology. 48: 612-622.<br />
Pagel, M. 2000. Discrete, version 4.0. A computer program distributed by the author.<br />
Pagel, M. 2002. Multistate, version 0.6. A computer program distributed by the author.<br />
Schluter D, T. Price, A.O. Mooers, D. Ludwig. 1997. Likelihood of ancestor states in adaptive radiation. Evolution. 51: 1699-1711.<br />
Swofford, D.L. 2002. PAUP*. Phylogenetic Analysis Using Parsimony (*and Other Methods), Version 4.0. Sinauer Associates, Sunderland, Massachusetts.
<!-- InstanceEndEditable -->

    
    
    </div>
	<!-- This clearing element should immediately follow the #mainContent div in order to force the #container div to contain all child floats --><br class="clearfloat" />
  <div id="footer">
    <p> ©2018 David R. Maddison &amp; Wayne P. Maddison</p>
  <!-- end #footer --></div>
<!-- end #container --></div>

<script type="text/javascript">
<!--


var OverviewOfMesquitePanel = null;
var GettingStartedPanel = null;
var BasicMesquiteOperationPanel = null;
var TaxaTreesCharactersPanel = null;
var AnalysesPanel = null;
var WorkflowsPanel = null;
var GettingHelpPanel = null;
var TechnicalDetailsPanel = null;
var HistoryPanel = null;
var PublishingResultsPanel = null;

if (qsParm['BasicMesquiteOperationPanel']=='open')
 	BasicMesquiteOperationPanel = new Spry.Widget.CollapsiblePanel("BasicMesquiteOperationPanel", {contentIsOpen: true});
else
 	BasicMesquiteOperationPanel = new Spry.Widget.CollapsiblePanel("BasicMesquiteOperationPanel", {contentIsOpen: false});
	
if (qsParm['OverviewOfMesquitePanel']=='open')
 	OverviewOfMesquitePanel = new Spry.Widget.CollapsiblePanel("OverviewOfMesquitePanel", {contentIsOpen: true});
else
 	OverviewOfMesquitePanel = new Spry.Widget.CollapsiblePanel("OverviewOfMesquitePanel", {contentIsOpen: false});
	
if (qsParm['GettingStartedPanel']=='open')
 	GettingStartedPanel = new Spry.Widget.CollapsiblePanel("GettingStartedPanel", {contentIsOpen: true});
else
 	GettingStartedPanel = new Spry.Widget.CollapsiblePanel("GettingStartedPanel", {contentIsOpen: false});

if (qsParm['TaxaTreesCharactersPanel']=='open')
 	TaxaTreesCharactersPanel = new Spry.Widget.CollapsiblePanel("TaxaTreesCharactersPanel", {contentIsOpen: true});
else
 	TaxaTreesCharactersPanel = new Spry.Widget.CollapsiblePanel("TaxaTreesCharactersPanel", {contentIsOpen: false});

if (qsParm['AnalysesPanel']=='open')
 	AnalysesPanel = new Spry.Widget.CollapsiblePanel("AnalysesPanel", {contentIsOpen: true});
else
 	AnalysesPanel = new Spry.Widget.CollapsiblePanel("AnalysesPanel", {contentIsOpen: false});

if (qsParm['WorkflowsPanel']=='open')
 	WorkflowsPanel = new Spry.Widget.CollapsiblePanel("WorkflowsPanel", {contentIsOpen: true});
else
 	WorkflowsPanel = new Spry.Widget.CollapsiblePanel("WorkflowsPanel", {contentIsOpen: false});

if (qsParm['GettingHelpPanel']=='open')
 	GettingHelpPanel = new Spry.Widget.CollapsiblePanel("GettingHelpPanel", {contentIsOpen: true});
else
 	GettingHelpPanel = new Spry.Widget.CollapsiblePanel("GettingHelpPanel", {contentIsOpen: false});
    
if (qsParm['TechnicalDetailsPanel']=='open')
 	TechnicalDetailsPanel = new Spry.Widget.CollapsiblePanel("TechnicalDetailsPanel", {contentIsOpen: true});
else
 	TechnicalDetailsPanel = new Spry.Widget.CollapsiblePanel("TechnicalDetailsPanel", {contentIsOpen: false});

if (qsParm['HistoryPanel']=='open')
 	HistoryPanel = new Spry.Widget.CollapsiblePanel("HistoryPanel", {contentIsOpen: true});
else
 	HistoryPanel = new Spry.Widget.CollapsiblePanel("HistoryPanel", {contentIsOpen: false});
if (qsParm['PublishingResultsPanel']=='open')
 	PublishingResultsPanel = new Spry.Widget.CollapsiblePanel("PublishingResultsPanel", {contentIsOpen: true});
else
 	PublishingResultsPanel = new Spry.Widget.CollapsiblePanel("PublishingResultsPanel", {contentIsOpen: false});


//-->
</script>
</body>
<!-- InstanceEnd --></html>
