<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><!-- InstanceBegin template="/Templates/MesquiteTemplate.dwt" codeOutsideHTMLIsLocked="false" -->
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />


<!-- InstanceBeginEditable name="doctitle" -->
<title>History of New Features</title>
<!-- InstanceEndEditable -->

<![endif]--><!--[if IE]>
<style type="text/css"> 
/* place css fixes for all versions of IE in this conditional comment */
.thrColFixHdr #sidebar2, .thrColFixHdr #sidebar1 { padding-top: 30px; }
.thrColFixHdr #mainContent { zoom: 1; }
/* the above proprietary zoom property gives IE the hasLayout it needs to avoid several bugs */
</style>
<![endif]-->

<style type="text/css">
<!--
-->
</style>
<link href="CSS/mesquitePackage.css" rel="stylesheet" type="text/css" />
<link href="SpryAssets/SpryCollapsiblePanel.css" rel="stylesheet" type="text/css" />

<script src="SpryAssets/SpryCollapsiblePanel.js" type="text/javascript"></script>
<script type="text/javascript" src="SpryAssets/SpryURLUtils.js"></script>
<script type="text/javascript" src="javascript/mesquite.js"></script>

<!-- InstanceBeginEditable name="head" -->
<!-- InstanceEndEditable -->
</head>

<body bgcolor="#ffffff" class="thrColFixHdr" >

<div id="container">
  <div id="header">
    <table width="100%" border="0" cellpadding="4">
  <tr>
    <td width="51%"><strong id="PackageTitle">Mesquite</strong></td>
    <td width="33%" align="right"><div id="subtitle"><!-- InstanceBeginEditable name="subtitle" -->History of New Features<!-- InstanceEndEditable --></div></td>
    <td width="2%">&nbsp;</td>
    <td width="14%">Version 3.6<br />
      27 December 2018<br /></td>
  </tr>
</table>

</div>
  
  
<div id="sidebar1">
  
  
  
  
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('http://www.mesquiteproject.org')">Mesquite Website Home</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('home.html')">Introduction</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('Installation.html')">Download & Installation</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('History%20of%20New%20Features.html')">Recent Changes</a>
  </div>

  <div id="OverviewOfMesquitePanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Overview of Mesquite ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('What%20Mesquite%20does.html');">What Mesquite Does</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Mesquites%20Features.html');">List of Features</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Development%20Team.html');">Development Team</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Contact%20Us.html');">Contact Us</a></td></tr>
		</table>
    </div>
  </div>
  <div id="GettingStartedPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Getting Started ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Installation.html');">Download and Installation</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Additional%20Mesquite%20Packages.html');">Additional Packages</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Beginning%20Mesquite.html');">Beginning Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Example%20Files.html');">Example Files</a></td></tr>
		</table>
    </div>
  </div>
  <div id="BasicMesquiteOperationPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Basic  Operations ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Files.html');">Files</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Menus.html');">Menus</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Windows.html');">Windows</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Charts.html');">Charts</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Selection.html');">Selection</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Modules.html');">Modules</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simplifying%20the%20Interface.html');">Simplifying Mesquite</a></td></tr>
		</table>
    </div>
  </div>
  <div id="TaxaTreesCharactersPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Taxa, Trees, Characters ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Characters%20&%20Matrices.html');">Characters &amp; Matrices</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Taxa.html');">Taxa</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Trees.html');">Trees</a></td></tr>
		</table>
    </div>
  </div>
  <div id="AnalysesPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Analyses ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Character%20Evolution.html');">Character Evolution</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Align.html');">Sequence Alignment</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Analyzing%20Molecular%20Data.html');">Molecular Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Diversification.html');">Diversification</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Gene%20Trees.html');">Gene Trees</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simulations%20&%20Randomizations.html');">Simulations &amp; Randomizations</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Analyzing%20Continuous%20Characters.html');">Continuous Characters</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Studies%20of%20Tree%20Topology.html');">Studies of Tree Topology</a></td></tr>
		</table>
    </div>
  </div>
  <div id="WorkflowsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Workflows ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Managing%20Molecular%20Data.html');">Managing Molecular Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Managing%20Continuous%20Data.html');">Managing Continuous Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Interactions%20with%20Other%20Programs.html');">Interactions with other programs</a></td></tr>
		</table>
    </div>
  </div>
  <div id="GettingHelpPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Getting Help ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Learning%20how%20to%20use%20Mesquite.html');">Learning Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simplifying%20the%20Interface.html');">Simplifying Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Bugs%20and%20Troubleshooting.html');">Bugs &amp; Troubleshooting</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Getting%20Help.html');">Getting Help</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Suggesting%20New%20Features.html');">Your Suggestions</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Glossary.html');">Glossary</a></td></tr>
  			<tr><td><a href="javascript:pageLink('FAQ.html');">FAQ</a></td></tr>
		</table>
    </div>
  </div>
  <div id="TechnicalDetailsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Technical Details ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('How%20Mesquite%20Works.html');">How Mesquite Works</a></td></tr>
  			<tr><td><a href="javascript:pageLink('MesquiteLogSupport.html');">Log & Support Files</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Scripts%20&%20Macros.html');">Scripts &amp; Macros</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Source%20Code.html');">Source Code</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Developers%20Documentation.html');">Developer's Documentation</a></td></tr>
		</table>
    </div>
  </div>
  <div id="HistoryPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">History ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('History%20of%20Mesquite.html');">History of Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('whyMesquite.html');">Why Mesquite?</a></td></tr>
  			<tr><td><a href="javascript:pageLink('History%20of%20New%20Features.html');">New Features & Bug List</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Release%20Dates.html');">Release Dates</a></td></tr>
		</table>
    </div>
  </div>
  <div id="PublishingResultsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Publishing Results ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Documenting%20Your%20Work.html');">Documenting your work</a></td></tr>
  			<tr><td><a href="javascript:pageLink('How%20to%20Cite%20Mesquite.html');">Citing Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Publishing%20Results.html');">Publishing results</a></td></tr>
		</table>
    </div>
  </div>

  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('Acknowledgements.html')">Acknowledgments</a>
  </div>

<h3>&nbsp;</h3>
  <!-- end #sidebar1 --></div>

  <div id="mainContent">
    
<!-- InstanceBeginEditable name="content" -->
    
    
<h1 id="toc0"><a name="New Features, Bugs, and History of Features and Bug Fixes"></a>New Features, Bugs, and History of Features and Bug Fixes</h1>
 <p>Here are some of the more notable new features and bug fixes since version 1.0. If the previous way Mesquite worked could have yielded errors in substantive results (e.g., there was a bug in a phylogenetic calculation), those bugs are marked with <img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />.<br />
  <br />
   A list of new modules in the latest version can be seen by choosing &quot;List new modules&quot; from the Help menu<br />
  <br />
   There is also a list of <a  href="Release%20Dates.html">release dates.</a><br />
  <br />
  <a name="New Features, Bugs, and History of Features and Bug Fixes--Features in the next release version of Mesquite; the latest source code in the development branch of the MesquiteProject/MesquiteCore repository on GitHub contains these changes" id="New Features, Bugs, and History of Features and Bug Fixes--Features in the next release version of Mesquite; the latest source code in the development branch of the MesquiteProject/MesquiteCore repository on GitHub contains these changes"></a>Features to come in the next release version of Mesquite are listed <a href="https://github.com/MesquiteProject/MesquiteCore/wiki/New-features-and-bug-fixes-in-the-development-version">on this web page</a>. These new features and bug fixes are already contained in latest source code in the development branch of the MesquiteProject/MesquiteCore repository on GitHub.</p>
 <p>&nbsp;</p>
 <hr />
 <h3>Version 3.6</h3>
 <h4 id="toc82">New Features</h4>
 <ul>
   <li>Names of selected taxa in tree windows can be enlarged (for instance, so that ingroup can be shown in larger font than outgroups). See Display&gt;Highlight for Selected Taxa.</li>
   <li>There are new options for color and style of names of taxa in tree windows. In addition to the old options of coloring by assigned color or taxon group, the name of a taxon can be colored according to how many matrices it has data in, or to some value for the taxon. See Display&gt;Color and Style of Taxon Names.</li>
   <li>Compatibility with Chromaseq 1.5 (including the <strong>new tools for submission to GenBank</strong>).</li>
   <li>There is now a search-and-replace function for the OTU ID code in the Taxa List Window.</li>
   <li>Copy Newick Tree for Open Tree available under Tree&gt;Utilities.</li>
   <li>There is a new character selection tool built to search for diagnostic characters or synapomorphies of groups, e.g. for molecular data. In a character matrix editor, first select the taxa whose distinguishing features you are seeking, then choose Select&gt;Characters&gt;Select Characters&gt;With State Distinguish Selected Taxa.</li>
   <li>One can now ask to have missing data cells show in paler text using Display&gt;Lighten&gt;Lighten Missing</li>
 </ul>
 <br />
 <strong> Bug Fixes and Workarounds</strong><br />
 </p>
 <ul>
   <li>Dropping FASTA and other files on a matrix always led to adjusting the imported sequences relative to a reference, even if that option was not chosen. Fixed in build 906.</li>
   <li>FASTA file import failed if the first line of the file was blank. Fixed in build 910.</li>
   <li>The dialog box for choosing colors was often not responsive. Fixed in build 906.</li>
   <li>On Windows computers with high-resolution monitors, Mesquite's text is now normally sized, not overly small. This was a bug in Java on Windows, but an available workaround was added to build 917.<br />
   </li>
 </ul>
 <hr /><h3>Version 3.51</h3>
<br /><strong>
Bug Fixes and Workarounds</strong><br />
 </p>
 <ul>
   <li>Mesquite wouldn&rsquo;t work if it was contained within a directory whose name included some accented letters and other high ASCII characters.</li>
   <li>Fix bug whereby reading in a treefile with bootstrap frequencies attached as node labels were attached to nodes rather than branches even though the user chose to interpret them as bootstrap frequencies in the &quot;How to interpret numbers in tree description&quot; dialog box. Thanks to Austin Baker for reporting this. Fixed in build 892.<br />
     <br />
   </li>
 </ul>

	  
 <hr />
<h3 id="toc1"><a name="New Features, Bugs, and History of Features and Bug Fixes--Features in the next release version of Mesquite; the latest source code in the development branch of the MesquiteProject/MesquiteCore repository on GitHub contains these changes"></a>Version 3.5<br />
</h3><h4 id="toc2">New Features</h4>
 <ul>
   <li>Improvements were done to support a new version of <a href="http://zephyr.mesquiteproject.org">Zephyr</a> that supports IQ-TREE.</li>
   <li>The manual was moved away from online-only at Wikispaces to being available both online at <a href="http://www.mesquiteproject.org">Github</a> and offline within the distributed Mesquite_Folder (at Mesquite_Folder/docs/home.html).</li>
 </ul>
 <strong><br />
Bug Fixes and Workarounds<br />
 </strong>
 <ul><li>Scrolling in various windows using the scroll wheel is sometimes very chaotic. This is a bug not in Mesquite but one introduced into Java 1.8 update 161. A workaround was introduced into Mesquite Build 879.</li><li>Scrolling with the scroll wheel sometimes caused Mesquite to freeze. Fixed in build 881.</li><li>Can now read CharSet token embedded within a CharPartition command in the NEXUS file. Build 884.</li></ul><br />
<hr />
<h3 id="toc3"><a name="New Features, Bugs, and History of Features and Bug Fixes--Verstion 3.40"></a>Version 3.4</h3>
 <br />
<h4 id="toc4"><a name="New Features, Bugs, and History of Features and Bug Fixes--Verstion 3.40-New Features"></a>New Features</h4>
 <ul><li>A new tool has been added to the data matrix window for non-molecular categorical data, the Popup State Entry tool. When touched on a cell, a popup menu will appear that lists the states for that character, allowing easy data entry if the states are named.</li><li>The number of items in the project panel is limited to save memory and time. The default of 50 can now be changed under File&gt;Defaults&gt;Maximum # Items in Project Panel....</li></ul><br />
<h4 id="toc5"><a name="New Features, Bugs, and History of Features and Bug Fixes--Verstion 3.40-Bug Fixes and Workarounds"></a>Bug Fixes and Workarounds</h4>
 <ul><li>Mesquite (as of build 875) can now run under Java 9. This required building a new startup protocol to accommodate Java 9 changes and restrictions.</li><li>The classpaths.xml file has a new format; it is now a simple text file (classpaths.txt) with the paths listed one per line.</li><li>Alignment using ClustalW was broken. Fixed in build 860.</li><li>Matrices formed using the Concatenate Selected Matrices feature sometimes omitted state names that were present in the original matrices. Fixed in build 861.</li></ul><br />
<hr />
<h3 id="toc6"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.31"></a>Version 3.31</h3>
 New Features<br />
<ul><li>There are a few minor enhancements to the Project panel and notifications to the user.</li></ul><h4 id="toc7"> </h4>
 <h4 id="toc8"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.31-Bug Fixes and Workarounds"></a>Bug Fixes and Workarounds</h4>
 <ul><li>Alignment using external programs (e.g., MAFFT, MUSCLE) was not functioning for old (1.7 and before) versions of Java. Fixed.</li><li>Several small bugs fixed.</li></ul><br />
<hr />
<h3 id="toc9"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.3"></a>Version 3.3</h3>
 <h4 id="toc10"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.3-New Features"></a>New Features</h4>
 <ul><li>Many changes were made to Mesquite's core in preparation for <a  href="http://zephyr.mesquiteproject.org">Zephyr 2.0</a>, which allows communication with external inference programs.</li><li>New options were added to better <strong>import trees with support values</strong>, from other programs. Some programs store support values (e.g., posterior probabilities) in Newick tree descriptions as labels for nodes (e.g. &quot;(((A,B)0.95,C),D)&quot;). If Mesquite reads a tree description and finds that there are numerical labels, it will now ask you how to interpret them: as bootstrap values, posterior probabilties, etc. You can also specify whether the information is to be attached to the node versus the branch (this has an effect on how it behaves when the tree is rerooted; <a  href="History%20of%20New%20Features.html#RerootedInformation">see this</a>). This function does not work for information attached to nodes using special &amp; comments, e.g. by TreeAnnotator. See <a  href="History%20of%20New%20Features.html#ImportInformationAtNodes">this note</a> for details about how to recover that information.</li><li>Process Data files, which reads a series of separate files and modifies or exports them, has a new option for compiling all the matrices read into a single NEXUS file (<strong>Compile Matrices into One File</strong>). This does not concatenate them intro a single matrix, but will allow you to read, for example, many Fasta files each representing a separate locus, and compiling them into a single NEXUS file. To access this option, select File&gt;Open Other&gt;Process Data Files..., indicate you want to process the files, and choose Compile Matrices Into One File as a file processor.</li><li>You can now enter a <strong>target value for the magic wand</strong> tool in the list windows (e.g., List of Characters window). This way, you can request selecting all items with a value of greater than, say 0.1, without finding and touching on an item with this exact value. To use this, touch on the tool in the palette to bring the drop down menu, select Set Target Value... to enter the string to be compared. Also, turn on the target search by making sure Use Entered Target Value is selected. Use the equals, greater than, or less than items to set what you would like to select. Then, touch on the relevant column in which the search value is to be found.</li><li>Mesquite can now calculate the <strong>number of SPR rearrangements</strong> for a given tree. This is available in all contexts for which a number for a tree would be needed. Build 805.</li><li>The ThinkTrees package was added to Mesquite. ThinkTrees provides a few tools for teaching phylogenetics, including the ability to randomly rotate nodes.</li></ul><br />
<h4 id="toc11"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.3-Bug Fixes and Workarounds"></a>Bug Fixes and Workarounds</h4>
 <ul><li>On Windows computer there were problems with communications between Mesquite and external alignment programs. Fixed.</li><li>On Windows computer there were problems with communications between Mesquite and programs required by <a   href="http://chromaseq.mesquiteproject.org" rel="nofollow">Chromaseq</a> for newer versions of Cygwin. Fixed.</li><li>On NCBI changed how BLAST searches are conducted programmatically, and Mesquite's Matrix &gt; Search &gt; Top BLAST Matches on the NCBI server failed as a result. Fixed in Build 803.</li><li>A bug was introduced in Mesquite 3.2 that changes how taxon names were simplified; in particular, in various contexts in which underscores should have been used in taxon names, they were removed. This may cause various file incompatibilities, and would alter the behavior of Simplify Taxon Names. Fixed in Build 807.</li><li>Drawing of coalescent trees contained within another tree was broken in version 3.1. Fixed in Build 809.</li><li>Separated suppression of TITLE and LINKs commands from suppression of ID and BLOCKID commands for better compatibility with PAUP* (see File&gt;Defaults). Build 811.</li><li>Fused Matrix Export (NEXUS) wrote incorrect FORMAT statements. Fixed in Build 828.</li></ul><br />
<hr />
<h3 id="toc12"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.2"></a>Version 3.2</h3>
 <h4 id="toc13"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.2-New Features"></a>New Features</h4>
 <ul><li>The <strong>Export Matrices and Batch</strong> feature (used for parametric bootstrapping) now saves a file giving a summary of the parameters of the modules that created the matrices and batch files, allowing you to record, for example, the parameters used in simulations. In addition, a summary file is saved with statistics about each matrix (if matrices were created).</li><li>There is a new option available to alter matrices: <strong>Impose Indel Pattern of Other Matrix</strong>. This will take convert any cell in the matrix to be altered into a gap (inapplicable) if the equivalent cell in another matrix has a gap. If the matrix to be altered already has a gap in a cell, it will not be altered.</li><li><strong>Local BLAST</strong> searches can now search over multiple databases at once. Build 788.</li><li><strong>Clade Frequencies in Trees</strong> can now calculate, for each node in the tree, the maximum frequency across a set of trees of a clade that <em><strong>contradicts</strong></em> the node, as well as the difference between the frequency of the clade and the maximum frequency of a contradictory clade. (Available in the Analysis:Trees&gt;Values for Nodes menu.) Build 789.</li><li>External alignment programs such as MAFFT that are started by Mesquite can now be stopped using a stop button on the progress indicator. Build 793.</li><li>By default Mesquite stores its log files and user preferences in a folder <strong>Mesquite_Support_Files</strong> in the user's home directory. We have added an alternative location, <strong>.Mesquite_Support_Files</strong>, to permit Linux and MacOS X users to have the folder be invisible. Simply change the name of the folder by adding the &quot;.&quot; at the beginning, and Mesquite will find it.</li><li>You can now export matrices as descriptions (i.e., including listings of states in taxa) similar to the feature available in MacClade.</li></ul><br />
<h4 id="toc14"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.2-Bug Fixes and Workarounds"></a>Bug Fixes and Workarounds</h4>
 <ul><li>There is a bug in Oracle's implementation of Java on the MacOS X that cause increasingly slowness the more you use Mesquite. Although we could not completely remove this (as only Oracle can fix their bug), we developed workarounds to avoid some of the slowdown.</li><li>MacOS 10.12 Sierra changed the way applications were handled, requiring a workaround in Mesquite installation.</li><li>Tree drawing, especially for circular trees, is now much improved, as the lines and curves are drawn more accurately.</li><li>MAFFT Alignment, and possibly alignment under other external aligners, was non-functional on at least Windows 7. Fixed in Build 792.</li><li>Diagonal character names in the matrix are now displayed correctly if character info strips are used. Build 782.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Using copy and paste by control-clicking in a Taxon Has Data column in the Taxon List window would fail to appropriately paste states for morphological matrices that were displaying full state names and those state names were complex with punctation. Fixed in build 790.</li><li>Many minor bugs have been fixed</li></ul><br />
<hr />
<h3 id="toc15"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.11"></a>Version 3.11</h3>
 <br />
<h4 id="toc16"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.11-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) A bug was introduced into 3.10 (it was not present in 3.04) that caused the codon positions of a DNA sequence matrix to be miswritten in specific circumstances, the most likely of which would be that a single non-coding site (not a series of sites in a contiguous stretch) appeared near the start of the matrix, and there was a later stretch of non-coding sites. When re-written, the labelled codon positions would have been scrambled, with (for instance) subsequent sites with codon position 1 being labelled as non-coding. Fixed in 3.11 (build 766).</li></ul><br />
<hr />
<h3 id="toc17"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.10"></a>Version 3.10</h3>
 <br />
<h4 id="toc18"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.10-New Features"></a>New Features</h4>
 <ul><li>Mesquite can now read the tree files produced by <strong>MrBayes's sumt</strong> command, and thus you can now use Mesquite to visualize the values on branches, etc. To do this, you will need to open the file using the File&gt;Open Other&gt;<strong>Special NEXUS</strong> ... command, and ask when queried to using the &quot;Import MrBayes Consensus Tree File&quot; feature. This will tell Mesquite that the trees being read in are annotated using MrBayes extensions to the tree description format. To see the values for each node, go to a treewindow displaying the tree, and choose either Display&gt;Node-Associated Values&gt;Choose Values to Show (which will give you the choice to show &quot;prob&quot; (the Bayesian Posterior Probability), &quot;probstddev&quot;, &quot;lengthmean&quot;, and &quot;lengthmedian&quot;), or Display&gt;Node-Associated Text&gt;Choose Associated Text to Show (with will give you the choice to show &quot;probrange&quot;, &quot;prob+-sd&quot;, or &quot;length95%HPD&quot;). You may wish to turn off drawing of the values on the terminal branches, by unchecking &quot;Show on Terminal Branches&quot; in the Node-Associated Values or Node-Associated Text menu items. You can also get a quick look at associated values using the &quot;?&quot; tool (Branch Info) and touching on a branch. [Thanks to Kip Will for suggesting this feature.]</li><li>Menus for the data editor have been redesigned, in particular to separate functions for altering the matrix into their own, organized menu, called <strong>Alter</strong>.</li><li>There is a new facility for reading and processing many data files within a folder (e.g. for phylogenomics pipelines), available at File&gt;Open Other&gt;<strong>Process Data Files...</strong>. At present it is crude, but with it you can (for instance) convert many files from one format to another, or align sequence matrices in many files, or do a calculation for files in many matrices. You will be asked to choose the folder, what is the format of the data files, whether only data files with a certain extension (e.g. &quot;.nex&quot;, &quot;.fas&quot;, &quot;.txt&quot;, etc.) are to be processed, and whether to automatically save all processed files as NEXUS files. You can then indicate you want to process each file in various ways.</li><li>Alterations of data matrices can now be performed en masse in the List of Character Matrices by selecting the matrices to be changed, and then selecting List&gt;Utilities&gt;<strong>Alter/Transform Selected Matrices</strong>.</li><li>There are now more flexible options for <strong>converting missing to gaps or gaps to missing</strong> (using Alter&gt;Convert Gap/Missing/Polymorph/Uncertain&gt; <strong>Gaps&lt;-&gt;Missing</strong>), including restricting the conversion to internal regions of a sequence or only terminal regions, available in the Alter menu of character matrix editors.</li><li>In merging taxa (using Matrix&gt;Taxon Utilities&gt;<strong>Merge Taxa</strong> or List&gt;Taxon Utilities&gt;Merge Taxa), you now have multiple new options. You have the option to keep the original, unmerged taxa. There are options as to how the merged taxon is named. You can specify whether or not merging that creates multistate characters sets them as uncertainties (especially valuable for DNA data) or polymorphisms.</li><li>You can now <strong>BLAST a local database</strong>, perhaps of your own sequences, using a new options in the <a  href="Analyzing%20Molecular%20Data.html#TopBlast">Top BLAST Matches</a> option available in Matrix&gt;Search&gt;Top BLAST Matches.</li><li>The information panel for the tree window, available by touching on the blue 'i&quot; at the bottom of the window, now shows a section for the source of trees. This allows you to see, for example, details about the RAxML run that found those trees, if the run was coordinated through <a  href="http://zephyr.mesquiteproject.org">Zephyr</a> .</li><li>When concatenating tree blocks (List&gt;Utilities&gt;<strong>Concatenate Tree Blocks</strong> in the List of Tree Blocks window), you now have the option to prefix each tree name in the concatenated tree block with the block name from which the tree came.</li><li>When grafting subtrees onto trees (Tree&gt;Alter/Transform Tree&gt;<strong>Graft Other Tree</strong>...) you can now select an internal branch, or one or more terminals. You also now have the option of deleting the selected taxa or branch from the tree, which you would want to do if you want the subtree to replace the selected taxa or branch.</li><li>You can now select those taxa that match the taxa listed in a file. In the List of Taxa window, choose List&gt;Select Taxa&gt;<strong>Select Taxa from List in File</strong>, and choose the file. The file will need to have each the name of each taxon to be selected on a separate line.</li><li>You can now set <strong>Alternative Names</strong> (in the List of Taxa window) to those listed in a separate file that consists of a translation table. On each line of the text file should be the current name of taxon, a tab, and then the new alternative name.</li><li>A new Character Info Strip (Display&gt;Add Char Info Strip when in data matrix) has been added, <strong>Number of Data Cells Strip</strong>, that shows by color the number of cells in a character that contain data. If the value is &lt;10, the value is listed in the info strip, and colors are used; for values 10 or higher, a grayscale is used, with characters containing little data being pale gray, and those with complete data being black.</li><li>The <strong>Pairwise Aligner</strong> tool in the matrix now has an option to check to see if reverse-complementing the DNA sequence being aligned increases the alignment score, and if so, it will reverse complement the sequence before aligning.</li><li>A new tool has been added, the <strong><a  href="Align.html#shifter">Pairwise Shifter</a></strong> tool. With this tool, if you drag a sequence onto a second sequence, will shift the first sequence to increase its match with the second sequence. By default, it finds a region in the first sequence that matches the region in the sequence on which it is dropped, and shifts the first sequence so that those two regions are lined up. You can optionally change the behavior (using the tool's menu or by holding down the Option/Alt key) so that it finds a region in the second sequence that matches the region by which the first sequence was dragged, and move the first sequence so that those two regions are lined up.</li></ul><h4 id="toc19"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.10-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>Various changes were made to make Mesquite more compatible with Java on macOS El Capitan (10.11).</li><li>Shift Other to Match was not shifting sequences that it reverse complemented; thus, there might be some sequences it should have shifted, but didn't, unless you used the tool a second time. This has been fixed (build 728).</li><li>If one dropped sequence files onto a data matrix, and asked to have those sequences adjusted (e.g., aligned) relative to a reference, this was not always done correctly if the reference you selected was something other than the first sequence.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) In the matrix editor, the &quot;Matrix&gt;Move Selected Block&quot; feature will erase the selected data if you asked to move the data beyond the edge of the matrix .</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Pairwise comparison correlation methods gave incorrect counts of positive and negative pairs if all of the terminal taxa of one of the two descendant clades from the root of the tree had missing data. This has been fixed (build 730).</li><li>If only one tree block was selected in the List of Tree Blocks window, then List&gt;Utilities&gt;Duplicate Tree Blocks failed to work. This has been fixed (build 740).</li><li>Incorrect bin boundaries with tracing of continuous characters have been fixed.</li><li>BLAST in Web Browser now gives errors in the web page much less frequently.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) For functions that use an external program through a script (e.g. Zephyr, alignment), there can be issues where the external program's file writing is not complete before Mesquite tries to harvest the results of the script. This can cause the results to be corrupted or incomplete. We have not seen this on Mac OS X or Windows, but we have seen it in Linux. To work around this you can add a delay of a few seconds before the results are harvested, by selecting Delay on Script File Recovery under File&gt;Defaults.</li><li>Many other minor bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc20"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.04"></a>Version 3.04</h3>
 <br />
New Features<br />
<ul><li>The <strong>Pairwise Comparisons</strong> calculations have been reformed in various ways, most of which are enhancements, but some are bug fixes.<ul><li><span style="line-height: 1.5;">Characters with missing data (and other issues, like polymorphisms) can now be handled. The calculation does this by ignoring problem taxa. If taxa have been ignored, there will be a notice in the legend and in the text view of the window. Any taxa with missing or inapplicable state in either the independent or dependent variable are ignored in the calculations. In addition, for specific Pairs Selectors, the following taxa are ignored:</span><ul><li>Pairs Contrasting in State of One Character: a taxon is ignored if its state in the independent variable is polymorphic or uncertain or if it has a state greater than 1 (i.e. is not a binary character), or if its state in the dependent variable is categorical and is polymorphic or uncertain.</li><li>Pairs Contrasting in State of Two Characters: a taxon is ignored if its state in either variable is polymorphic or uncertain or if it the independent variable has a state greater than 1 (i.e. is not a binary character).</li></ul></li><li>The calculation will not be done if there are polytomies in the tree. A notice will appear in the legend and in the text view of the window.</li></ul></li><li>There is a new way to color cells of a matrix (categorical or molecular): <strong>Matching First Taxon</strong>. This colors cells according to whether or not their state matches the state in the first taxon. This is based on a similar coloring method used in MacClade.</li><li>There are new menu items to <strong>delete character matrices</strong>, to <strong>delete tree blocks</strong>, to <strong>close all</strong> tabs of a project, or close all tree windows, or close all character matrix editors, or close all list of characters windows.</li><li><strong>Footnotes</strong> (annotations) and other information about taxa blocks, character matrices, and tree blocks are now shown in the Project Panel at left when the cursor hovers over their names. Footnotes are also now shown in the Text view of corresponding windows (such as a character matrix editor or List of Trees window).</li><li>You can now duplicate a tree block, either using the drop down menu in the Project Panel, or by selecting the tree block in the List of Tree Blocks window and selecting List&gt;Utilities&gt;<strong>Duplicate Tree Blocks</strong>....</li><li>There are new features for the <strong>Has Data In Matrix</strong> column in the List of Taxa window. This column, available by selecting either Column&gt;Has Data in Matrix, or List&gt;Show Columns for All Matrices, indicates whether a taxon has any data in a particular matrix. It is useful for molecular data to see if there is a sequence available for that taxon. You can now copy and paste entire sequences from one taxon to another, by right-clicking on the Has Data cell for a taxon that has data (a green cell), choosing <strong>Copy Data</strong> in the drop down menu, and then right-clicking on another cell and choosing <strong>Paste Data</strong> in the drop down menu.</li><li>The <strong>Tree Inference</strong> submenu has been moved to the Analysis menu.</li><li>The behavior of <strong>tree windows</strong> has changed when tree blocks are deleted. If the window is showing trees from a stored tree block, and all tree blocks were deleted, in the past Mesquite would ask for a replacement source of trees (e.g. simulated trees). Now, the last tree shown continues to be shown, but the tree window switches to the &quot;editing by hand&quot; mode, just as when one starts a tree window with &quot;With Tree to Edit by Hand&quot;.</li><li>When printing or saving a PDF of the tree in the Tree Window, you can choose not to have the tree name added using the menu item Display&gt;Add Name to Printed Tree.</li><li>The graphics shown when branches are grabbed and moved in the Tree Window have been improved.</li><li>A new default option is available under the File&gt;Defaults submenu, <strong>Close Calculations if Matrices Used are Deleted</strong>. This may be useful to enable if you find that you are frequently deleting matrices, and you find it annoying that Mesquite queries you about substituting another source of matrices.</li></ul><br />
Bug Fixes<br />
<ul><li><span style="line-height: 1.5;">(</span><img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" /><span style="line-height: 1.5;">) </span>Adding or deleting taxa could cause trees to be corrupted in some circumstances, leading to error messages about unrecognized taxon names. This has been fixed.</li><li>If you dropped fasta files on a matrix window to add sequences, then did manual alignment on the added sequences, a checksum error could have been reported and prevented you from doing the alignment. This has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Counting the number of stop codons and translating nucleotide to protein sequences could fail (yielding underestimates or partial translations) if codon positions were not assigned in sequence 123123.., for example if it were 1231123.... This has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Pairwise Comparisons with the pair selector &quot;Most Pairs&quot; would given inconsistent results if there was missing data in the characters. This has been fixed by excluding taxa with missing data (see under New Features).</li><li>File or folder names including accented characters (or other unusual characters) could cause preferences to be misremembered or scripts (such as those used in alignment or by <a  href="http://zephyr.mesquiteproject.org">Zephyr</a>) to fail. This has been fixed.</li><li>The last line of a data matrix could sometimes not be easily selected; the row would be selected, and then be immediately deselected. This has been fixed.</li><li>Footnotes (annotations) for taxa blocks, tree blocks, and character matrices were not saved and read properly from files; this has been fixed.</li><li>Various other small bug fixes were made.</li></ul><br />
<br />
<hr />
<h3 id="toc21"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.03"></a>Version 3.03</h3>
 <br />
New Features<br />
<ul><li><span style="line-height: 1.5;">The GBLOCKS algorithm (Castresana, 2000) has been added to Mesquite. This will let you select characters that fit (or don't fit) the GBLOCKS criterion, making it easy to exclude characters using this algorithm. In addition to the</span> options provided in Castresana's program, Mesquite's version allows sites to be rejecting using any user-chosen fraction of gaps (e.g., 0.25), not just &quot;none&quot;, &quot;half&quot;, or &quot;all&quot;. You can also have it count the fraction of identities over only those taxa with non-gaps at the site, rather than all taxa. To use the GBLOCKS algorithm, go into a matrix and choose choose Select&gt;Select Characters&gt;GBLOCKS Selector... or go to a list of characters window and choose List&gt;Select Characters&gt;GBLOCKS Selector. You can then manipulate those characters as you see fit, for example by excluding them, or deleting them.</li><li><span style="line-height: 1.5;">Proportion Unique States in Taxon is a new number that can be calculated for each taxon. It is the proportion of characters for which a taxon has states that are entirely unique among the taxa in the matrix, among the characters for which a taxon has applicable data.</span></li><li><span style="line-height: 1.5;">If you touch on a Boolean Info Strips in the data editor, there is now the option to select all characters that match the criterion. (You can see Info Strips using Display&gt;Add Char Info Strip&gt;Boolean Info Strip&gt; when in the data matrix.)</span></li><li><span style="line-height: 1.5;">When a new file is created, or when a file is imported from FASTA, PHYLIP or other formats, the window now shows the matrix or taxa block by default.</span></li><li><span style="line-height: 1.5;">A new file exporter &quot;Conservative NEXUS&quot; is available in File&gt;Export... This writes a NEXUS file that avoids TITLES and other commands, and yet includes all of the data and metadata. This may be helpful for compatibility with PAUP* and other programs that read NEXUS files.</span></li><li><span style="line-height: 1.5;">There has been a change in the default of writing NEXUS files that may lead to occasional incompatibilities with PAUP* or other programs. When matrices have names, then the &quot;(CHARACTERS=&lt;MatrixName&gt;)&quot; command is inserted into the character specifications (codon positions, groups, weights, parsimony models, etc.). One solution for any incompatibility is to choose File&gt;Export&gt;Conservative NEXUS to export a NEXUS file that avoids such fancy stuff. Alternatively, you can turn off the writing of TITLEs by </span>selecting File&gt;Defaults&gt;Suppress TITLE in NEXUS files, but this may result in the loss of the names of taxa blocks and character matrices..</li></ul><br />
Bug Fixes<br />
<ul><li><span style="line-height: 1.5;">Export for PartitionFinder has several bug fixes. It writes a more complete representation of the data, such as including noncoding regions as a partition.</span></li><li>Bugs in Java 1.6 for OS X 10.10 (Yosemite) were causing crashes. We have therefore facilitated the use of Java 1.7/1.8 on OS X by building executable files (the icons you click on to start Mesquite) that use Java 1.7 or Java 1.8. These are available in the download labelled for OS X 10.10 or higher.</li><li>Reverse and Reverse Complement of whole matrix or contiguous block of whole characters now reverses associated information like codon positions.</li><li>Fixed bug where cloned windows would appear blank.</li><li>Fixed bug on Windows that could affect external alignment programs and packages like Zephyr and Chromaseq that use scripts to run external programs. When names of folders and files included characters like spaces, !, or other special characters, the external program would not work properly and a message might have been given that a file could not be found. This has been fixed.</li><li>Various other small bug fixes were made.</li></ul><br />
<br />
<hr />
<h3 id="toc22"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.02"></a>Version 3.02</h3>
 <br />
New Features<br />
<ul><li>Selecting characters using a text description of character ranges: In the List of Characters window, you can choose List&gt;Select Characters&gt;Character Ranges... and you will be presented with a dialog box in which you can enter text that describes the character ranges to be selected. Two formats are allowed, including one that matches the output of GBLOCKS, in particular, of the form &quot;[23 45] [100 120]&quot;. In that particular example, characters 23 through 45 inclusive, and 100 through 120 inclusive, will be selected.</li><li>Taxa can be reordered (sorted) to match the sequence from left to right in a phylogenetic tree. To do this, have the tree showing in a Tree Window, and select the menu item Tree&gt;Utilities&gt;Reorder Taxa by Tree.</li><li>Various improvements to BLAST tools. Word size added to BLAST options. Also, explicitly set blastn as the task for blastn calls. Ability to append query taxon name name to top hit name.</li></ul><br />
Bug Fixes<br />
<ul><li>Windows beneath the frontmost would mistakenly appear to cover the frontmost on OS X with Java 1.7 and 1.8, due to bugs in those versions of Java. A workaround has been implemented.</li><li>Dialog boxes could not be closed on OS X 10.10 &quot;Yosemite&quot; because of a bug in Java 1.6. A workaround has been implemented, although it means that dialog buttons will be handled immediately upon being touched.</li><li>Checked menu items were often checked or unchecked inappropriately. Fixed.</li><li>Reading tree files, especially simple PHYLIP-Newick tree files, would often generate a crash in MesquiteTree.flagNodesInTree. Fixed.</li><li>In reading simple PHYLIP-Newick tree files, Mesquite had mistakenly used NEXUS conventions as to what constituted punctuation in taxon names. This has been fixed: reading Newick tree files now uses Newick punctuation conventions.</li><li>Coloring of Trace Character by Grayscale, Redscale, Greenscale, etc. had been broken. Fixed.</li><li>Characters not yet given names were mistakenly sorted by their implicit names when the Sort Tool was applied to the names column of the List of Characters window. Fixed: they are now sorted as if they have blank names.</li><li>A bug was introduced just before 3.0's release that broke reconnectability in Zephyr, which means that if you saved a file with an ongoing Zephyr run in 3.0 or 3.01, and then close the file, upon reopening the file Zephyr could not reconnect to that run. Fixed</li></ul><br />
<hr />
<h3 id="toc23"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.01"></a>Version 3.01</h3>
 <br />
New Features<br />
<ul><li>The Tree Window has changed its behavior when you have edited trees and then change the source of trees. By default, it shows the requested tree, but you can recover the edited tree by a menu item under Tree. You can change the behavior using the menu item Edited Tree Handling Options.</li><li>In the windows for List of Character Matrices, List of Taxa Blocks, and List of Tree Blocks, you can now right-click (or control click) on the name to get a popup menu of options for that item.</li></ul><br />
Bug Fixes<br />
<ul><li>Some operations were inadvertently slowed down in 3.0; these have been fixed</li><li>Custom binning for traced continuous characters was broken. Reported by Laura Whitefleet-Smith. Fixed.</li><li>In some contexts (e.g., reading in consensus frequency values stored in a tree in a file) in which there is a mixture of integer and real values, Mesquite did not read in integers. Fixed.</li><li>When new taxa were added to a taxa block there were circumstances in which this caused problems in tree processing yielding apparently reticulate trees. Fixed.</li><li>Some other minor bugs have been fixed.</li></ul><br />
<hr />
<h3 id="toc24"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.0"></a>Version 3.0</h3>
 <h4 id="toc25"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.0-New Features"></a>New Features</h4>
 <ul><li>Source code moved to <a   href="https://github.com/MesquiteProject/MesquiteCore" rel="nofollow">GitHub</a>, and the manual and other documentation moved to a wiki.</li><li>Revised core executables that allow Mesquite to start up smoothly on more recent, security-conscious operating systems.</li><li>Interface revised and cleaned up, with a revised menu structure. Most menu items controlling appearance have been moved within a unified <strong>Display</strong> menu. Some elements that used to be buttons on the screen (such as the choice between window view modes) have been moved into menus. Menus specific to the window are distinguished by being between &lt;&lt; and &gt;&gt;.</li><li>Character Matrices<ul><li>In the List of Character Matrices you can now duplicate and concatenate matrices. You can choose <strong>Duplicate Selected Matrices</strong> under List&gt;Utilities to duplicate the selected matrices, or <strong>Concatenate Selected Matrices</strong> to concatenate them.</li><li>There is a feature to duplicate selected taxa in the List of Taxa or Character Matrix editor windows. Select the rows of the taxa to be duplicated, and choose Taxon Utilities&gt;<strong>Duplicate Taxa...</strong> The character states in all matrices are copied into the duplicate taxa.</li><li>You can now <strong>include or exclude characters directly from within the data editor</strong> by selecting columns and choosing items from the Matrix&gt;Character Inclusion/Exclusion submenu.</li><li>In the data editor, you can now ask to have various characters made paler by using elements in the Display&gt;<strong>Lighten</strong> menu, including having excluded characters made paler.</li><li>There is a new way to create a matrix, called <strong>Condense to Pattern Matrix</strong>, that will make a copy of an existing matrix but one in which &quot;redundant&quot; characters (those characters with an identical distribution of states) are reduced to a single character whose character weight is the number of copies of that pattern in the original matrix.</li></ul></li><li>Molecular data (alignment, editing)<ul><li>The Align package now supports alignment by <strong>MAFFT</strong></li><li>In the data editor, if you are using the Wand tool, and hold the Option/Alt key down, then the tool will select a <strong>contiguous block</strong> of data-containing cells within on taxon (this is useful for DNA sequence data when you are doing manual alignment).</li><li>In the data editor, if you have a selected block of cells, then Matrix&gt;<strong>Move Selected Block</strong> will allow you to specify a particular distance left or right that you wish that block to be moved.</li><li>The Matrix&gt;Alter/Transform&gt;Shift Other To Match feature now has the option to <strong>automatically reverse complement</strong> sequences if that improves the match</li><li>Upon importing sequences into an existing matrix, there is the option to have the <strong>new sequences automatically adjusted</strong> (reverse complemented if necessary, and aligned) relative to a reference sequence.</li><li>The <strong>Sequence Length</strong> feature that calculates the number of non-gaps in a sequence (available in the List of Taxa window [under Columns&gt;Number for Column&gt;Sequence Length] and elsewhere) now has the option to count sites in excluded characters.</li><li>The <strong>Clean Up Matrix</strong> features provides a quick means to reverse complement any sequences that are in the opposite orientation from a reference sequence, as well as conduct an alignment.</li></ul></li><li>Continuous data<ul><li>You can now import into a continuous matrix a character from another matrix or character source, using <strong>Add Character...</strong>, available under Matrix&gt;Alter/Transform Data.</li><li>You can invert (1/x) the values of a continuous matrix using <strong>Inverse of Value</strong>. You can also multiply a character's values by the values from another character (for the respective taxon) using <strong>Multiply by Character</strong>. Both of these are available under Matrix&gt;Alter/Transform Data in the Character Data Editor.</li></ul></li><li>Data Management<ul><li><span style="line-height: 1.5;">OTU ID code and OTU ID code Database fields added to the List of Taxa window. These are used by Chromaseq, as well as by the new tool for renaming taxa based upon ID codes and names contained in a file (see below).</span></li><li><strong><span style="line-height: 1.5;">Rename Taxa Based on ID Code File</span></strong><span style="line-height: 1.5;">: If OTU ID codes are specified for a taxa, and you have a tab-delimited text file the first column of which is the OTU ID, then this tool will allow you to rename taxa based upon the entries in the second and later columns in that text file.</span></li></ul></li><li>Trees<ul><li>A new package to link to tree inference programs, <a  href="http://zephyr.mesquiteproject.org">Zephyr</a>, has been released separately. With it, programs like GARLI and RAxML can be run from within Mesquite.</li><li>There is a new tree drawing form called <strong>Square Line Tree</strong> that is cleaner and more suited for publication-quality trees than Square Tree. It is the new default tree form.</li><li>There is a new value (<strong>Clade Frequencies in Trees</strong>) that can be calculated for nodes (and thus shown on the tree) of the frequencies of that clade in a collection of trees. It is available in Analysis:Tree&gt;Values for Nodes&gt;Clade Frequencies in Trees. You could, for example, have a maximum likelihood tree on the screen, and have a separate collection of bootstrap trees. By using this feature you could ask Mesquite to show on the branches of the tree the percentage of those bootstrap trees that have each of the clades, thus allowing you to calculate the bootstrap values less than 50%.</li><li><strong>Cut Selected Taxa from Tree</strong> (available via Tree&gt;Cut Selected Taxa) will removed taxa that are selected (for example, in the List of Taxa window) from the current tree in the tree window.</li></ul></li><li>Ancestral State Reconstruction<ul><li>Coloring of terminal nodes in Trace Character has changed when the taxon has missing data. A special missing data pattern is shown. This replaces the former coloring, which had presented an estimate of the state the taxon might have.</li></ul></li><li>Data exporting<ul><li>You can now export files for use by <strong>PartitionFinder</strong>. This will also export the configuration file needed by PartitionFinder.</li><li>You can export fused matrices of different types (e.g., DNA and protein) into a Phylip file configured for RAxML. This will <strong>Fused Matrix Export (Phylip/RAxML)</strong> will also write a model file suitable for use in RAxML.</li></ul></li><li>Many small new features</li></ul><br />
<h4 id="toc26"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 3.0-Bug fixes"></a>Bug fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) The Topology Congruent calculator, which calculates whether two trees are topologically congruent, sometimes failed to give the correct answer. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Chart problem: If you have a chart of values across characters, and codon positions are defined, then the bars will be colored by codon position. However, if you then exclude characters, the colors in the chart will be messed up. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Summarize Changes: if the states with the highest state number are ONLY present in taxa that have multiple states for this character, then those states might not be listed in the summary of changes. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Bug whereby various features involving translating from nucleotides to amino acids were in error when a codon triplet was split by a non-coding region. Bug reported by Mitchell Nunn.</li><li>Many miscellaneous bug fixes</li></ul><br />
<hr />
<h3 id="toc27"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.75"></a>Version 2.75</h3>
 <h4 id="toc28"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.75-New Features"></a>New Features</h4>
 <ul><li>In the List of Trees window, there is now a small magnifying glass cursor. When you touch on a tree in the list, the tree is shown in a Tree Window.</li><li>In the Balls&amp;Sticks tree drawing mode, you can now turn off the balls on internal nodes. This can be useful to make an illustration with character states shown only at the tips of the tree. First, turn on Trace Character History, then select the menu item Balls on Internal Nodes to turn off the display of balls on internal nodes.</li><li>Continuous characters can now be discretized by selecting the characters in the Character Matrix Editor, then choosing Discretize Selected Characters under the Utilities submenu. Before doing that, you need to have a categorical data matrix available to which the discretized characters will be added.</li><li>There is a new column available in the List of Taxa window, &quot;Mark Taxa&quot;. With it you can mark taxa with a name, and toggle the named mark off or on. Thus if you name the mark &quot;Resequence&quot; then marked taxa will have &quot;Resequence&quot; in their row in that column. To mark a taxon, touch on its row in that column with the arrow cursor. To change the name of the mark, use the drop down menu at the top of the column.</li><li>It is now possible to rearrange the order of columns in the List windows (List of Taxa, List of Characters, and so on) using the menu item Move Selected Columns... in the Column menu.</li></ul><br />
<h4 id="toc29"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.75-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) There had been a bug with consensus tree calculation when not all taxa were included in the input trees. In this situation, the consensus tree insofar as it was shown was correct, but it could have been an incomplete tree, with some taxa missing. This bug has been fixed.</li><li>If tree blocks were deleted or moved in sequence while a calculation using tree blocks was in effect, and the file saved, on reopening the calculation may have used a different tree block than intended. This bug has been fixed.</li><li>FASTA file export now handles taxon names with spaces and punctuation without adding underscores or quotes.</li><li>Alignment using Muscle sometimes failed under Windows; this has been fixed.</li></ul><br />
<hr />
<h3 id="toc30"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.74"></a>Version 2.74</h3>
 <h4 id="toc31"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.74-New Features"></a>New Features</h4>
 <ul><li>You can now assign different bin boundaries and colors when using continuous characters in Trace Character History. By default there are 11 bins of fixed widths for the shading of the character states, and the colors are either a pre-defined spectrum or grayscale. Now, you can use Set Number of Bins in the Trace menu to change the number of bins, and Set Bin Boundaries to change bin boundaries. To change colors used, double click on the color rectangles in the legend for Trace Character History and choose the new color. These binnings are specific to a given Tree window. The colors are specific to a given Tree window unless you choose Set Custom Colors as Defaults.</li><li>If you have a tree whose stored branch lengths represent not branch lengths but divergence times, then you can ask Mesquite to reinterpret these numbers so as to assign actual branch lengths, using &quot;Branch Lengths from current, reinterpreted as Divergence Times&quot; in the Tree Window's Alter/Transform Branch Lengths submenu. Thus, if a node had &quot;50.4&quot; as branch length but in fact that was a divergence time, and its immediate ancestral node was assigned &quot;56.7&quot;, also a divergence time, then if you used &quot;Branch Lengths from current, reinterpreted as Divergence Times&quot; the node's branch length would be reassigned as 6.3 (= 56.7 - 50.4). This function can be useful if you are importing trees from a program such as TreeAnnotator that assigns divergence times based on BEAST runs. (See item below for a scenario of how this would be used.)</li><li><a name="NumbersAtNodes"></a>If you have information attached to the nodes of a tree as specially attached numbers, you can transfer these numbers to become the tree's new branch lengths using Obtain Branch Lengths from Values Attached to Nodes in the Alter/Transform Branch Lengths submenu. This function can be useful if you are importing trees from a program such as TreeAnnotator that places divergence times as comments to nodes in the tree file. (See item below for a scenario of how this would be used.)</li><li><a name="ImportInformationAtNodes"></a>Reading of trees in NEXUS files was enhanced slightly to permit import of information attached to nodes from BEAST and TreeAnnotator. These programs store important information about nodes as comments beginning with &quot;[&amp;&quot;. This is not standard NEXUS, insofar as the &quot;[&amp;&quot; comments should include just a single word. You will first need to edit the NEXUS text file by replacing all ampersands to percent signs in the tree description (&quot;[&amp;&quot; to &quot;[%&quot;) to permit Mesquite to read the comments. Be careful when you make that replacement: if you change &amp; to % outside the tree descriptions in the TREES blocks, you may render the NEXUS file unreadable). After making this change, read in the tree file. Then open a tree window to show a tree. To check to see if the information was read properly, select the branch info tool (which looks like a question mark) and touch on a branch. Therefore, to obtain branch lengths in the tree derived from TreeAnnotator divergence times, there are three steps:<ol><li>Prepare the tree file from TreeAnnotator by replacing &amp; to % in the comments in the tree descriptions, then read the file with Mesquite.</li><li>In the tree window, show the tree of interest. Select Obtain Branch Lengths from Values Attached to Nodes from the Alter/Transform Branch Lengths submenu, and select the divergence time you want to use (e.g., height or height_median). This will transfer these divergence times into the branch lengths of the tree. The problem with this is that, of course, divergence times are not the same as branch lengths. So, you need to ask Mesquite to recalculate the branch lengths.</li><li>Next, select Branch Lengths from current, reinterpreted as Divergence Timesfrom the Alter/Transform Branch Lengths submenu. This will take the stored branch lengths, intepret them as divergence times, do the appropriate subtractions, and then reassign the branch lengths so as to imply the divergence times. If you now look at the tree with Branches Proportional to Lengths, you will see it is drawn according to those divergence times.</li></ol></li></ul><br />
<ul><li>There is a new menu item in the Tree menu of the tree window, Node-Associated Text, that permits you to display text that is attached to the nodes of a tree. This can be used to display, for instance, text such as the height 95% HPD from TreeAnnotator.</li></ul><br />
<h4 id="toc32"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.74-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Protein NEXUS files written by MacClade may be misread by Mesquite 2.73, with some amino acids being coverted to others (specifically, Serine, S, converted to Cysteine, C). This bug only affects version 2.73, and was introduced in an attempt to fix the problem caused by misformatted NEXUS files created by Clustal. More specifically, it is caused by problems in interpreting protein NEXUS files in which there is a SYMBOLS subcommand in the FORMAT command, which is a frequent occurrence with files written by MacClade. [Thanks to Wendy Moore for reporting this.]</li><li>Top BLAST Matches didn't work if used for amino acid sequences; it now works.</li><li>Various small bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc33"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.73"></a>Version 2.73</h3>
 <h4 id="toc34"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.73-New Features"></a>New Features</h4>
 <ul><li>There is a new method to graft subtrees into larger trees, available in the Tree Window under Tree&gt;Alter/Transform Tree&gt;Graft Other Tree. For instance, you could edit a tree of genera, and a separate tree of species within one of the genera, and then graft the second tree to replace the taxon representing the genus in the first. To use this, all of the taxa need to be in the same taxa block (e.g. by using Merge Taxa &amp; Matrices from File). Then, open up a Tree Window showing the receiving tree (e.g. of genera) and a Tree Window showing the subtree (e.g. of species). Go to the receiving tree and select the terminal taxon name (e.g. the genus name) to be replaced by the grafted tree,and choose Graft Other Tree. If you don't select a terminal taxon in the receiving tree, then the other trees is grafted at the base. To work, the subtree and receiving tree cannot share any taxa.</li><li>There is a new file exporter Export Partitions as Separate NEXUS files that writes a separate file for each character group (partition) of a character matrix. This can be useful, for instance to create files to run ModelTest on separate partitions (e.g. 1st, 2nd and 3rd codon positions). You need to first assign the characters to groups by showing the column &quot;Group Membership (Characters)&quot; in the List of Character window and the drop down menu at the top of this column. Then, choose File&gt;Export and select &quot;Export Partitions as Separate NEXUS files&quot;.</li><li>Two Taxon Name modifiers have been added: (1) Remove Tokens, which allows you to remove a specified number of words (=tokens) from the start or end of taxon names; (2) Simplify Names, which will simplify names in a standard way for use in MrBayes, RAxML, etc. These are available from the Taxon Names submenu.</li><li>There are now more options for coloring matrix cells by the character state. If Matrix &gt; Color Matrix Cells &gt; Character State is chosen for categorical matrices (other than molecular sequence matrices), then two new options will appear under the Matrix menu: (1) &quot;All Characters Use Same Maximum Value for Cell Coloring&quot; will force Mesquite to use the same color for each state for all characters (e.g., state 1 will always be blue), as opposed to having the state colors vary from character to character as a function of the maximum state for each character; (2) &quot;Set Maximum Value for Cell Coloring&quot; allows you to set the state range to be used for this constant coloring.</li><li>A note in a Notes window now persists when the window is closed; in previous versions of Mesquite it was lost when the window was closed. The note can be seen again by going to Window&gt;Current Windows and selecting the Notes window. To delete a note, you now need to do it explicitly by opening up the Notes window and selecting Delete Note from the window's Note menu.</li><li>You can now export ancestral state reconstructions to a tree description with embedded ancestral states in the format of <a   href="http://www.simmap.com/" rel="nofollow">SIMMAP</a> 1.5, which can then be subsequently read by <a   href="http://tree.bio.ed.ac.uk/software/figtree" rel="nofollow">FigTree</a>. To do this, have the ancestral states recontructed in a Tree window using Trace Character History. Then, in the Trace menu, choose Export Ancestral States Trace&gt;SIMMAP 1.5 Format. The tree file saved can be read by FigTree.</li><li>The default color table for non-DNA and RNA data has been changed so that it is easier to distinguish states visually.</li><li>In importing GenBank files, Mesquite no longer puts GenBank accession numbers as footnotes; instead it stores accession numbers in a special location. GenBank accession numbers can then be viewed in the List of Taxa window by choosing GenBank Number under the Columns menu. It also now uses the DEFINITION tag rather than the ORGANISM tag to obtain the sequence name.</li><li>The Tree Window can be opened in a new mode for editing trees by hand. When requesting a new Tree Window a submenu gives two choices: (1) With Trees from Source, which offers to display trees from a source such as Stored Trees or Simulated Trees, and (2) With Tree To Edit By Hand, which shows a default tree that can be edited and saved into a tree block.</li><li>The &quot;checksum&quot; system for checking the integrity of matrices read from files has been improved to eliminate many of the false warnings given, and to give more diagnostic information if an error is detected.</li><li>Pairwise comparison correlation analysis now supplies text output that can be seen in the text window or harvested using Step Through Trees.</li></ul><br />
<h4 id="toc35"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.73-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) ClustalX 2, when writing a NEXUS file, inappropriately inserts an unnecessary but problematic SYMBOLS subcommand in the FORMAT command. This is a serious bug in Clustal, because the implication of this subcommand is incorrect according to the NEXUS standard, and should lead to misinterpretation and data corruption. To preserve the data a program should ignore this incorrect subcommand, as MacClade did. Mesquite 2.72 neither ignored the subcommand nor followed the NEXUS standard: it interpreted the subcommand to override the default ACGT symbols list, which caused the character states in the matrix to be mis-read in a different way. As a protection against such files, Mesquite now ignores the SYMBOLS subcommand for DNA and RNA matrices. (This means that technically Mesquite 2.73 is not NEXUS-compliant, but we believe these files misformatted by CLUSTAL would be encountrered much more frequently than DNA matrices with a legitimate SYMBOLS subcommand.) This bug in Clustal has been there for many years, but apparently only was noticed once the files started being read by Mesquite, which previously treated ill-formatted FORMAT commands in a different way than other programs had done. The bug in ClustalX will be fixed in version 2.0.13 (<a   href="http://www.clustal.org/" rel="nofollow">http://www.clustal.org</a>).</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Export to TNT now writes CCODEs correctly; previously, the characters to which weights, etc., are applied could be in error.</li><li>Various small bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc36"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.72"></a>Version 2.72</h3>
 <h4 id="toc37"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.72-New Features"></a>New Features</h4>
 <ul><li>Parsimony analyses can now be done with meristic data matrices, using Linear Parsimony (analogous to Ordered parsimony with categorical data).</li><li>There is a new tree source, Concatenate Multiple Tree Sources, that brings together several tree sources into one.</li><li><a name="RerootedInformation"></a>New options have been provided for intepreting how information is attached to nodes and branches of trees, which affects how the information behaves when the tree is rerooted. Branch lengths are treated as if they belong to the branch ancestral to a node, and so when the tree is rerooted, the branch length is reassociated so as to continue to refer to the same branch between nodes. However, the name of a clade needs to remain associated with the same node regardless of rerooting. Ned Young pointed out that other information always behaves as if attached to the nodes (not the branches between nodes). We have changed the behaviour so that branch widths and branch color behave as if attached to the branch. We have also provided a function &quot;Reinterpret Internal Node Labels&quot;, available in the Alter/Transform Tree submenu of the Tree menu of a Tree window. When this is selected you can change information associated with the tree as node labels and associate it with the nodes or branches in a different form. This can be useful, for instance, if you have imported a tree from MrBayes with posterior probabilities attached as node labels. Being attached as node labels, the probabilities will be mis-associated with nodes if the tree is rerooted. You can move the probabilities from node labels to numbers attached to branches using Reinterpret Internal Node Labels.</li><li>The reporting of crashes and checksum errors has been improved.</li></ul><br />
<h4 id="toc38"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.72-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) When taxa were deleted, there was a chance that a crash would occur, and this crash could corrupt data. This crash occurred inconsistently, but if it did occur, then you would have received a warning that data might be corrupted. If you ignored the warning, then your character data or other information could have become associated with the wrong taxa. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) When a 3 dimensional (i.e. multi-item) continuous character matrix was exported to a file, and in a few other contexts, the exported matrix would have the items written incorrectly. This did not affect typical continuous matrices, only those with multiple items in each cell of the matrix. This bug has been fixed.</li><li>Reading and writing of DNA matrices was fixed so that it preserved lower case symbols</li><li>Trees exported to PHYLIP and other formats are now written without &quot;decorations&quot; such as branch colors, which tended to render the trees unreadable by other programs</li><li>Sorting of columns now respects numerical order (and thus 2.0 is sorted as between 1.0 and 10.0)</li><li>Various small bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc39"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.71"></a>Version 2.71</h3>
 <h4 id="toc40"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.71-New Features"></a>New Features</h4>
 <ul><li>The crash reporting system now invites you to enter an email address so that the Mesquite developers can contact you if we need more information to diagnose or fix a problem. Entering an email address is optional.</li><li>A new scripting command, getValuesTabbed, can be sent to chart windows to obtain the chart values as a simple list of numbers separated by tabs.</li></ul><br />
<h4 id="toc41"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.71-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>Various small bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc42"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.7"></a>Version 2.7</h3>
 <h4 id="toc43"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.7-New Features"></a>New Features</h4>
 <ul><li>There is a new crash reporting system that can send reports to the Mesquite development team. When certain crashes occur, a dialog box will appear offering to send a report. We urge you to permit the report to be sent; it will help us fix bugs.</li><li>Speed and memory use has been improved considerably for large matrices and trees. This permits many more taxa and characters to be conveniently handled.</li><li>Consistency Index and Retention Index can be calculated for trees with individual characters and with matrices. These are available in various places, for instance in the List of Characters Window, Column&gt;Number for Character&gt;Character Value with current tree...</li><li>NEXUS tree files without translation tables can now be read directly, without first reading or making the corresponding taxa block.</li><li>The BiSSE simulation of trees and characters (available through Trees &amp; Diversification Characters in the Taxa &amp; Trees menu) now asks in the simulating dialog box what prior probability to put on state 1 at the root. The default is to use the equilibrium (stationary) frequencies as the prior, but you can enter 0.0 to indicate the state at the root is necessarily 0, 1.0 to indicate it is necessarily 1.</li><li>Tree simulations normally are set to be repeatable; i.e. if you ask for tree 1 several times, you will get the same tree 1. This can now be turned off by deselecting the menu item Simulation Repeatable. This could be useful in scripting under different scenarios to ensure each scenario uses an independent random number sequence.</li><li>There is a new tree window tool, Node Age Constraints (shaped like an anchor), that permits age constraints to be assigned to nodes. These constraints are not yet used by any standard Mesquite module, but could be used by optional modules. More details are given <a   href="Trees.html#nodeageconstraints" rel="nofollow">here</a>.</li><li>The function to save trees to a file without holding them in memory, previously available under File&gt;Export as &quot;Export NEXUS Tree File from Tree Source&quot; has been moved to Taxa&amp;Trees&gt;Save Trees To File from.</li><li>A new source of trees is Concatenate 2 Tree Sources, which supplies trees from two separate sources as if they were one source. Trees from the first source have an attached value of &quot;1&quot;, while trees from the second source have an attached value of &quot;2&quot;. Concatenating tree sources can be useful for example to concatenate two tree files that are too large to read into memory at once. More details are given <a   href="Trees.html#concat2sources" rel="nofollow">here</a>.</li><li>There is a Revert To Saved menu item in the File menu that reverts the project to its last saved state. Choosing this menu item is the equivalent of closing the file without saving, and then rereading it from disk. If you have linked files open, then choosing it is equivalent to closing all files in the project without saving, and then rereading the home file of the project from disk.</li><li>There is a new submenu, &quot;Tree Inference&quot;, in the Taxa&amp;Trees menu, that shows functions to infer phylogenetic trees. Currently the only built-in tree inference methods are Mesquite's heuristic Add &amp; Rearrange, and simple Cluster Analysis. However, we anticipate that optional packages will soon be released to add functions.</li><li>The Tree Window has a new scripting command, getAsTable, that returns a string in the form of a table that can be converted into a tab delimited table. It summarizes ongoing calculations on the tree such as traced characters and values at nodes. The format is &quot;Calculation Type&lt;tab&gt;node value&lt;tab&gt;node value&lt;tab&gt;...&quot;.</li><li>Character matrices can be marked as &quot;hidden&quot;, which means that they will not be included in most menus and dialog boxes in which you can choose matrices, and they will not appear in the project panel at the left of the window. To set the visibility, go to the List of Character Matrices, Column&gt;Visibility of Matrix. Once the column appears, select the rows for the matrices whose visiblity you want to change, and choose Toggle Visibility from the drop down menu at the top of the column.</li><li>You can now suppress decorations on taxon names in the tree window (e.g. the &quot;*&quot; that indicates a footnote is available) by deselecting Show Footnotes Etc. under Taxon Names in the Drawing window.</li><li>The tree window now has a menu item, Legends to Default Positions, that can recover legends that have become lost off screen.</li><li>The Square Tree drawing form now has a Corners submenu that permits drawing with rounded corners:<br />
<img src="files/roundedCorners.jpg" alt="roundedCorners.jpg" title="roundedCorners.jpg" /></li><li>You can assign a name to your project other than the name of the home file. The project name is then displayed on the window's title bar. To change the name of the project, touch on &quot;Project&quot; near the top of the project panel at the left of the window. In the drop down menu, choose the menu item Name of Project...</li><li>When a duplicate taxa block is read, for instance in a linked file, the first taxa block is not reordered. (Actually, it is reordered temporarily to complete reading of the linked file, but then reverted to the original order.)</li><li>Mesquite now requests more memory by default.</li><li>Mesquite's phone home system now sends the following information to the Mesquite server when you start up Mesquite: your IP address, your operating system, and your version of Java. (Similar information is sent to any web server you visit when you browse the web.) This is sent so that we can better prioritize what operating systems and Java versions to support. If you want to turn this reporting off, deselect &quot;Check for Notices on Mesquite Server&quot; from the Defaults submenu of the File menu.</li><li>With a character matrix open, you can now quickly remove all invariant characters by choosing Remove Invariant Characters from the Alter/Transform menu of the Matrix menu.</li><li>The Merge Taxa command now allows you some control over how the name of the merged taxon is formed.</li></ul><br />
<h4 id="toc44"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.7-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) There was a bug that could scramble the Taxa Associations used to indicate how taxa in one taxa block are associated with taxa in another block (e.g., for genes in species). Scrambling could occur if taxa were reordered then new taxa added, and would result in the wrong taxa being listed as associated. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) There was a bug that could place incorrect data into a continuous valued character matrix if a section was cut, and then Undo called. The error would have likely been obvious. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) There was a bug that caused Mesquite to misinterpret typed character states when editing a character matrix that used certain non-standard lists of symbols. This would have generated a warning about checksum errors. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) The Translate DNA to Protein feature had a bug in that it on occasion removed the last amino acid; this has been fixed.</li><li>Trace Character over Trees could sometimes display the incorrect character name when characters were excluded. This bug has been fixed.</li><li>The BiSSE viewer would record the wrong character number into a file when saved, so that when reopened the file would disply statistics based on a different character than it had previously. This is fixed.</li><li>We have fixed a bug that would cause Mesquite to issue an unnecessary warning that the log file could not be written.</li><li>There was a bug in the Undo system when a cell of a character matrix was being edited; repeated requests to Undo could place unexpected states into the cell. This is fixed.</li><li>Some bugs with tree window scrolling, introduced in version 2.6, have been fixed.</li><li>Various other small bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc45"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.6"></a>Version 2.6</h3>
 <h4 id="toc46"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.6-New Features"></a>New Features</h4>
 <ul><li>There is a new <a   href="Simplifying%20the%20Interface.html" rel="nofollow">interface simplification system</a>.</li><li>There is a new system for installing updates for Mesquite and extra packages. If an update is available, you will be notified when Mesquite starts up. Updates that you choose not to install will still be available via File&gt;Available to Install or Update.</li><li>The Tree Window's automatic sizing of the tree drawing has been improved, and the menu items controlling <a   href="Trees.html#drawingSize" rel="nofollow">tree sizing</a> have changed.</li><li>You can now display boolean values for characters in the information strip at the top of the data matrix, using the Matrix&gt;Add Info Strip&gt;Boolean Info Strip submenu. The two initial booleans whose values you can display in this way are whether or not the character has two or more states, and whether or not it is included. You can remove the information strip by choosing Remove Info Strip in the menu you get when you touch on the info strip.</li><li>You can now export distance matrices from various sources (patristic distances, pairwise sequence distances, geographic distances) to a text file. This can be done by selecting Export from the File menu, and choosing &quot;Export Taxa Distance Matrix&quot;.</li><li>Mesquite can now read treefiles produced by BEST (Bayesian Estimation of the Species Tree), the program by Liang Liu.</li><li>The menu items for Summarize State Changes Over Trees and Summarize Changes In Selected Clade have been moved to the Analysis menu</li><li>The &quot;phone home&quot; system by which Mesquite checks its website to notify you of updates now tells the server which version of Mesquite you are using. Mesquite does not transmit any other data about you, except what your web browser does when you go to a website. (You can disable the &quot;phone home&quot; system in the Defaults submenu of the Edit menu.)</li></ul><br />
<h4 id="toc47"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.6-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) If consensus trees were constructed over trees with branch lengths stored, the average branch lengths of the terminal branches of the consensus tree was incorrect. This has been corrected.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) If new characters or taxa were inserted in an existing matrix or taxa block anywhere but in last position, then subsequent characters or taxa may have had their status in selection sets corrupted. This would not directly lead to a substantive error, but could indirectly if those selection sets were subsequently used for manipulating data. This has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) If new characters or taxa were inserted in an existing matrix, for example by importing FASTA files, then subsequent use of the manual alignment tools (e.g., the multiple sequence splitter) might in rare circumstances cause loss of a few bases from the ends of sequences. As far as we know, Mesquite would have given a warning if this happened. (Thanks to Jolanta Miadlikowksa for reporting this.)</li><li>Mesquite can now more successfully import and export PHYLIP files.</li><li>Undoing some operations in the data matrix caused Mesquite to crash. This has been fixed, but only by preventing the ability to undo some actions that could lead to changes in the entire matrix if the matrix is very large. You can set the maximum size of the matrix that is fully undoable using File&gt;Defaults&gt;Matrix Limits for Undo...</li><li>INTERLEAVED NEXUS files with more than 2000 characters per interleaved block were written incorrectly by Mesquite; this has been fixed. (Thanks to Jolanta Miadlikowksa and Johan Nylander for reporting this.)</li><li>Moving multiple sequences in Mesquite using the &quot;Splits multiple sequences&quot; tool was very slow; this has been fixed.</li><li>Shift to Match Other was not finding many sequences that matched that were matches; it is now much improved.</li><li>Missing data in FASTA files is now correctly written.</li><li>Various other small bugs were fixed</li></ul><br />
<hr />
<h3 id="toc48"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.5"></a>Version 2.5</h3>
 <h4 id="toc49"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.5-New Features"></a>New Features</h4>
 <ul><li>The default appearance of Mesquite's windows have changed. You can alter the general colors used by choosing items from the File&gt;Defaults&gt;Color Theme menu.</li><li><a   href="Trees.html#consensus" rel="nofollow">Consensus tree</a> calculations are now available in the standard Mesquite packages (there are separate consensus tree calculations in the TreeSetViz package). Strict, semistrict, and majority rules consensus are available.</li><li>Results of calculations over multiple trees in the Tree Window are available without scripting through the new <a   href="Trees.html#stepthroughtrees" rel="nofollow">Step through Trees</a> feature.</li>
 <li>Mesquite's parsimony ancestral state reconstructions can now examine individual <a   href="Ancestral%20States.html#mprs" rel="nofollow">Most Parsimonious Reconstructions (MPRs)</a>, and calculate how many there are.</li>
 <li>There are two <a   href="Ancestral%20States.html#SCOT" rel="nofollow">Summarize State Changes Over Trees</a> modes, both of which summarize the frequencies of each sort of character change (e.g., from state 0 to state 1) based upon character histories reconstructed over each of a collection of trees.</li>
 <li>There is a new <a   href="Windows.html#projectPanel" rel="nofollow">Project Panel</a> at the left of most windows to help you keep track of your data objects such as taxa blocks, character matrices and tree blocks. With it you can display, export, rename, or delete these objects.</li><li>The Tree Window has a new Tree Info Panel, available by touching the blue &quot;i&quot; (<img src="files/showInfo.gif" alt="showInfo.gif" title="showInfo.gif" />) in the tree window. This panel shows basic information about the tree and its notes. It also has a Values section in which you can display values from various calculations (character likelihood, etc.).</li><li>Selection of taxa and branches in the Tree Window has been changed to be more natural, using the <a   href="Trees.html#selecting" rel="nofollow">standard arrow cursor via clicking and dragging</a>.</li><li>The Character Matrix editor has various improvements:<ul><li>There is a Matrix Info Panel, available by touching the blue &quot;i&quot; (<img src="files/showInfo.gif" alt="showInfo.gif" title="showInfo.gif" />) in the matrix editor window. This panel shows basic information about the matrix and its entries. For categorical matrices, you can edit character names and character state names; for sequence matrices you can edit a comment for the particular esquence; for continuous matrices you can adjust the matrix's items and their names.</li><li>Copy/paste of molecular sequences have been improved.</li><li>Character names can be written diagonally for categorical and continuous matrices.</li><li>You can now change the symbols used for missing data and inapplicable (gaps) using Matrix&gt;Current Matrix&gt;Missing Data Symbol and Matrix&gt;Current Matrix&gt;Inapplicable Symbol</li><li>For DNA or RNA sequence data, there is a new feature to alter data (available in the Matrix&gt;Alter/Tranform&gt;Convert to RY that converts A or G to R, and C or T to Y.</li></ul></li></ul><br />
<ul><li>There are two new tools for manual sequence alignment. These are described in the a href=&quot;../../mesquite/align/aAlignIntro/index.html&quot;Align manual:<ul><li>Selected Block Mover</li><li>MultiBlock Splitter</li></ul></li></ul><br />
<ul><li>There is a new data type, meristic, for counts (number of teeth, etc.). This is not widely supported in analyses, and has been introduced now for several internal bookkeeping functions.</li><li>The boxes for taxon names and character names in the Character Matrix Editor are now colored independently from the internal cells of the matrix. These have their own submenus in the Matrix menu to control them.</li><li>The Tree Legend has been renamed as &quot;Values for Current Tree&quot;. Its functions are also duplicated in the Tree Info Panel &quot;values&quot; section.</li><li>By default menu items and dialog boxes show all choices, not just the primary choices.</li><li>The tree drawing menu has been reorganized to put Taxon Names Angle under Names.</li><li>Select taxa through search of taxon names. This feature allows you to select taxa if a portion of their names match the specified text. It is available from the Select&gt;Select Taxa submenu in the character matrix editer, or in List&gt;Select Taxa submenu in the List of Taxa window.</li><li>There is a new module to calculate the average tree depth, that is, the average length of the paths between terminal taxa and the root, called &quot;Average Tree Depth&quot;.</li><li><a   href="Trees.html#exportTreesFromSource" rel="nofollow">Export NEXUS Tree File from Tree Source</a>. This feature allows you to save a text file directly from a tree source.</li></ul><br />
<h4 id="toc50"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.5-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) The correlation analysis using pairwise comparisons occasionally would display results that were not complete (not based on a full sample of pairings).</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) When taxa were added, deleted or reordered, some special information such as branch widths, colors and notes associated with the nodes of trees became associated to the wrong nodes. Branch lengths were not affected by this.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Simulations of tree and character evolution simultaneously using &quot;BiSSE Trees &amp; Characters&quot; (available via Trees &amp; Diversification Characters menu item in Taxa &amp; Trees) in some circumstances would have been done using parameters different from those specified in the dialog box, depending on the sequence with which parameters were specified.</li><li>Hennig86/NONA file export had incorrectly counted comments. Hennig86/NONA file export now writes CCODE specifications explicitly, not assuming a default.</li><li>The autotab feature to the quick key selector tool now works correctly in the data editor.</li><li>Many other smaller bugs were fixed.</li></ul><br />
<hr />
<h3 id="toc51"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.01"></a>Version 2.01</h3>
 <h4 id="toc52"><a name="new201" id="new201"></a>New Features</h4>
 <ul><li>&quot;Phone home&quot; system. Mesquite now checks its website on startup to see if there are notices for users, for instance about updates or bugs. (This can be disabled in the Defaults submenu of the Edit menu.) Other packages can also be set to check their websites. This checking does not transmit data about you other than what your web browser does when you go to a website.</li><li>The search facility to help you find features (&quot;Search Features&quot;, available in each window: <img src="files/searchFeaturesAND.jpg" alt="searchFeaturesAND.jpg" title="searchFeaturesAND.jpg" />) has been improved to include menu items and tools in open windows.</li><li>To help minimize user confusion, there are new options in how Mesquite by default presents choices of modules to the user. First, dialog boxes that offer a choice of module may now show more detailed choices. For instance, whereas previously you would be offered a choice &quot;Tree value using character&quot;, now there will also be a list beneath it of the particular values available:<br />
<img src="files/treeValues.jpg" alt="treeValues.jpg" title="treeValues.jpg" /><br />
This new default can be reverted to the old style by deselecting &quot;Show Subchoices in Module Dialogs&quot; in the Defaults submenu. Second, calculations that use characters or matrices can be set to use Stored Characters and Stored Matrices by default, without asking. This default may help novice users who otherwise would be confused by a request about simulations and so on. You can set this default as explained on the<a   href="Characters.html#source" rel="nofollow">Characters</a> page.</li><li>The Defaults menu is now a submenu of the File menu, and is available from all windows.</li><li>GenBank and GenPept import. Mesquite can now import GenBank and GenPept files. If multiple files are dropped onto a molecular matrix of the same sort (e.g., DNA), then these sequences will be added to the matrix.</li><li>Drag and drop of sequence files. You can now drag a collection of FASTA, NBRF, or GenBank files and drop them onto an existing molecular matrix. Mesquite will ask you to specify the file format, and then it will read in the sequences and append them to the bottom of hte matrix</li><li>New alignment tools (in the Align package):<ul><li>Align DNA to Protein. A DNA matrix can be aligned to match that of a translated amino acid alignment.</li><li>Muscle Align. A portion of a sequence matrix can be sent to MUSCLE to be aligned and then reimported back into the matrix.</li><li>Shift to Minimize Stop Codons. Sequences will individually be shifted 0, 1, or 2 based to minimize the number of stop codons in each sequence. This is useful to set the reading frame of each sequence, for example for subsequent translation to amino acids.</li></ul></li></ul><br />
<ul><li>Tree of Life Web Project linking and other online databases. Trees from the Tree of Life Web Project can be downloaded and navigated. In addition, a new tool in the tree window will search for taxon names in various on-line databases, including TreeBase, iSpecies, etc.</li><li>Export NEXUS Tree File. This new exporter allows you to save a tree file (optionally including a taxa block) from a Mesquite project.</li><li>A Search Within Clade Tool is available in the tree window. This tool rearranges the branches above the node touched looking for a better tree. The objective function used (e.g., treelength, or something else) can be chosen by the user. This is similar to MacClade's search tool in that it directly rearranges the current tree.</li><li>Randomly Resolve Polytomies has a new convention. In previous versions it paid little attention to branch lengths, simply using the same convention as when a user moves branches. This resulted in the descendant nodes acquiring different distances from the root than they had had before resolution. The new convention is to preserve distance from root when resolving polytomies. The user however must specify what portion of branch lengths to put on new branches formed during the resolution.</li><li>The Contained Associates tree drawer, most commonly used to display gene trees within species trees, has been renamed as Contained Gene (or Other) Trees.</li><li>The Defaults menu of the Log window has been moved to a submenu of the File menu, and is now available in all windows.</li></ul><br />
<h4 id="toc53"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.01-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>Various bugs concerning behavior of windows and dialogs under Linux and Mac OS X were fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Deep coalescences were miscounted if a species or population lacked any containing genes. Such a species or population would contribute -1 to the deep coalescence count in such a circumstance (it should have contributed 0). This could, for instance, result in a deep coalescence score of a negative number, but it could also simply lower the total. This bug did not affect relative scores among trees (as long as they contained the same set of populations and genes) and so many analyses, such as searching for optimal species trees, would be unaffected. This bug was fixed in build j28 of version 2.01. The very first build of 2.01, j27, released for a few hours, has this bug.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) The Number of Stops in a taxon in some circumstances was underestimated if there were some characters whose codon positions were not specified.</li></ul><br />
<hr />
<h3 id="toc54"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.0"></a>Version 2.0</h3>
 <h4 id="toc55"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.0-New Features"></a>New Features</h4>
 <ul><li>Search system — A &quot;Search&quot; text entry box is now available at the top of most windows (in the Log window, at the bottom). Enter text there and hit Return, and the text will be sought. There are several modes, marked by different symbols:<ul><li>&amp; — Search Features, AND: searches through information about Mesquite's features and returns a description of those features matching ALL of the search terms</li><li>/ — Search Features, OR: searches through information about Mesquite's features and returns a description of those features matching ANY of the search terms</li><li>d — Search Data: searches the current data file or project and returns objects (e.g. taxa, characters, sequences) that match the search string.</li></ul></li></ul>You can choose the mode by touching on the symbol until the desired mode appears. In the Search Features modes, the results also give indication as to how the feature might be accessed. (There may even be a Try It! link that offers to invoke the feature.)<br />
<ul><li>New windowing system — By default, all of the windows of a single project now appear as tabbed panels within a single window. This change was made to limit window proliferation. If you want to &quot;pop out&quot; a tab to be a separate window, hit the curved arrow button within the tab (you can &quot;pop in&quot; the window again using the curved arrow button at the upper left side of the separated window). You can revert to the old system of many windows using the Open Windows as Tabs item in the Defaults submenu of the File menu.</li><li>Wizard-style dialogs — In many contexts Mesquite needs to ask a series of questions when a feature is requested. In previous versions these would appear as a series of dialog boxes; in version 2 these now appear as a single dialog that moves from one question to another, in the style of a wizard. This change was made to make an area in the dialog to give a more thorough explanation of the request, and to minimize the visual distraction of changing dialog sizes. You can revert to the old style of separate dialog boxes (which may work better on some operating systems) using the Use Wizard-style Dialogs item of the Defaults submenu of the File menu.</li><li>Diverse package — There is a new package of modules that estimates and simulates models of speciation and extinction. Some of its features are:<ul><li>Estimating speciation/extinction rates</li><li>Detecting state-dependent differential speciation/extinction (BiSSE)</li><li>Birth/Death tree simulation</li><li>Simulations with evolving characters that affect speciation/extinction</li><li>Lineages through time plots</li></ul></li></ul><br />
<ul><li>Coalescence simulation with migration — Coalescence within a population/species tree can now be simulated with a selected rate of migration per generation among populations.</li><li>Trace Character Over Trees — This feature has been improved in several ways. The fraction of trees without a node is now shown, as is the fraction of trees with the node but not counted as they have ambiguous state assignment (by whatever criteria are chosen). The states at the terminal taxa are now shown. There is a new calculation mode, Average Frequencies Across Trees, available if the ancestral state reconstruction module yields frequencies or probabilities; this calculates, for each state, the average frequency across all trees (including trees in which the frequency for that state is very low or zero).</li><li>Sample Trees Directly from File (now called Sample Trees from Separate File) — One can choose as a source of trees a random subsample of the trees contained within a treefile.</li><li>Undo — You can now undo some changes you might make to the character matrix in the character matrix window. In particular, edits of an individual cell, or changes made by the items in the Matrix&gt;Alter/Transform menu, are now undoable. Other changes (e.g., deleting a character) are not yet undoable.</li><li>Consensus sequences — One or more consensus sequences can be displayed in the character matrix editor.</li><li>Adding sequences to an existing matrix — FASTA or NBRF files dropped onto a matrix of DNA or protein sequences can be read in and appended to those matrices.</li><li>Fusing genes to single matrix — In concatenating matrices, you can now contatenate matrices with different sets of taxa, facilitating fusion of different genes into a single matrix</li><li>Alternative naming schemes for taxa — This feature was introduced in version 1.1 as &quot;Archived names&quot;. It has been expanded to permit multiple <a   href="Taxa.html#alternativeNames" rel="nofollow">archived naming schemes</a>.</li><li>Matrix coloring to aid manual sequence alignment — There are two new Color Cells options that aid with manual alignment. One (&quot;Aligning Colors&quot;) uses colors that emphasize purine-pyrimidine differences; the other highlights sections that appear to be slightly misaligned.</li><li>Various other smaller improvements have been made, for instance you can now import a PHYLIP tree file directly (without having to establish a taxa block first), and you can choose darker colors when assigning colors to objects.</li></ul><br />
<h4 id="toc56"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 2.0-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) There was a bug in writing decimal numbers as strings. Frequently, the 7th or 8th significant decimal place of a number would be written incorrectly. This did not affect how data were stored in the file; it affected only the reporting of results and screen display. This is unlikely to have a serious effect on published results for two reasons. First, it usually affected only the 7th or 8th significant decimal place. Second, the situations in which it affected a more important decimal place were obscure and perhaps rarely used by users. These are: Density output in Scattergram; Parameters Explorer for newer likelihood calculations; Location of changes in Stochastic Character Mapping. In these cases, the 3rd or 4th decimal place could be affected. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) On Windows operating systems, pasting into the data matrix often resulted in the loss of the first character of some rows. This may have been benign, but occasionally would result in data being pasted incorrectly (e.g. 1.234 received as 0.234). This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Among the coalescence calculations, the method to resolve a polytomous contained gene tree failed to resolve all polytomies needed, leaving some unresolved. This resulted in inflated Deep Coalescence counts. This bug has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Fixed bug in which parsimony step matrix would be incorrectly written and could not be re-read properly if it refered to characters with more than 10 states.</li><li>Fixed bug in which Recode dialog box would disappear unexpectedly.</li></ul><br />
<br />
<hr />
<h3 id="toc57"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.12"></a>Version 1.12</h3>
 <h4 id="toc58"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.12-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) In versions 1.1 and 1.11 of Mesquite several of the simulations and randomizations had a bug in choosing the random number seed for the first item simulated or the first randomization. The designed behavior is for the simulations to begin with a random number seed based on the current clock, and then from that calculate base seeds for each of the replicates using the procedure described below under version 1.1 bug fixes. Version 1.12 has been corrected to use this designed behavior. However, in 1.1 and 1.11 for several of the simulations the first replicate always used 1 as its random number seed, although all subsequent replicates used the correct method as designed. This bug did not affect independence of replicates within a simulation of multiple replicates (except when multiple tree blocks were simulated using &quot;Simulated Tree Blocks&quot;), but it did affect independence among separate simulations/randomizations. Since the bug affected only the first replicate, it would not be expected to affect significance values unless very few replicates were used. The affected simulations/randomizations are: Simulated Trees (first tree used seed 1 always); Simulated Tree Blocks (first tree of each block used seed 1 always); Randomly Modify Trees (first tree used seed 1 always); Randomly Modify Matrices (first matrix used seed 1 always).</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) When taxa are selected by the user and Rarefy Tree is used to generate trees with a requested number of taxa randomly deleted, taxa are chosen for deletion randomly only from among those currently selected. In previous versions of Mesquite this did not work correctly, in that frequently fewer than the requested number of taxa were deleted. This may have been evident to the user because the modified tree would have contained more taxa than expected. This bug is fixed in 1.12.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) The likelihood calculations for categorical characters were designed only for characters with a contiguous series of states starting at 0, and to prohibit other characters. That is, a character with states 0, 1, 2 would be permitted, but a character with states 1 and 2, or a character with states 0 and 2, would not be permitted. This requirement was mentioned in the manual, but the requirement was not enforced properly within the program, and thus Mesquite permitted the calculations with some such characters. The reconstructions are incorrect. In version 1.12 the prohibition is correctly in place; Mesquite will refuse to calculate likelihoods with such characters.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) A single character matrix can be modified from various windows, such as the List of Characters window (e.g., by moving, deleting characters) and the Character Matrix Editor. When a character matrix editor is open and a cell is being edited, the editing field could sometimes be misapplied to the wrong character or taxon if elsewhere characters or taxa are deleted, added or changed in order. The effect of this is that the state of one character or taxon can get copied inadvertantly to another character or taxon. This bug has been fixed.</li><li>A bug in the shading of character states for Trace Character History for continuous characters has been fixed. In previous versions there were situations in which a branch was given a color slightly offset from its correct color according to the legend and its state value. (The correct state could always be seen, however, by holding the cursor over the branch.) This has been fixed.</li><li>Fixed bug in Find Sequence (Matching sequence).</li></ul><br />
<br />
<hr />
<h3 id="toc59"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.11"></a>Version 1.11</h3>
 <h4 id="toc60"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.11-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) In previous versions of Mesquite, a Taxa Association between taxa blocks, for example one which records what haplotypes belong to what species, could come to have incorrect linkings if taxa were reordered in the file. This is not directly a substantive bug, but could yield calculations with errors if the user does not notice that taxa have become mis-associated. This bug has been fixed</li><li>Clustal Align now works under Windows</li><li>Various other small bugs were fixed.</li></ul><br />
<br />
<hr />
<h3 id="toc61"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.1"></a>Version 1.1</h3>
 18 May 2006<br />
<h4 id="toc62"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.1-New Features"></a>New Features</h4>
 <ul><li>Character evolution<ul><li><a   href="Processes%20of%20Character%20Evolution.html#pagel94" rel="nofollow">Pagel's (1994) correlation test</a> for 2 binary characters</li><li><a   href="Ancestral%20States.html#scm" rel="nofollow">Stochastic character mapping</a> for simple categorical (non-molecular) data</li><li>Likelihood reconstruction of ancestral states now permits root to have <a   href="Ancestral%20States.html#asymmmk" rel="nofollow">equilibrium frequencies</a> implied by model</li><li>Missing data permitted with categorical character likelihood and continuous parsimony</li></ul></li><li>Gene tree/species trees<ul><li>Fit of gene tree-species tree can now be assessed by counting <a   href="Gene%20Trees.html#duplext" rel="nofollow">duplication and extinction</a> events</li><li>new graphical editor for specifing <a   href="Gene%20Trees.html#establishing" rel="nofollow">associations</a> between containing and contained taxa</li></ul></li></ul><br />
<ul><li>Alignment (see the a href=&quot;../../mesquite/align/aAlignIntro/index.html&quot; Align Package manual)<ul><li>Manual alignment tools in editor (Block Mover, Sequence Pusher)</li><li>Automated pairwise alignment tool in editor</li><li>Submission of a selected region in the matrix to be aligned by ClustalW and then reincorporation of the newly aligned region into the matrix</li></ul></li>
<li><a   href="Analyzing%20Molecular%20Data.html#geneticcodes" rel="nofollow">Genetic Codes</a>
  <ul><li>Assignment of genetic codes to individual characters</li><li>Translation of nucleotide data to protein data</li></ul></li><li>Molecular Data<ul><li>various tools for <a   href="Analyzing%20Molecular%20Data.html#altertransform" rel="nofollow">altering molecular data</a>, including reverse complement</li></ul></li>
  <li><a   href="Analyzing%20Molecular%20Data.html#distances" rel="nofollow">Distances</a>
    <ul><li>Jukes-Cantor, Felsenstein 81, Kimura 2-Parameter, Felsenstein 84 distances</li><li>Great Circle distances in the separately-released <a   href="http://mesquiteproject.org/packages/cartographer/" rel="nofollow">Cartographer package</a></li></ul></li><li>Charts<ul><li>Values for Taxon Pairs now available in Bar &amp; Line Charts and Scattergrams, allowing one to plot, for example, uncorrected distance versus corrected distance for all taxon pairs within a matrix.</li><li>Line charts</li><li>Bar &amp; Line charts can now show cumulative results in various fashions (e.g., cumulative, cumulative average, reverse cumulative average)</li></ul></li>
    <li><a   href="Analyzing%20Molecular%20Data.html#genbank" rel="nofollow">Interactions with GenBank</a>
      <ul><li>Import into matrix the top BLAST matches to sequences selected in matrix</li></ul></li><li>Interaction with <a   href="Interactions&20with%20Other%20Programs.html#mrbayes" rel="nofollow">MrBayes</a><ul><li>live updating of trees produced by ongoing MrBayes analysis</li><li>exporter that <a   href="Analyzing%20Molecular%20Data.html#fusedMatrices" rel="nofollow">merges matrices</a> and includes a MrBayes block summarizing the partitions in the resulting matrix. Merged matrix can be mixed (DNA, protein, and morphological) using MrBayes's modified NEXUS format.</li></ul></li><li>Managing matrices and filesRandomization (reshuffling) of matrices can now respect taxa partitions; also can reshuffle within taxa<ul>
          <li><a   href="Analyzing%20Molecular%20Data.html#fusedMatrices" rel="nofollow">fuse matrices</a> into a single matrix, even if matrices refer to different taxon names (e.g., sequences of different genes), using a <a   href="Gene%20Trees.html#establishing" rel="nofollow">table of correspondences</a> among taxa blocks</li><li><a   href="Taxa.html#alternativeNames" rel="nofollow">alternative taxon names</a> (permitting you to have alternative naming schemes for taxa in a file)</li><li><a   href="Analyzing%20Molecular%20Data.html#includetaxafromfile" rel="nofollow">include taxa</a> from NEXUS, NBRF or FASTA files into current file, e.g. to add newly acquired sequences to an existing matrix</li><li>import/export Phylip tree files</li></ul></li><li><br />
</li><li>PDF file saving of graphics of trees and other windows</li><li>Colors of character tracings can now be changed</li><li>Simplified installation process for Windows computers</li><li>A <a   href="Module%20Index.html" rel="nofollow">complete list</a> of standard modules has been added to the manual.</li></ul><br />
<h4 id="toc63"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.1-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) In previous versions of Mesquite, simulations and randomizations of trees and characters used a shortcut to go to the n'th replicate, cycling the (pseudo-)random number generator n times before entering the replicate. This caused adjacent replicates to be not entirely independent. In most cases we expect this bug would not have had a biasing effect if sample sizes were large. This bug has been fixed by adding 1 to the random number seed after cycling n times to generate independent random number sequences for each replicate.</li><li>Various bugs and inefficiencies affect memory use have been solved in file reading, tree display and printing</li><li>There were some copy/paste bugs, especially under Windows, that have been fixed</li><li>Swapping in heuristic tree search now retains only unique trees</li><li>NEXUS files with high-ASCII characters (e.g., letters with accents) are better handled</li><li>Various other minor bugs and inefficiencies have been fixed</li><li>build h61 (released 23 May 2006) has a few extra bugs fixed: (1) Categorical likelihood calculations no longer complain if root has zero length; (2) Bug in Java 1.5 on Mac OS X caused list dialogs not to respond properly to clicking &quot;Show Secondary Choices&quot;; workaround implemented; (3) Incompatibility with PDAP file exporter solved; (4) directory choice dialog under Mac OS X improved.</li></ul><br />
<hr />
<h3 id="toc64"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.06"></a>Version 1.06</h3>
 <h4 id="toc65"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.06-New Features"></a>New Features</h4>
 <ul><li>The window showing annotatons for taxa, characters and cells of a matrix has been integrated into the Character Matrix Editor as a panel within that window. This makes it easier to annotate without having to manipulate two windows. This panel can be requested by the Show Annotations Panel item in the Matrix menu, or by touching on the small button (<img src="files/annot.gif" alt="annot.gif" title="annot.gif" />) near the lower left of the Matrix Editor (beneath the taxon names). The pencil tool (<img src="files/notesTool.gif" alt="notesTool.gif" title="notesTool.gif" />) has disappeared. Another change is that images in annotations can now be remote (i.e. URL's). Also, the annotations panel is also available in the List of Taxa window (for annotating taxa) and the List of Characters window (for annotating characters). In the list windows, the panel can be requested by touching on the small button (<img src="files/annot.gif" alt="annot.gif" title="annot.gif" />) at lower left of the table.</li><li>Annotations can now be attached to the states of a categorical character, within the State Names Editor (available from the Matrix menu of the Character Matrix Editor). The annotations can be viewed by choosing Show State Annotations in the State_Names menu or by touching on the small button at lower left of the table (<img src="files/annot.gif" alt="annot.gif" title="annot.gif" />).</li><li>A legend is now available for the colors in a Character Matrix Editor. The legend can be shown by selecting Show Color Legend in the Matrix menu, or by touching the small button(<img src="files/colorLegend.gif" alt="colorLegend.gif" title="colorLegend.gif" />) at lower left of the Matrix Editor (beneath the taxon names). If you double click on a color in the matrix, the editor will move to a cell with that color.</li><li>There are buttons at the lower left of the Character Matrix Editor to open the List of Characters window (<img src="files/listC.gif" alt="listC.gif" title="listC.gif" />) and the List of Taxa window (<img src="files/listT.gif" alt="listT.gif" title="listT.gif" />). Reciprocally, the List of Characters window has a button (<img src="files/matrix.gif" alt="matrix.gif" title="matrix.gif" />) to show the Character Matrix Editor.</li><li>Character matrices can now be viewed in a &quot;Birds-eye view&quot; with very narrow columns.</li><li>There is a State Names Strip available for categorical matrices under the Matrix menu. This shows a strip along the bottom of the matrix in which the names of characters and states are shown. Similarly when Bird's Eye View is used for a matrix, a Bird's Eye Closeup Strip appears below the matrix so that you can more clearly see the states of the characters of focus. Both of these strips can be made higher by going to their upper edges, clicking and dragging upward.</li><li>You can now view a character matrix in more than one window, by requesting an Extra Matrix Editor from the Characters menu. This is useful, for instance, if you want to have the cells of the two matrices colored differently.</li><li>The order of characters can now be managed through stored character orders. A Default Order of characters is automatically stored as characters are created. This default order is visible in the List of Characters window as the first column. If the order of characters is accidentally or intentionally changed (for instance using the Sort Tool) then the default order can be recovered easily by applying the Sort Tool to the Default Order column of the List of Characters window. In addition, you can store and reload other character orderings by showing the Stored Character Order in the List of Characters window (Columns menu). You can store the current ordering and recover previously stored orderings using the drop down menu from the Stored Order column heading. When you do this, you will see the numbers in the column change, but the characters will not immediately be reordered. The reason for this is that what is stored and displayed is simply a number for each character marking its position. It is up to you to use that ordering to resort the characters. Thus, to recover a former sequence of characters, load the stored ordering, then touch on its column with the sort tool.</li><li>There are new facilities for searching for sequences in molecular data files. Thes are found in the Edit menu for editor windows of molecular sequence data. These are:<ul><li>Copy Sequence (at bottom of Edit menu): This copies the selected cells of the matrix into the computer's clipboard as a sequence. That is, whereas the standard Copy would place into the clipboard selected pieces of the matrix in tab-delimited text format (e.g., if the sequence AATCA is selected, &quot;A-tab-A-tab-T-tab-C-tab-A&quot; would be copied), this modified Copy Sequence command does not include tabs (thus, &quot;AATCA&quot; would be copied). This style of copying is useful when interacting with programs like <a   href="http://www.genecodes.com/" rel="nofollow">Sequencher (TM)</a>. For instance, if you want to find a piece of sequence in a matrix in Mesquite within a chromatogram viewer of Sequencher, do the following: select the sequence in Mesquite, choose Copy Sequence, then go to Sequencher, select Find Bases, and paste the sequence as the search string.</li><li>Maintain Target Match (in the Find Sequence submenu of the Edit menu): This highlights and maintains highlighted the first occurence of a given sequence in a given taxon. First, you are asked which taxon to search in. Then, it displays a panel like this:<br />
<img src="files/maintainTargetMatch.jpg" alt="maintainTargetMatch.jpg" title="maintainTargetMatch.jpg" /><br />
underneath the matrix. The first button (red X) is to close the panel; the second pauses the search; the third allows you to select another taxon as your focus. If you type a sequence into the text area, the matching sequence (if any) will be highlighted in the matrix. Mesquite is constantly monitoring this text, and so you don't need to give any command to find again if you change the text. This is useful if working with a program like Sequencher. If you see a stretch of sequence while viewing chromatograms that you'd like to find in the matrix in Mesquite, type in the sequence into the text box and you will quickly be taken to it in the taxon.</li><li>Maintain Clipboard Match (in the Find Sequence submenu of the Edit menu): This is similar to Maintain Target Match, except that it obtains the search string not from the text area but from the clipboard. If the clipboard changes, the function will automatically find the sequence again in the matrix. This is useful if working with a program like Sequencher. If you turn on Maintain Clipboard Match, then you can copy stretches of a sequence within Sequencher, and Mesquite will automatically highlight it, without your having to return to Mesquite or give any other command to it. (Mesquite is constantly monitoring the clipboard to see if it changes).</li></ul></li></ul><br />
<ul><li>There is an optional system for assigning and maintaining ID strings to uniquely identify taxa and characters. This can be turned on via the menu items Invent Unique IDs for New Taxa and Invent Unique IDs for New Characters in the Defaults submenu of the File menu. When enabled, newly created characters and taxa are automatically assigned IDs which are maintained even if the taxa or characters are rearranged through reordering, additions or deletions. These unique IDs are little-used in Mesquite at present, but eventually will permit users to reassociate taxa and characters from different files even after names have been changed (for instance, to re-read a tree file saved with different taxon names). The IDs will also be used in database connectivity and collaborative features. To assign IDs to already-created taxa or characters, go to the List of Taxa or List of Characters windows and choose Unique IDs from the Columns menu. IDs can be generated using the drop down menu from the column's header. The one disadvantage of assigning IDs is that the IDs are stored in the Taxa block and Characters block of the NEXUS file, which causes some programs to refuse to read the file. (PAUP* by default will refuse to read these files, but you can request that PAUP* ignore these extra commands by the command &quot;set errorstop = no;&quot; which is available in its Options Menu, Warnings and Errors.)</li><li>There has been a shift in NEXUS file format for annotations — the AN command is now used. This, unfortunately, produces files that are not readable by version 1.05. The reason for this change is to minimize text in the file when dealing with a large matrix with thousands of annotations.</li><li>Pairwise comparisons were previously forced to find all possible pairs. Users can now impose a limit so that only the first n pairs are found and considered. This is important if there are too many pairs to efficiently consider.</li><li>Support for the new <a   href="http://mesquiteproject.org/packages/cartographer/" rel="nofollow">Cartographer</a> package.</li><li>Groups of characters and taxa can now be assigned grayscale values instead of just colors.</li><li>Rarefy Tree (a random tree modifier) now uses taxon selection. If some of the taxa are selected, than Rarefy Tree chooses which taxa to exclude randomly only from among the selected taxa.</li><li>There is a new option to export files, Old-Fashioned NEXUS, for use by programs that understand only a small subset of the NEXUS format such as MrBayes.</li><li>Trees can be drawn in &quot;Eurogram&quot; style by using Square Tree and selecting the menu item Cut Corners.</li><li>The <a   href="Developers Documentation.html" rel="nofollow">Developer's documentation</a> has been updated</li></ul><br />
<h4 id="toc66"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.06-Bug fixes"></a>Bug fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Fixed bug that caused labels for internal nodes (including Bayesian support values) to be scrambled if the branch interchange or reroot tool was used.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) The PAUPConstraintTestParsimony template included with Batch Architecture for doing the test of monophyly has one of the searches incorrectly searching for the optimal trees with the converse of the constraint rather than the constraint itself; this has been fixed.</li><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) Fixed bug that caused tree statistics to be displayed incorrectly in columns in the List of Trees window if trees were deleted.</li><li>Fixed bug in dialog boxes with lists from which to choose that made them unusable under Mac OS X 10.4.</li><li>Fixed bug that prevented use of high ASCII characters in NEXUS files (e.g. &quot;é&quot;, &quot;ü&quot;, and so on).</li><li>Fixed bug that greatly slowed file reading under Java 1.5</li><li>Fixed various issues to improve performance under Mac OS X Java 1.4. (Some windows are still misdrawn, but that is due to bugs in Mac OS X.)</li><li>Fixed other minor bugs (including, but not only: table copy/paste under Windows; use of arrow keys in matrix under Windows; retention of full character information in concatenated matrices; stepmatrix reading when character matrix ambiguous; fill tool with gaps)</li></ul><br />
<hr />
<h3 id="toc67"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.05"></a>Version 1.05</h3>
 <h4 id="toc68"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.05-New Features"></a>New Features</h4>
 <ul><li>added a new Find feature (in the Edit menu) that allows you to search for sequences within DNA data matrices. The first example of a module doing this is &quot;Matching Sequence&quot; which finds sequences matching a target sequence the user enters. The search allows a certain number of mismatches. Optionally, it can search for the reverse, complement and reverse complement of the target sequence.</li><li>added a new item in the Select menu of Character Matrix Editor, &quot;Select Lowercase Ends&quot;. This selects the ends of a DNA sequence up to the first block of N sites in a row with uppercase symbols. The user chooses N. Assuming that lowercase letters are used for less certain base calls, this can be used to select and then trim (by painting with gaps) poorly-sequenced terminal regions.</li><li>the Concatenate Matrices command will now allow molecular sequence data to be concatenated to the end of a matrix of standard categorical characters.</li><li>in the Mac OS X download, added an executable to run Mesquite under Java 1.4. This may be less stable than the standard version running under Java 1.3.1. The Java 1.4 version should be used only if you have Java 1.4.2 update 2 or later on Mac OS X. (The problems with early versions of Java 1.4 appear to be restricted to Mac OS X.)</li></ul><br />
<h4 id="toc69"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.05-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) fixed bug in the Brownian motion simulations of continuous character evolution. Brownian motion puts more change on long branches, less change on short branches. The bug in the simulations caused this contrast between long and short branches to be heightened — long branches had even more change relative to short branches than they should have. (Inadvertantly the effect of branch length had been squared in the calculation.)</li></ul><br />
<br />
<hr />
<h3 id="toc70"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.04"></a>Version 1.04</h3>
 <h4 id="toc71"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.04-New Features"></a>New Features</h4>
 <ul><li>added Utilities submenu to Tree menu of Tree Window. Currently there is only one utility, E-mail Tree, which prepares a text image of the tree and puts it in the body of an e-mail message requested using a web browser's mailto: command.</li></ul><br />
<h4 id="toc72"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.04-Bug fixes"></a>Bug fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) fixed bug in two tree simulators, Uniform speciation (Yule) and Uniform Speciation with Sampling. In versions 1.03 and before, these failed to randomize tip labels, resulting in trees whose shapes matches the model's expectations, but whose particular relationships did not. Thus, in previous versions the taxa were added one at a time to a tree growing through time. For example, if there are 6 taxa, then the divergence between the 6th taxon and its sister species was always the most recent divergence in the tree. While this would not affect questions about general tree symmetry (e.g. &quot;What is the distribution of tree asymmetries under uniform speciation?&quot;) or branch length distributions, it would affect questions about similarity between an observed tree and those expected under a null model of speciation.</li><li>fixed bug that caused a crash when characters were deleted from a matrix which had annotations, images or colors attached to the cells but which didn't have footnotes.</li><li>fixed a few minor bugs.</li></ul><br />
<hr />
<h3 id="toc73"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.03"></a>Version 1.03</h3>
 <h4 id="toc74"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.03-New Features"></a>New Features</h4>
 <ul><li>added automatic file backup, which retains copies of previously saved versions of files. This feature was added to allow the user to return to a previous version in case of errors or other problems. To enable this, in the Log window or Projects &amp; Files window select Automatic NEXUS backup in the Defaults submenu of the File menu. Indicate the number of backups to be saved. If you enter, for instance, 3, then when you save a file named &quot;myfile&quot;, the previous version of the file (if any) will be retained and named myfileBKP1. It will be in the same directory as myfile. The version before that will be called myfileBKP2, the version before that myfileBKP3. When you save the files again, the file ending in BKP3 will be deleted, BKP2 will be named BKP3, BKP1 renamed BKP2, and the newest backup will be called myfileBKP1.</li><li>enabled grayscale shading in Trace Character History for continuous-valued characters. This can be turned on in the Trace menu by selecting the &quot;Use Grayscale If Continuous&quot; menu item.</li><li>Matrices can be compared by selecting cells that differ. To request this, choose &quot;Select by Matrix Comparison&quot; from the Select menu of the character matrix editor. This is similar to the Utility &quot;Compare matrices&quot;, but instead of giving a textual summary of the differences between two matrices, &quot;Select by Matrix Comparison&quot; highlights those cells of the matrix that differ.</li></ul><br />
<h4 id="toc75"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.03-Bug fixes"></a>Bug fixes</h4>
 <ul><li>fixed a bug that caused NOTES blocks to be written in duplicate form into all linked files (opened via Link File...) if all files were saved. This bug was generally innocent except if a linked file was subsequently unlinked from the main data file. This would leave the linked file with a relictual copy of the NOTES block. If the main data file was then edited (e.g., footnotes changed, characters deleted or moved), and then the secondary file was later linked again to the main data file, its relictual (and now out of date) NOTES block could override the current NOTES block in the main file. This could cause footnotes in the main data file to revert to former ones, or to be applied to inappropriate characters or taxa. In addition to fixing this bug, we have added a warning that is given if Mesquite detects that a NOTES block is being overridden. If you suspect that your footnotes may have been corrupted by this bug, please contact us for assistance.</li><li>fixed a bug in Trees Directly from File (now called &quot;Use Trees from Separate File&quot;) that gave spurious warnings.</li><li>fixed various other small bugs</li></ul><br />
<hr />
<h3 id="toc76"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.02"></a>Version 1.02</h3>
 <h4 id="toc77"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.02-New Features"></a>New Features</h4>
 <ul><li>multiple notes with images can now be associated with taxa, characters and cells of character matrices. Prior to 1.02 a single image could be associated with each taxon, or a single footnote could be associated with each taxon, character or cell of the matrix. Now, multiple notes can be attached to each taxon, character, or cell of a matrix. Each note can have a comment, a reference, an image, and an author. Labels can be attached to the image. To use this, select the notes tool in the Character Matrix Editor. If you touch on a cell, a <a   href="Characters.html#annotations" rel="nofollow">Annotations window</a> will appear with the notes for that cell, if any. Buttons in this window can be used to delete and add notes and images, and to make additional notes windows. Behavior of the window can be modified using its Annotations menu. NOTE: the single footnote system still exists and is independent of this multiple notes system.</li><li>MRP matrices can be generated from a set of trees, for use by Matrix Representation with Parsimony analyses to construct supertrees. To use this, select MRP Matrices from trees as a source of matrices (e.g., under Characters menu, Make New Matrix submenu).</li><li>a search facility has been added to the Character Matrix editor and to the List windows. The cell of the matrix found by the search is highlighted and made visible. To use this, select Find String or Find Footnote from the Edit menu. Find String searches through the row names first (e.g., taxon names for the Character Matrix Editor), then the column names (e.g., character names), then the internal cells of the table (e.g., the character state codings). Find Footnote searches the footnotes of the Character Matrix and highlights cells containing the requested text in their footnotes. You can also search for stretches of sequences or state distributions matching the currently selected stretch using Select Same Sequence and Select Same Distribution of the Select menu of the Character Matrix Editor.</li><li>More choices were added for coloring character matrices. Both text and background of a cell can now be colored according to properties of the cell. A paintbrush tool (<img src="files/color.gif" alt="color.gif" title="color.gif" />) was added to allow arbitrary coloring of cells of a character matrix.</li><li>taxa and characters can be moved by a menu item &quot;Move Selected [taxa or characters]To...&quot; in the List of Taxa and List of Characters windows. Prior to 1.02 taxa and characters could be moved by clicking and dragging, but using this it was difficult to move them long distances through a matrix. To use this, select the taxa or characters to be moved in their respective List window, and select Move Selected from the List menu.</li><li>Taxon selecting in the tree window has been modified, with a new taxon select tool (<img src="files/selectTaxa.gif" alt="selectTaxa.gif" title="selectTaxa.gif" />).</li><li>Utilities to modify taxon names have been enhanced and added to the Character Matrix Editor in the Taxon Names submenu of the Matrix menu.</li><li>Added import/export of FASTA files</li></ul><br />
<h4 id="toc78"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.02-Bug Fixes"></a>Bug Fixes</h4>
 <ul><li>(<img src="files/substantiveBug.gif" alt="substantiveBug.gif" title="substantiveBug.gif" />) fixed calculation of Shared Partitions, which counted too many shared partitions between trees in some circumstances if the trees had different terminal taxa included</li><li>resolved conflict in interpetation of tree descriptions that would result in Mesquite's interpreting trees from MrBayes and other programs as having reticulations whenever labels for internal nodes were duplicated. Mesquite now interprets these labels as cosmetic only, and hence not as indicating reticulations</li><li>fixed bug involving the State Names Editor, by which a character might mistakenly receive a name duplicated from another character</li><li>fixed bug in moving taxa or characters by dragging past the end of the matrix</li><li>fixed bug in recoding characters by which state names and footnotes would not be adjusted</li><li>fixed various other small bugs</li></ul><br />
<hr />
<h3 id="toc79"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.01"></a>Version 1.01</h3>
 <h4 id="toc80"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.01-New Features"></a>New Features</h4>
 <ul><li>added on-the-fly filtering and transforming of trees. Thus, trees being supplied from a file or simulation can be filtered according to various criteria or can be transformed with the available tree or branch length alterers. To use this, whenever a source of trees is needed, choose first &quot;Filter Trees from Other Source&quot; (if you want to select trees satisfying a criterion; more details <a   href="Trees.html#filtering" rel="nofollow">here</a>) or &quot;Transform Trees from Other Source&quot; (if you want to transform trees, e.g. reroot, scale branch lengths, etc.; more details <a   href="Trees.html#modifying" rel="nofollow">here</a>), then indicate what source of trees is to be filtered or transformed, and what filter or transformation is to be used.</li><li>added module to read trees directly from a file one at a time; allows processing of much larger tree files (&gt;50,000 trees). To use this, request &quot;Use Trees from Separate File&quot; as your source of trees for charts and other calculations. You can also build a tree block from a sample of trees from a large file by selecting &quot;Include partial contents&quot; from the &quot;Import File With Trees&quot; submenu of the Taxa&amp;Trees menu.</li><li>available tree and branch length alterers can be applied to all trees in a tree block. To use this, go to the Utilities submenu of the List menu of the List of Trees window.</li><li>added a new tree alterer: outgroup rerooting of trees. To reroot trees using a selected set of outgroups, you first must select the outgroup taxa, then, choose the menu item &quot;Root tree with selected taxa as outgroup&quot;, for instance in the Alter/Transform Tree submenu of the Tree menu of the tree window. More details are given on the page on<a   href="Trees.html#modifying" rel="nofollow">trees</a>.</li><li>added integrated BLAST search. To use this, go to a character matrix editor window showing a molecular matrix. Select one or more stretches of sequence and choose BLAST Search from the Search submenu of the Matrix menu. Mesquite will then send a request to your web browser to perform a BLAST search at NCBI's BLAST server, one request for each of the selected stretches of sequence. The results will be available in your web browser.</li><li>added -w (windowless) and -b (backgroundable) flags for running long Mesquite jobs in the background in Unix from the command line</li><li>improved user interface of tables (List Windows, Character Matrix Editor), including navigation with arrow keys and more informative explanations and footnotes</li><li>Added autotab to the Character Matrix Editor. With this, the edited cell shifts automatically to the right or downward when a state is entered. To use this, select the Select and Type tool (with the letter &quot;K&quot;) and use its drop down menu to indicate autotab right or autotab down. Autotab functions only when the Select and Type tool is the active tool.</li><li>selection of taxa, characters and trees is now remembered when files are saved</li><li>calculations of charts and Trace Character Over Trees can now be stopped part way to show partial results</li></ul><br />
<h4 id="toc81"><a name="New Features, Bugs, and History of Features and Bug Fixes--Version 1.01-Bug fixes"></a>Bug fixes</h4>
 <ul><li>fixed tree printing bug by which node numbers and branch lengths always appeared</li><li>fixed bug by which pressing the Delete key while editing in a table would be misinterpreted as a requesting to delete a character, taxon or other object</li><li>fixed bugs in NEXUS file reading and writing (concerning SYMBOLS list, blank character names, tokenization with tab or square brackets, near-duplicate taxa blocks in same file)</li><li>fixed bugs in State Names Editor and Character Matrix Editor concerning footnote display and editing</li><li>fixed bug with Mac OS X 10.3 by which dialog boxes with lists always chose first item (this was not a bug in Mesquite, but rather in Mac OS X, but we have found a workaround)</li><li>fixed bug with Windows by which pie diagrams would be misdrawn as entirely black for traced characters in the Balls and Sticks tree drawing mode (this was not a bug in Mesquite, but rather in Windows/Java, but we have found a workaround)</li><li>several other smaller, less notable bugs were also fixed</li></ul>
<!-- InstanceEndEditable -->

    
    
    </div>
	<!-- This clearing element should immediately follow the #mainContent div in order to force the #container div to contain all child floats --><br class="clearfloat" />
  <div id="footer">
    <p> ©2018 David R. Maddison &amp; Wayne P. Maddison</p>
  <!-- end #footer --></div>
<!-- end #container --></div>

<script type="text/javascript">
<!--


var OverviewOfMesquitePanel = null;
var GettingStartedPanel = null;
var BasicMesquiteOperationPanel = null;
var TaxaTreesCharactersPanel = null;
var AnalysesPanel = null;
var WorkflowsPanel = null;
var GettingHelpPanel = null;
var TechnicalDetailsPanel = null;
var HistoryPanel = null;
var PublishingResultsPanel = null;

if (qsParm['BasicMesquiteOperationPanel']=='open')
 	BasicMesquiteOperationPanel = new Spry.Widget.CollapsiblePanel("BasicMesquiteOperationPanel", {contentIsOpen: true});
else
 	BasicMesquiteOperationPanel = new Spry.Widget.CollapsiblePanel("BasicMesquiteOperationPanel", {contentIsOpen: false});
	
if (qsParm['OverviewOfMesquitePanel']=='open')
 	OverviewOfMesquitePanel = new Spry.Widget.CollapsiblePanel("OverviewOfMesquitePanel", {contentIsOpen: true});
else
 	OverviewOfMesquitePanel = new Spry.Widget.CollapsiblePanel("OverviewOfMesquitePanel", {contentIsOpen: false});
	
if (qsParm['GettingStartedPanel']=='open')
 	GettingStartedPanel = new Spry.Widget.CollapsiblePanel("GettingStartedPanel", {contentIsOpen: true});
else
 	GettingStartedPanel = new Spry.Widget.CollapsiblePanel("GettingStartedPanel", {contentIsOpen: false});

if (qsParm['TaxaTreesCharactersPanel']=='open')
 	TaxaTreesCharactersPanel = new Spry.Widget.CollapsiblePanel("TaxaTreesCharactersPanel", {contentIsOpen: true});
else
 	TaxaTreesCharactersPanel = new Spry.Widget.CollapsiblePanel("TaxaTreesCharactersPanel", {contentIsOpen: false});

if (qsParm['AnalysesPanel']=='open')
 	AnalysesPanel = new Spry.Widget.CollapsiblePanel("AnalysesPanel", {contentIsOpen: true});
else
 	AnalysesPanel = new Spry.Widget.CollapsiblePanel("AnalysesPanel", {contentIsOpen: false});

if (qsParm['WorkflowsPanel']=='open')
 	WorkflowsPanel = new Spry.Widget.CollapsiblePanel("WorkflowsPanel", {contentIsOpen: true});
else
 	WorkflowsPanel = new Spry.Widget.CollapsiblePanel("WorkflowsPanel", {contentIsOpen: false});

if (qsParm['GettingHelpPanel']=='open')
 	GettingHelpPanel = new Spry.Widget.CollapsiblePanel("GettingHelpPanel", {contentIsOpen: true});
else
 	GettingHelpPanel = new Spry.Widget.CollapsiblePanel("GettingHelpPanel", {contentIsOpen: false});
    
if (qsParm['TechnicalDetailsPanel']=='open')
 	TechnicalDetailsPanel = new Spry.Widget.CollapsiblePanel("TechnicalDetailsPanel", {contentIsOpen: true});
else
 	TechnicalDetailsPanel = new Spry.Widget.CollapsiblePanel("TechnicalDetailsPanel", {contentIsOpen: false});

if (qsParm['HistoryPanel']=='open')
 	HistoryPanel = new Spry.Widget.CollapsiblePanel("HistoryPanel", {contentIsOpen: true});
else
 	HistoryPanel = new Spry.Widget.CollapsiblePanel("HistoryPanel", {contentIsOpen: false});
if (qsParm['PublishingResultsPanel']=='open')
 	PublishingResultsPanel = new Spry.Widget.CollapsiblePanel("PublishingResultsPanel", {contentIsOpen: true});
else
 	PublishingResultsPanel = new Spry.Widget.CollapsiblePanel("PublishingResultsPanel", {contentIsOpen: false});


//-->
</script>
</body>
<!-- InstanceEnd --></html>
