<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml"><!-- InstanceBegin template="/Templates/MesquiteTemplate.dwt" codeOutsideHTMLIsLocked="false" -->
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />


<!-- InstanceBeginEditable name="doctitle" -->
<title>Module Index</title>
<!-- InstanceEndEditable -->

<![endif]--><!--[if IE]>
<style type="text/css"> 
/* place css fixes for all versions of IE in this conditional comment */
.thrColFixHdr #sidebar2, .thrColFixHdr #sidebar1 { padding-top: 30px; }
.thrColFixHdr #mainContent { zoom: 1; }
/* the above proprietary zoom property gives IE the hasLayout it needs to avoid several bugs */
</style>
<![endif]-->

<style type="text/css">
<!--
-->
</style>
<link href="CSS/mesquitePackage.css" rel="stylesheet" type="text/css" />
<link href="SpryAssets/SpryCollapsiblePanel.css" rel="stylesheet" type="text/css" />

<script src="SpryAssets/SpryCollapsiblePanel.js" type="text/javascript"></script>
<script type="text/javascript" src="SpryAssets/SpryURLUtils.js"></script>
<script type="text/javascript" src="javascript/mesquite.js"></script>

<!-- InstanceBeginEditable name="head" -->
<!-- InstanceEndEditable -->
</head>

<body bgcolor="#ffffff" class="thrColFixHdr" >

<div id="container">
  <div id="header">
    <table width="100%" border="0" cellpadding="4">
  <tr>
    <td width="51%"><strong id="PackageTitle">Mesquite</strong></td>
    <td width="33%" align="right"><div id="subtitle"><!-- InstanceBeginEditable name="subtitle" -->Module Index<!-- InstanceEndEditable --></div></td>
    <td width="2%">&nbsp;</td>
    <td width="14%">Version 3.6<br />
      27 December 2018<br /></td>
  </tr>
</table>

</div>
  
  
<div id="sidebar1">
  
  
  
  
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('http://www.mesquiteproject.org')">Mesquite Website Home</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('home.html')">Introduction</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('Installation.html')">Download & Installation</a>
  </div>
  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('History%20of%20New%20Features.html')">Recent Changes</a>
  </div>

  <div id="OverviewOfMesquitePanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Overview of Mesquite ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('What%20Mesquite%20does.html');">What Mesquite Does</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Mesquites%20Features.html');">List of Features</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Development%20Team.html');">Development Team</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Contact%20Us.html');">Contact Us</a></td></tr>
		</table>
    </div>
  </div>
  <div id="GettingStartedPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Getting Started ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Installation.html');">Download and Installation</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Additional%20Mesquite%20Packages.html');">Additional Packages</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Beginning%20Mesquite.html');">Beginning Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Example%20Files.html');">Example Files</a></td></tr>
		</table>
    </div>
  </div>
  <div id="BasicMesquiteOperationPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Basic  Operations ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Files.html');">Files</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Menus.html');">Menus</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Windows.html');">Windows</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Charts.html');">Charts</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Selection.html');">Selection</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Modules.html');">Modules</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simplifying%20the%20Interface.html');">Simplifying Mesquite</a></td></tr>
		</table>
    </div>
  </div>
  <div id="TaxaTreesCharactersPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Taxa, Trees, Characters ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Characters%20&%20Matrices.html');">Characters &amp; Matrices</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Taxa.html');">Taxa</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Trees.html');">Trees</a></td></tr>
		</table>
    </div>
  </div>
  <div id="AnalysesPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Analyses ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Character%20Evolution.html');">Character Evolution</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Align.html');">Sequence Alignment</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Analyzing%20Molecular%20Data.html');">Molecular Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Diversification.html');">Diversification</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Gene%20Trees.html');">Gene Trees</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simulations%20&%20Randomizations.html');">Simulations &amp; Randomizations</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Analyzing%20Continuous%20Characters.html');">Continuous Characters</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Studies%20of%20Tree%20Topology.html');">Studies of Tree Topology</a></td></tr>
		</table>
    </div>
  </div>
  <div id="WorkflowsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Workflows ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Managing%20Molecular%20Data.html');">Managing Molecular Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Managing%20Continuous%20Data.html');">Managing Continuous Data</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Interactions%20with%20Other%20Programs.html');">Interactions with other programs</a></td></tr>
		</table>
    </div>
  </div>
  <div id="GettingHelpPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Getting Help ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Learning%20how%20to%20use%20Mesquite.html');">Learning Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Simplifying%20the%20Interface.html');">Simplifying Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Bugs%20and%20Troubleshooting.html');">Bugs &amp; Troubleshooting</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Getting%20Help.html');">Getting Help</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Suggesting%20New%20Features.html');">Your Suggestions</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Glossary.html');">Glossary</a></td></tr>
  			<tr><td><a href="javascript:pageLink('FAQ.html');">FAQ</a></td></tr>
		</table>
    </div>
  </div>
  <div id="TechnicalDetailsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Technical Details ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('How%20Mesquite%20Works.html');">How Mesquite Works</a></td></tr>
  			<tr><td><a href="javascript:pageLink('MesquiteLogSupport.html');">Log & Support Files</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Scripts%20&%20Macros.html');">Scripts &amp; Macros</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Source%20Code.html');">Source Code</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Developers%20Documentation.html');">Developer's Documentation</a></td></tr>
		</table>
    </div>
  </div>
  <div id="HistoryPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">History ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('History%20of%20Mesquite.html');">History of Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('whyMesquite.html');">Why Mesquite?</a></td></tr>
  			<tr><td><a href="javascript:pageLink('History%20of%20New%20Features.html');">New Features & Bug List</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Release%20Dates.html');">Release Dates</a></td></tr>
		</table>
    </div>
  </div>
  <div id="PublishingResultsPanel" class="CollapsiblePanel">
    <div class="CollapsiblePanelTab" tabindex="0">Publishing Results ˇ</div>
    <div class="CollapsiblePanelContent" style="display: block">
  		<table width="100%" border="0" cellpadding="1">
  			<tr><td><a href="javascript:pageLink('Documenting%20Your%20Work.html');">Documenting your work</a></td></tr>
  			<tr><td><a href="javascript:pageLink('How%20to%20Cite%20Mesquite.html');">Citing Mesquite</a></td></tr>
  			<tr><td><a href="javascript:pageLink('Publishing%20Results.html');">Publishing results</a></td></tr>
		</table>
    </div>
  </div>

  <div id="NonCollapsiblePanel">
    <a href="javascript:pageLink('Acknowledgements.html')">Acknowledgments</a>
  </div>

<h3>&nbsp;</h3>
  <!-- end #sidebar1 --></div>

  <div id="mainContent">
    
<!-- InstanceBeginEditable name="content" -->
    
    
<h1 id="toc0"><a name="Mesquite's standard modules"></a><span style="line-height: 1.5;">Mesquite's standard modules</span></h1>
 <span style="line-height: 1.5;">The following is a list of the modules installed by default in this version of Mesquite, along with the explanations of what they do.</span><br />
<ul><li><span style="line-height: 1.5;">Absolute Distance (General) — Absolute distance from a character matrix.</span></li><li><span style="line-height: 1.5;">ACGT Compositional Bias — Returns continuous characters which are the proportion G+C, and separately A, C, G, T, for each taxon in a DNA data set.</span></li><li><span style="line-height: 1.5;">Add and Fill Tree Block — Manages tree block filling as an alteration to a file.</span></li><li><span style="line-height: 1.5;">Add character — Alters continuous data by adding a continuous character. All items of the matrix are similarly modified.</span></li><li><span style="line-height: 1.5;">Add Characters from Source — Adds characters from a source of characters.</span></li><li><span style="line-height: 1.5;">Add item... — Adds an item in each cell of a continuous matrix.</span></li><li><span style="line-height: 1.5;">Add Noise (Gaussian) — Adds normally distributed noise to existing states. Only the first item of a continuous matrix is modified.</span></li><li><span style="line-height: 1.5;">Add Noise to Branch Lengths — Adds noise to branch lengths of tree. Noise is Normally distributed, with variance as selected. By default the variance is be proportional to current branch length, so that if branch length is 10 and you've indicated a variance multiplier of 0.1, the noise added will have a variance of 1.0. Negative branch lengths are not allowed, and are changed to zero.</span></li><li><span style="line-height: 1.5;">Add Note On Tree — Draws editable notes in a tree display.</span></li><li><span style="line-height: 1.5;">Add Selected Taxa To Tree — To a tree with some taxa excluded, adds those taxa that are selected.</span></li><li><span style="line-height: 1.5;">Add Taxa To Tree — To a tree with some taxa excluded, adds taxa.</span></li><li><span style="line-height: 1.5;">Add text to taxon names — Adds text to taxon names.</span></li><li><span style="line-height: 1.5;">Add value — Alters continuous data by adding a value. All items of the matrix are similarly modified.</span></li><li><span style="line-height: 1.5;">Add/Delete Data — Adds and deletes characters and taxa in a Character Matrix editor. This can be done through the Add characters..., Add taxa..., and Delete Selected menu items and via the tools that add taxa and characters where touched.</span></li><li><span style="line-height: 1.5;">Adjust scaling factor of lineage widths — Provides a tool with which to adjust the scaling factor used for lineage widths in a tree window.</span></li><li><span style="line-height: 1.5;">Align DNA to Protein — Realigns a DNA matrix to match the alignment in an amino acid alignment.</span></li><li><span style="line-height: 1.5;">Align Package — Align is a Mesquite package providing tools for the alignment of sequence data.</span></li><li><span style="line-height: 1.5;">Align Score/RC Align Score To Ref. — Reports the Alignment Score / Reverse Complement Alignment Score for a taxon in a data matrix. The score for each taxon is the cost of aligning that taxon's sequence against the reference taxon using Mesquite's default pairwise aligner, divided by the score for aligning the reverse complement of the sequence against the reference. Scores greater than one mean that alignment is better if one of the sequences is reverse complemented .</span></li><li><span style="line-height: 1.5;">Align Sequences — Sends the selected sequence to be aligned.</span></li><li><span style="line-height: 1.5;">Align To Dropped — Supplies an alignment tool that can be used on a set of sequences. Sequences dropped by this tool on another sequence will be aligned to that other sequence (pairwise).</span></li><li><span style="line-height: 1.5;">Alignment Score To Ref. — Reports the Alignment Score for a taxon in a data matrix. The score for each taxon is the cost of aligning that taxon's sequence against the reference taxon using Mesquite's default pairwise aligner. Lower scores mean a better alignment.</span></li><li><span style="line-height: 1.5;">All Branch Lengths to 1.0 — Assigns a value of 1.0 for branch length for all of a tree's branches.</span></li><li><span style="line-height: 1.5;">All Rerootings — Reroots tree at all nodes.</span></li><li><span style="line-height: 1.5;">All Trees — Supplies all possible dichotomous rooted or unrooted trees (only works if number of taxa is &lt;=11).</span></li><li><span style="line-height: 1.5;">All Unassigned Branch Lengths to 1.0 — Assigns a value of 1.0 for branch length for all of a tree's branches that have not yet had a branch length assigned.</span></li><li><span style="line-height: 1.5;">Alter All Matrices — Manages data-transforming modules to transform all matrices in a file.</span></li><li><span style="line-height: 1.5;">Alter Branch Lengths — Alters branch lengths of selected trees.</span></li><li><span style="line-height: 1.5;">Alter Data — Manages data-transforming modules.</span></li><li><span style="line-height: 1.5;">Alter Trees — Alters selected trees.</span></li><li><span style="line-height: 1.5;">Alternative Names — Lists the alternative names for the taxon.</span></li><li><span style="line-height: 1.5;">Amino Acid Molecular Weight — Returns molecular weight of amino acid</span></li><li><span style="line-height: 1.5;">Amino Acid Properties — Returns property of amino acid</span></li><li><span style="line-height: 1.5;">Among-group PCA — Performs among-group principal components analysis on a continous-valued matrix.</span></li><li><span style="line-height: 1.5;">Ancestral States All — Reconstructs ancestral states at nodes for all characters supplied by character source.</span></li><li><span style="line-height: 1.5;">Ancestral States Reconstruction Package Introduction — The ancestral states reconstruction package provides modules to coordinate and display reconstruction of ancestral states on the branches of trees.</span></li><li><span style="line-height: 1.5;">Annotate Branches — Supplies a tool for tree windows to attach and view footnotes for branches.</span></li><li><span style="line-height: 1.5;">Annotation Panel module — Provides tools with which to attach notes (including images) to cells of the matrix and show them.</span></li><li><span style="line-height: 1.5;">Annotation Panel module (list of characters) — Provides tools with which to attach notes (including images) to characters and show them.</span></li><li><span style="line-height: 1.5;">Annotation Panel module (list of taaa) — Provides tools with which to attach notes (including images) to characters and show them.</span></li><li><span style="line-height: 1.5;">Append numbers — Appends to each taxon name the number of the taxon. This will ensure that each taxon name is unique.</span></li><li><span style="line-height: 1.5;">Arbitrarily Resolve State Ambiguities — Alters categorical data by arbitrarily resolving ambiguous sites to one of the possible states.</span></li><li><span style="line-height: 1.5;">Arbitrarily Ultrametricize — Adjusts a tree's branch lengths so that the distances among terminal taxa are ultrametric (i.e. like a molecular clock, all tips reaching to same level). This is not done with any sophisticated smoothing algorithm; rather, branches are just stretched until they reach to same level.</span></li><li><span style="line-height: 1.5;">Arcsine transform — Alters continuous data by arcsine transforming values.</span></li><li><span style="line-height: 1.5;">Assign All Branch Lengths — Assigns a value for branch length for all of a tree's branches.</span></li><li><span style="line-height: 1.5;">Assign Branch Lengths From Node Names — Assigns a value for branch length for all of a tree's branches based upon node names that represent numbers.</span></li><li><span style="line-height: 1.5;">Assign Branch Lengths Less than or Equal to Zero... — Assigns a user-provided value to all branches with lengths less than or equal to zero.</span></li><li><span style="line-height: 1.5;">Assign Selected Branch Lengths — Assigns a value for branch length for all of a tree's selected branches.</span></li><li><span style="line-height: 1.5;">Assign taxon names — Assigns to each taxon name a string followed by a number. For example, if the string is &quot;g&quot;, then taxon 3 will be assigned the name g3.</span></li><li><span style="line-height: 1.5;">Assigned Colors — Provides a tool with which to color cells of a matrix.</span></li><li><span style="line-height: 1.5;">Associated Taxa — Lists and edits what other taxa (e.g. contained or containing) are associated with these.</span></li><li><span style="line-height: 1.5;">Asymmetrical 2-param. Markov-k Model — Defines and maintains simple Markov k-state asymmetrical 2-parameter stochastic models of character evolution.</span></li><li><span style="line-height: 1.5;">Asymmetry Likelihood Ratio Test — Calculates the test statistic for the likelihood ratio test comparing the asymmetrical and one parameter models [2ln(L(Asymm.)/L(Mk1)], on a tree for a given character.</span></li><li><span style="line-height: 1.5;">Attached Value — Supplies a number attached to the tree</span></li><li><span style="line-height: 1.5;">Attachment — Displays an item attached to trees.</span></li><li><span style="line-height: 1.5;">Augment Tree Randomly — Augments tree by random placement of excluded taxa. There are three modes: adding only to original branches, ignoring lengths; adding to any branch, considering branch length; adding to any branch, ignoring branch length. With lengths considered, probability of placing taxon on a branch is proportional to the branch's length.</span></li><li><span style="line-height: 1.5;">Average Difference of Values Associated with Nodes — Calculates the average differences in values associated with nodes across those clades shared between two trees (excludes the clade consisting of all taxa).</span></li><li><span style="line-height: 1.5;">Average of Several Distances — Average of several distances among taxa, e.g. average from several data matrices.</span></li><li><span style="line-height: 1.5;">Average Taxon Value among Trees — Averages among trees a value calculated for a taxon using a tree.</span></li><li><span style="line-height: 1.5;">Average Tree Depth — Calculates the average path length (in branch length) from terminals to the root, treating unassigned lengths as 0. The length of the root is not counted.</span></li><li><span style="line-height: 1.5;">Average Tree Value — Calculates the average of some value for trees in a tree block.</span></li><li><span style="line-height: 1.5;">Balls &amp; Sticks — Draws trees with spots and the nodes and thin lines for branches.</span></li><li><span style="line-height: 1.5;">Bar &amp; Line Chart — Helps make bar &amp; line charts.</span></li><li><span style="line-height: 1.5;">Bar &amp; Line Chart for Character Matrices — Shows chart displaying values (such as treelength, etc.) for a series of character matrices.</span></li><li><span style="line-height: 1.5;">Bar &amp; Line Chart for Characters — Shows chart displaying values (such as likelihoods, parameter values, parsimony counts, etc.) for a series of characters.</span></li><li><span style="line-height: 1.5;">Bar &amp; Line Chart for Taxa — Makes a chart displaying values (such as proportion of gaps in a sequence, etc.) for each of a series of taxa.</span></li><li><span style="line-height: 1.5;">Bar &amp; Line Chart for Trees — Makes a chart displaying values (such as likelihoods, parsimony scores, imbalance statistics, correlations,etc.) for a series of trees.</span></li><li><span style="line-height: 1.5;">Basic Draw Names for Tree Display — Draws taxon names on a tree. Chooses orientation of names according to orientation of tree.</span></li><li><span style="line-height: 1.5;">Basic File Coordinator — Coordinates the reading, maintenance and linkages of files, which include Taxa, Trees, CharacterData, and other objects. This module is hired directly by the Mesquite trunk module.</span></li><li><span style="line-height: 1.5;">Basic Tree Draw Coordinator — Coordinates the drawing of a tree by maintaining the basic TreeDisplay and by hiring a DrawTree module.</span></li><li><span style="line-height: 1.5;">Basic Utilities Package Introduction — Provides basic utilities for Mesquite.</span></li><li><span style="line-height: 1.5;">Batch Architect Package Introduction — Provides facilities to coordinate the simulation of data and generate batch files for replicated analyses.</span></li><li><span style="line-height: 1.5;">Batch File Template Manager — Manages templates used in saving batch files.</span></li><li><span style="line-height: 1.5;">Birth/Death Process Trees — Generates tree by simple birth/death model with a constant rate of speciation (birth) and of extinction (death).</span></li><li><span style="line-height: 1.5;">BiSSE Ln Likelihood Difference — Calculates the difference in log likelihoods between two BiSSE speciation/extinction models.</span></li><li><span style="line-height: 1.5;">BiSSE Net Diversification Likelihood — Calculates likelihood with a tree of a species diversification model whose rates (r = spec. - ext.; a = spec./ext.) depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</span></li><li><span style="line-height: 1.5;">BiSSE Net Diversification Likelihood (Calculator) — Calculates likelihood with a tree of a species diversification model whose rates (r = spec. - ext.; a = spec./ext.) depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</span></li><li><span style="line-height: 1.5;">BiSSE Speciation/Extinction Likelihood — Calculates likelihood with a tree of a species diversification model whose speciation and extinction rates depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</span></li><li><span style="line-height: 1.5;">BiSSE Speciation/Extinction Likelihood Calculator — Calculates likelihood with a tree of a species diversification model whose speciation and extinction rates depend on the state of a binary character (BiSSE model, Maddison, Midford &amp; Otto, 2007).</span></li><li><span style="line-height: 1.5;">BiSSE Trees &amp; Characters — Generates tree by a speciation/extinction model in which a character controls rates of speciation/extinction. Time to next event drawn from negative exponential distribution.</span></li><li><span style="line-height: 1.5;">BLAST in Web Browser — BLASTs selected data against GenBank.</span></li><li><span style="line-height: 1.5;">BLAST NCBI GenBank Server — BLASTs the NCBI GenBank server at NIH</span></li><li><span style="line-height: 1.5;">Boolean for Character (in List of Characters window) — Supplies booleans for characters for a character list window.</span></li><li><span style="line-height: 1.5;">Boolean for Taxon (in List of Taxa window) — Supplies booleans for taxa for a taxon list window.</span></li><li><span style="line-height: 1.5;">Boolean for Tree (in List of Trees window) — Supplies booleans for trees for a trees list window.</span></li><li><span style="line-height: 1.5;">Boolean Info Strip — Shows an strip of boxes near the top of each a character column that shows the value of a boolean for that character in gray or white.</span></li><li><span style="line-height: 1.5;">Boolean Tree Value — Gives a tree a value of 1 if it satisfies the criterion, 0 otherwise.</span></li><li><span style="line-height: 1.5;">Bootstrap Resample Matrix — Resamples characters from a matrix for use in bootstrap methods.</span></li><li><span style="line-height: 1.5;">Branch Information — Provides a tool that shows information about branches.</span></li><li><span style="line-height: 1.5;">Branch Length from Root — For each taxon, calculates the sum of the branch lengths from that taxon to the root of the tree.</span></li><li><span style="line-height: 1.5;">Branch Lengths Adjust — Provides a tool to adjust branch lengths of trees.</span></li><li><span style="line-height: 1.5;">Branch Lengths from current, reinterpreted as Divergence Times — Reassigns branch lengths under the assumption that currently assigned branch lengths are actually divergence times.</span></li><li><span style="line-height: 1.5;">Brownian Model — Initializes Brownian motion character model for likelihood and other probability calculations.</span></li><li><span style="line-height: 1.5;">Calculated value for taxon — Supplies values for taxa reinterpreted as if characters.</span></li><li><span style="line-height: 1.5;">Canonical Variates Analysis — Performs canonical variates analysis on a continous-valued matrix.</span></li><li><span style="line-height: 1.5;">Catalogue of Life — Provides a URL to a taxon in the Catalogue of Life</span></li><li><span style="line-height: 1.5;">Categorical Data Package Introduction — Provides basic facilities for handling categorical characters.</span></li><li><span style="line-height: 1.5;">Cell Wand (data) — Provides a tool with which to select cells in a matrix automatically.</span></li><li><span style="line-height: 1.5;">Character Compositional Bias — Calculates the percent of taxa with particular nucleotides (GC bias, or individual frequency of A, C, G or T) for a character.</span></li><li><span style="line-height: 1.5;">Character Group Color — Colors by Character Group.</span></li><li><span style="line-height: 1.5;">Character Group Color in List — Shows color assigned to character group.</span></li><li><span style="line-height: 1.5;">Character Included — A boolean that is true if a character is currently included, and false if the character is currently excluded.</span></li><li><span style="line-height: 1.5;">Character Inclusion — Shows current character inclusion in character list window.</span></li><li><span style="line-height: 1.5;">Character Likelihood — Calculates the negative log likelihood of a tree for a given character.</span></li><li><span style="line-height: 1.5;">Character List — Makes windows listing characters and information about them.</span></li><li><span style="line-height: 1.5;">Character Loadings — Returns loadings of a character in an ordination</span></li><li><span style="line-height: 1.5;">Character Matrices List — Makes windows listing character data matrices and information about them.</span></li><li><span style="line-height: 1.5;">Character Matrices Package Introduction — Provides basic management and utilities for character data matrices.</span></li><li><span style="line-height: 1.5;">Character Matrix Editor — Makes editor windows to edit character data.</span></li><li><span style="line-height: 1.5;">Character Matrix manager — Coordinates the management of character data matrices.</span></li><li><span style="line-height: 1.5;">Character matrix selection coordinator — Coordinates selecting taxa and characters in data matrix.</span></li><li><span style="line-height: 1.5;">Character Models List — Makes windows listing character models and information about them.</span></li><li><span style="line-height: 1.5;">Character Reference Strip — Shows the strip at bottom of matrix with character reference.</span></li><li><span style="line-height: 1.5;">Character selection coordinator — Coordinates character selection.</span></li><li><span style="line-height: 1.5;">Character Sort (data) — Provides a tool with which to sort characters automatically.</span></li><li><span style="line-height: 1.5;">Character Source — Coordinates the supply of characters from various sources of characters.</span></li><li><span style="line-height: 1.5;">Character Source — Coordinates the supply of characters from various sources of characters.</span></li><li><span style="line-height: 1.5;">Character value using other character — Coordinates the calculation of a number for a character based on another character (e.g., a correlation between the two characters).</span></li><li><span style="line-height: 1.5;">Character value with current tree — Coordinates the calculation of a number for a character based on a current tree.</span></li><li><span style="line-height: 1.5;">Character Value with Respective Tree — Coordinates the calculation of a number for a character based on a respective tree.</span></li><li><span style="line-height: 1.5;">Character value with tree — Coordinates the calculation of a number for a character based on a tree.</span></li><li><span style="line-height: 1.5;">Character Wand (data) — Provides a magic wand tool with which to select characters automatically.</span></li><li><span style="line-height: 1.5;">Character-Associated Diversification — Coordinates analyses for the effect of a character on diversification (speciation/extinction).</span></li><li><span style="line-height: 1.5;">Character/State Information — Exports character and state information for categorical matrix.</span></li><li><span style="line-height: 1.5;">Characters from Matrix Source — Supplies characters from source of matrices.</span></li><li><span style="line-height: 1.5;">Characters from Ordinations — Supplies a character from an ordination of an existing matrix.</span></li><li><span style="line-height: 1.5;">Characters Scattergram — Makes a scatterplot displaying two values (such as likelihoods, parsimony scores, etc.) for a series of characters.</span></li><li><span style="line-height: 1.5;">Chart from Instruction File — Displays a chart summarizing data contained in files, as directed by an instruction file.</span></li><li><span style="line-height: 1.5;">Charting Package Introduction — Provides basic charting functions.</span></li><li><span style="line-height: 1.5;">Circular tree — Draws trees in circular form.</span></li><li><span style="line-height: 1.5;">Clade Frequencies in Trees — Calculates, for each clade in the tree, the frequency of that split among the trees in a specified tree source.</span></li><li><span style="line-height: 1.5;">Clade Match with Another Current Tree — Calculates a number relating each clade in the tree to another current tree (e.g., a tree-to-tree distance metric).</span></li><li><span style="line-height: 1.5;">Classic Square tree — Draws trees with standard square branches (&quot;phenogram&quot;)</span></li><li><span style="line-height: 1.5;">Clear All Node Names — Clears all node names from the tree.</span></li><li><span style="line-height: 1.5;">Clustal (DNA/RNA) — Imports Clustal files that consist of DNA or RNA sequence data.</span></li><li><span style="line-height: 1.5;">Clustal (protein) — Imports Clustal files that consist of amino acid sequence data.</span></li><li><span style="line-height: 1.5;">ClustalW Align — Sends the selected sequence to ClustalW to align.</span></li><li><span style="line-height: 1.5;">Cluster analysis — Supplies trees obtained from cluster analysis on distance matrices.</span></li><li><span style="line-height: 1.5;">Coalescence Contained within Current Tree — Generates tree by a simple coalescence model of a neutral gene with constant population size, within a current species tree from a Tree window or other tree context. Branch lengths are assigned according to generation of coalescence.The species tree used is a current tree found in a Tree Window or other tree context.</span></li><li><span style="line-height: 1.5;">Coalescence in Current Tree with Migration — Generates tree by a simple coalescence model of a neutral gene with constant population size, within a current species tree from a Tree window or other tree context. Branch lengths are assigned according to generation of coalescence. The default population size is 10000. The species tree used is a current tree found in a Tree Window or other tree context. Migration is allowed; the default probability is 0.00001 per individual generation. Migration can be isolated to a particular generation, or spread throughout the depth of the species/population history.</span></li><li><span style="line-height: 1.5;">Coalescence Package Introduction — Provides simulations and calculations for coalescence (genetic drift) within populations.</span></li><li><span style="line-height: 1.5;">Coalescence Simulated within Current Tree — Generates tree by a simple coalescence model of a neutral gene with constant population size, within species trees. Branch lengths are assigned according to generation of coalescence.The species tree used is a current tree found in a Tree Window or other tree context.</span></li><li><span style="line-height: 1.5;">Coalescent Trees — Generates tree by coalescence within a single panmictic population.</span></li><li><span style="line-height: 1.5;">Codon Position Rates Model — Defines and maintains models that specify rates for codon positions.</span></li><li><span style="line-height: 1.5;">Collapse Gaps to Right — Collapses gaps in the selected block of cells to the right, to yield unaligned sequences.</span></li><li><span style="line-height: 1.5;">Collapse Sequences to Left (Remove all Gaps) — Collapses sequences in the selected block of cells to yield unaligned sequences, by removing all gaps.</span></li><li><span style="line-height: 1.5;">Collapse Zero-length Branches — Collapses zero-length branches to yield polytomes.</span></li><li><span style="line-height: 1.5;">Colless's Imbalance — Calculates Colless's Imbalance statistic for tree asymmetry, normalized by maximum asymmetry. Polytmous nodes are ignored in the calculation.</span></li><li><span style="line-height: 1.5;">Color AAs to Check — Colors the cells of a DNA matrix by stop codons, partial triplets, and missing data.</span></li><li><span style="line-height: 1.5;">Color Branches — Provides a tool with which to color branches in a tree window.</span></li><li><span style="line-height: 1.5;">Color Branches by Partition — Colors the tree's branches by the taxon partition.</span></li><li><span style="line-height: 1.5;">Color By Amino Acid — Colors the cells of a DNA matrix by the amino acids for which they code.</span></li><li><span style="line-height: 1.5;">Color By Cell Value — Colors the cells of a character matrix according to a value for that cell or a moving window of cells.</span></li><li><span style="line-height: 1.5;">Color By Character Value — Colors the cells of a character matrix according to a value for the character or a moving window of characters.</span></li><li><span style="line-height: 1.5;">Color By Footnote — Colors the cells of a character matrix by whether or not they have a footnote.</span></li><li>Color By Match to First Taxon — Colors the cells of a character matrix blue if they match the first taxon, yellow if different, grey if cell or first taxon has missing or inapplicable data.</li><li><span style="line-height: 1.5;">Color By State — Colors the cells of a character matrix by their contained character states.</span></li><li><span style="line-height: 1.5;">Color By Taxon Value — Colors the cells of a character matrix according to a value for the Taxon or a moving window of taxa.</span></li><li><span style="line-height: 1.5;">Color For Aligning — Colors the cells of a character matrix for easy of alignment.</span></li><li><span style="line-height: 1.5;">Color If Excluded — Colors the excluded characters gray.</span></li><li><span style="line-height: 1.5;">Color Taxa — Provides paintbrush to color taxon names.</span></li><li><span style="line-height: 1.5;">Compact to (X,Y) matrix — Converts a matrix of alternating X and Y columns (e.g. for landmarks) to a matrix each cell of which has X,Y values.</span></li><li><span style="line-height: 1.5;">Compare Bases with Taxon — Compares nucleotide bases of selected taxa with those of a chosen taxon.</span></li><li><span style="line-height: 1.5;">Compare Other Matrix — Compares matrix in data editor with another.</span></li><li><span style="line-height: 1.5;">Compare with Another Current Tree — Calculates a number relating a tree and another current tree (e.g., a tree-to-tree distance metric).</span></li><li><span style="line-height: 1.5;">Compare with Other Trees — Calculates a number relating one tree to another in a sequence (e.g., a tree-to-tree distance metric), such that the i'th tree is compared to the i'th other tree in the sequence.</span></li><li><span style="line-height: 1.5;">Composite DNA Simulation Model — Defines and maintains simple stochastic models of DNA evolution for simulations.</span></li><li><span style="line-height: 1.5;">Concatenate Multiple Tree Sources — Concatenates multiple tree sources to yield a tree source supplying trees from each source. The trees are marked with values to indicate the source.</span></li><li><span style="line-height: 1.5;">Concatenate Other Matrix — Concatenates matrix onto one in data editor. Assumptions like weights and character models are NOT transferred. For categorical data, state names are not included. For continuous data, new items may need to be created to accommodate differences in items between the matrices.</span></li><li><span style="line-height: 1.5;">Concatenate Respective Matrices from Two Sources — Supplies matrices by concatenating respective matrices from two sources. The i'th matrix supplied is the concatenation of the i'th matrix from each of the two sources. Assumptions like weights and character models are NOT transferred. For categorical data, state names are not included. For continuous data, new items may need to be created to accommodate differences in items between the matrices.</span></li><li><span style="line-height: 1.5;">Concatenate Selected Matrices — Concatenates selected matrices in List of Character Matrices window. Only those compatible with first selected are concatenated into it.</span></li><li><span style="line-height: 1.5;">Concatenate tree blocks — Concatenates selected tree blocks into a new tree block.</span></li><li><span style="line-height: 1.5;">Concatenate Two Tree Sources — Concatenates two tree sources to yield a tree source first supplying trees from one source, then supplying trees from another. The trees are marked with values 1 and 2 to indicate the source.</span></li><li><span style="line-height: 1.5;">Condense taxon names — Condenses taxon names, for example for use by programs that cannot handle long names or names with punctuation.</span></li><li><span style="line-height: 1.5;">Condense to Matrix of Patterns — Condenses a categorical matrix to one having exactly one character of each pattern with that character weighted by the number of copies of that pattern in the original matrix.</span></li><li><span style="line-height: 1.5;">Condensed Tree Source — Produces a block of trees that are a condensed version of another block of trees, with duplicate trees removed.</span></li><li><span style="line-height: 1.5;">Consensus Sequence Strip — Displays a consensus sequence as in info strip in a character matrix editor.</span></li><li><span style="line-height: 1.5;">Consensus State for Character — </span></li><li><span style="line-height: 1.5;">Consensus Tree — Supplies consensus tree from a block of trees.</span></li><li><span style="line-height: 1.5;">Consensus Trees from Tree Blocks — Supplies consensus trees from Tree blocks.</span></li><li><span style="line-height: 1.5;">Consensus Trees Package Introduction — Calculations for consensus trees.</span></li><li><span style="line-height: 1.5;">Consistency Index for Character — Calculates the Consistency Index (CI) for a tree and character.</span></li><li><span style="line-height: 1.5;">Consistency Index for Matrix — Calculates the Consistency Index (CI) for a tree and matrix.</span></li><li><span style="line-height: 1.5;">Constellation — Draws trees with spots at the nodes in a more or less unrooted way.</span></li><li><span style="line-height: 1.5;">Contained Gene (or Other) Trees — Draws wide trees containing trees of associates (e.g., species tree containing gene tree)</span></li><li><span style="line-height: 1.5;">Continuous Data Package Introduction — Provides basic facilities for handling continuous-valued characters.</span></li><li><span style="line-height: 1.5;">Continuous Items Editor — Edits items for continuous data matrix within the editor info panel.</span></li><li><span style="line-height: 1.5;">Continuous state of taxon — State of a continuous character.</span></li><li><span style="line-height: 1.5;">Convert Negative Branch Lengths to Zero — Adjusts a tree's branch lengths so that negative branch lengths are zero</span></li><li><span style="line-height: 1.5;">Convert Polymorphisms to Uncertainties — Alters categorical data by converting polymorphic entries to uncertainties.</span></li><li><span style="line-height: 1.5;">Convert to RY — Converts nucleotide data to RY data in selected region.</span></li><li><span style="line-height: 1.5;">Convert Uncert./Polymorph. to Missing — Alters categorical data by converting uncertain or polymorphic entries to missing data.</span></li><li><span style="line-height: 1.5;">Convert Uncertainties to Polymorphisms — Alters categorical data by converting uncertain entries to polymorphisms.</span></li><li><span style="line-height: 1.5;">Coordinator of voucher information — Finds suppliers of voucher information.</span></li><li><span style="line-height: 1.5;">Correl Package — Correl is a Mesquite package for examining character correlation.</span></li><li><span style="line-height: 1.5;">Correlation Analysis — Performs a Correlation analysis for categorical characters.</span></li><li><span style="line-height: 1.5;">Current Codon Positions — Supplies current codon positions applied to characters for character list window.</span></li><li><span style="line-height: 1.5;">Current Genetic Codes — Shows current genetic codes applied to characters in character list window.</span></li><li><span style="line-height: 1.5;">Current Genetic Codes — Supplies the currently assigned genetic code (e.g., &quot;standard&quot;) for a character.</span></li><li><span style="line-height: 1.5;">Current Parsimony Models — Shows current parsimony models applied to characters in character list window.</span></li><li><span style="line-height: 1.5;">Current Parsimony Models — Supplies the currently assigned parsimony model (e.g., &quot;unordered&quot;) for a character.</span></li><li><span style="line-height: 1.5;">Current Probability Models — Supplies current stochastic models applied to characters for character list window.</span></li><li><span style="line-height: 1.5;">Current probability models — Supplies the currently assigned stochastic (likelihood, probability) model for a character.</span></li><li><span style="line-height: 1.5;">Current taxa partition — Supplies the current taxa partition.</span></li><li><span style="line-height: 1.5;">Current Tree — Supplies a single tree currently shown in a tree window.</span></li><li><span style="line-height: 1.5;">Current Weights — Supplies current weights applied to characters for character list window.</span></li><li><span style="line-height: 1.5;">Curvogram — Draws trees with curved branches (as PHYLIP's 'Curvogram').</span></li><li><span style="line-height: 1.5;">Cut Selected Taxa from Tree — Cuts the selected taxa from the tree.</span></li><li><span style="line-height: 1.5;">Data Painter — Supplies the paint can tool and dropper tool for filling cells in a character data editor.</span></li><li><span style="line-height: 1.5;">Data type for model — Indicates data type of model in list window.</span></li><li><span style="line-height: 1.5;">Data Window Coordinator — Coordinates the creation of basic data windows.</span></li><li><span style="line-height: 1.5;">Deassign Branch Lengths — Sets lengths of a tree's branches to unassigned.</span></li><li><span style="line-height: 1.5;">Deep Coalescences (gene tree) — Counts W. Maddison's (1997) number of extra gene lineages (&quot;deep coalescences&quot;) for gene tree within species tree</span></li><li><span style="line-height: 1.5;">Deep Coalescences (species tree) — Counts W. Maddison's (1997) number of extra gene lineages (&quot;deep coalescences&quot;) for species tree implied by a contained gene tree</span></li><li><span style="line-height: 1.5;">Deep Coalescences Multiple Loci — Counts W. Maddison's (1997) number of extra gene lineages (&quot;deep coalescences&quot;) for species tree containing a series of gene trees. The gene trees must be represented by a single taxa block.</span></li><li><span style="line-height: 1.5;">Default Order (characters) — Shows default order of character.</span></li><li><span style="line-height: 1.5;">Default Order (taxa) — Shows default order of taxa.</span></li><li><span style="line-height: 1.5;">Default String Matcher — Uses the default string-matching mechanism of the module that is doing the calling.</span></li><li><span style="line-height: 1.5;">Default Trees — Supplies simple default trees (bush, ladder).</span></li><li><span style="line-height: 1.5;">Defaults — Supervises some Mesquite-wide defaults</span></li><li><span style="line-height: 1.5;">Definite Tree Source — Supplies trees from from a tree source, arranged to be a definite number</span></li><li><span style="line-height: 1.5;">Delete Footnotes from Taxa — Deletes footnotes from taxa</span></li><li><span style="line-height: 1.5;">Delete Reticulations — Deletes secondary parents of nodes with more than one immediate ancestor. There is no control over which parents get deleted.</span></li><li><span style="line-height: 1.5;">Dependent Contained Tree Window — Displays a single tree (the same as contained within a tree in a tree window).</span></li><li><span style="line-height: 1.5;">Dependent Tree Window — Displays a single tree (the same as in a tree window).</span></li><li><span style="line-height: 1.5;">Detokenize Names — Detokenizes names. This can be useful for reading trees produced from exported files.</span></li><li><span style="line-height: 1.5;">Diagonal tree — Draws trees with standard diagonal branches (&quot;cladogram&quot;)</span></li><li><span style="line-height: 1.5;">Difference in steps in two characters — Calculates the difference between two characters in parsimony steps for a given tree.</span></li><li><span style="line-height: 1.5;">Difference in two values for character — Coordinates the calculation of the difference in two numbers for a character.</span></li><li><span style="line-height: 1.5;">Difference in two values for tree — Coordinates the calculation of the difference in two numbers for a tree.</span></li><li><span style="line-height: 1.5;">Discretize Selected Characters... — Bins the selected continuous characters to discretize them into categorical characters (ordered if more than two states), and adds them to an existing categorical matrix. If the continuous matrix is multi-item, only the first item is used.</span></li><li><span style="line-height: 1.5;">Display Character Loadings — Shows character loadings in ordination.</span></li><li><span style="line-height: 1.5;">Display Mean — Calculates and shows the mean value.</span></li><li><span style="line-height: 1.5;">Distance between taxa — Counts the difference between two taxa.</span></li><li><span style="line-height: 1.5;">Distance Matrix Package Introduction — Calculates and provides utilities for distances matrices.</span></li><li><span style="line-height: 1.5;">Distance of Contained Taxa — Distances among taxa according to distances among contained taxa (e.g., genes).</span></li><li><span style="line-height: 1.5;">Distances from Character Matrix — Distances calculated from a character matrix.</span></li><li><span style="line-height: 1.5;">Diverse Package — Diverse is a Mesquite package for examining speciation and extinction.</span></li><li><span style="line-height: 1.5;">Diversification (Char. Indep.) — Coordinates analyses of diversification (speciation/extinction).</span></li><li><span style="line-height: 1.5;">DManager Package Introduction — Provides special tools for managing data and taxa.</span></li><li><span style="line-height: 1.5;">DNA Empirical State Frequencies Model — Defines and maintains model of state frequencies that match those found in an empirical matrix.</span></li><li><span style="line-height: 1.5;">DrawHierarchy — Draws project and module trees</span></li><li><span style="line-height: 1.5;">Duplicate Selected Matrices — Duplicates selected matrices in List of Character Matrices window.</span></li><li><span style="line-height: 1.5;">Duplicate Taxa — Duplicates selected taxa and their character states.</span></li><li>Duplicate tree blocks — Duplicates selected tree blocks into a new tree block.</li><li><span style="line-height: 1.5;">E-mail Tree — Calls mailto: to email a text image of the tree.</span></li><li><span style="line-height: 1.5;">Edit State Names — Edits state names of a categorical data matrix.</span></li><li><span style="line-height: 1.5;">Enforce Minimum Node Age Constraints — Uses minimum node age constraints to set the branch lengths on the tree.</span></li><li><span style="line-height: 1.5;">Equiprobable Trees — Generates trees randomly so that each possible labelled tree topology is equally likely.</span></li><li><span style="line-height: 1.5;">Evolutionary Principal Components — Performs Maddison and Dyreson's evolutionary principal components analysis on a continous-valued matrix.</span></li><li><span style="line-height: 1.5;">Evolve Categorical Characters — Simulates character evolution for categorical characters on a tree.</span></li><li><span style="line-height: 1.5;">Evolve Continuous Characters — Simulates character evolution for continuous characters on a tree.</span></li><li><span style="line-height: 1.5;">Evolve DNA Characters — Simulates evolution of DNA sequences on a tree.</span></li><li><span style="line-height: 1.5;">Evolving Speciation Rate (Continuous Character) — Generates tree by a speciation model in which the speciation rate evolves by a Brownian motion model.</span></li><li><span style="line-height: 1.5;">Examples Navigator — Provides a Examples Navigator window with explanation and buttons to link to other files</span></li><li><span style="line-height: 1.5;">Export Ancestral States Trace — Exports the ancestral state trace from Trace Character History.</span></li><li><span style="line-height: 1.5;">Export for PartitionFinder — Exports a .phy file containing your data prepared for PartitionFinder and creates a .cfg file.</span></li><li><span style="line-height: 1.5;">Export for PartitionFinderProtein — Exports a .phy file containing your data prepared for PartitionFinderProtein and creates a .cfg file.</span></li><li><span style="line-height: 1.5;">Export Matrices &amp; Batch Files — Saves copies of matrices to separate files for subsequent batch analysis (e.g., parametric bootstrapping).</span></li><li><span style="line-height: 1.5;">Export NEXUS for MrBayes — Exports NEXUS files for use by MrBayes.</span></li><li><span style="line-height: 1.5;">Export NEXUS Tree File — Exports NEXUS file with a tree block, and optionally a taxa block.</span></li><li><span style="line-height: 1.5;">Export NEXUS Tree File from Tree Source — Exports NEXUS file with a tree block, and optionally a taxa block, based upon a source of trees. One major advantage of this is that it allows a collection of trees to a file without having them all in memory at once. For example, if the Tree Source is &quot;Transform Trees From Other Source&quot;, and the other source is &quot;Use Trees from Separate NEXUS file&quot;, then Mesquite will read in a tree from the other NEXUS file, transform the tree, and wrte it to the file, one at a tme.</span></li><li><span style="line-height: 1.5;">Export Partitions as Separate NEXUS files — Exports character partitions in a matrix as Separate NEXUS files.</span></li><li><span style="line-height: 1.5;">Export Taxa Distance Matrix — Exports a tab-delimited text file of distances among the taxa.</span></li><li><span style="line-height: 1.5;">Extract tree block — Extracts selected trees and places them in a new tree block.</span></li><li><span style="line-height: 1.5;">F81 distance — F81 (Felsenstein, 1981) distance from a character matrix.</span></li><li><span style="line-height: 1.5;">F84 distance — F84 (Felsenstein, 1984) distance from a DNA matrix.</span></li><li><span style="line-height: 1.5;">FASTA (DNA/RNA) — Imports and exports FASTA files that consist of DNA/RNA sequence data.</span></li><li><span style="line-height: 1.5;">FASTA (protein) — Imports and exports FASTA files that consist of amino acid sequence data.</span></li><li><span style="line-height: 1.5;">Fetch &amp; Add GenBank Sequences — Fetches GenBank nucleotide sequences given their GenBank accession numbers and adds them to the matrix.</span></li><li><span style="line-height: 1.5;">File Comment — Manages the main file comment in a NEXUS file.</span></li><li><span style="line-height: 1.5;">Fill — Fills cells with a uniform state in a character data editor.</span></li><li><span style="line-height: 1.5;">Filter Trees from Other Source — Filters trees from another source.</span></li><li><span style="line-height: 1.5;">Find Incomplete Codons — Finds the next occurrence of a partial codon triplet in a matrix of molecular data.</span></li><li><span style="line-height: 1.5;">Find Missing — Finds the next occurrence of missing data in a matrix of molecular data.</span></li><li><span style="line-height: 1.5;">Find Sequence — Finds the next occurrence of a sequence in a matrix of molecular data.</span></li><li><span style="line-height: 1.5;">Forward/Backward Rates — Uses maximum likelihood to estmate the rates of forward and backward changes (0 to 1 and 1 to 0 changes respectively), or alternatively the overall rate and the bias in gains versus losses, using the AsymmMk model on a tree for a given character.</span></li><li><span style="line-height: 1.5;">Fraction Applicable and Not Missing — Calculates the fraction of the matrix that is applicable and not missing data.</span></li><li><span style="line-height: 1.5;">Fraction of Clades that are Shared — Calculates the fraction of the total number of clades in two trees that are shared between the trees (excludes the clade consisting of all taxa).</span></li><li><span style="line-height: 1.5;">Frequency of State — Calculates the frequency of a state for a categorical character</span></li><li><span style="line-height: 1.5;">Fused Matrix Export (NEXUS) — Exports NEXUS files with matrices fused.</span></li><li><span style="line-height: 1.5;">Fused Matrix Export (Phylip/RAxML) — Exports Phylip matrices that consist of multiple matrices, preparing them for RAxML.</span></li><li><span style="line-height: 1.5;">Gamma Invar Rates Model — Defines and maintains Gamma rate variation models with a proportion of invariant characters.</span></li><li><span style="line-height: 1.5;">Gamma Rates Model — Defines and maintains Gamma rate variation models.</span></li><li><span style="line-height: 1.5;">Gaps to Missing — Converts gaps to missing data in selected region.</span></li><li><span style="line-height: 1.5;">GC — Returns 1 for G or C, 0 for A or T.</span></li><li><span style="line-height: 1.5;">GenBank (DNA/RNA) — Imports GenBank files that consist of DNA/RNA sequence data.</span></li><li><span style="line-height: 1.5;">GenBank Number — Lists the primers used in the reads composing a sequence.</span></li><li><span style="line-height: 1.5;">GenBank Taxon — Provides a URL to a taxon in NCBI GenBank</span></li><li><span style="line-height: 1.5;">GenBank/GenPept (Protein) — Imports GenBank/GenPept files that consist of amino acid sequence data.</span></li><li><span style="line-height: 1.5;">Gene Duplication-Extinction (gene tree) — Counts the number of duplication and extinction events for gene tree implied by a containing species tree</span></li><li><span style="line-height: 1.5;">Gene Duplication-Extinction (species tree) — Counts the number of duplication and extinction events for species tree implied by a contained gene tree</span></li><li><span style="line-height: 1.5;">Genesis Introduction — Implements stochastic models of molecular evolution.</span></li><li><span style="line-height: 1.5;">Get Taxon's tree from a web site — Supplies a tool for tree windows that gets tree for taxon touched from a web site.</span></li><li><span style="line-height: 1.5;">Get Taxon's tree from tolweb.org — Supplies a tool for tree windows that gets tree for taxon touched from tolweb.org</span></li><li><span style="line-height: 1.5;">Go-to Window — Provides a window to jump to another file</span></li><li><span style="line-height: 1.5;">Graft Other Tree — Grafts a tree in a tree window onto given tree; requires that the other tree includes none of the same terminal taxa. If a taxon in receiving tree is selected, graft occurs there; otherwise, graft occurs at root.</span></li><li><span style="line-height: 1.5;">Group Membership (characters) — Lists and allows changes to group membership in the current partition of characters, for List of Characters window.</span></li><li><span style="line-height: 1.5;">Group Membership (taxa) — Lists and allows changes to group membership in the current partition of taxa, for List of Taxa window.</span></li><li><span style="line-height: 1.5;">GTR Rate Matrix Model — Defines and maintains six-parameter general time-reversible (GTR) rate matrix models.</span></li><li>Has Data in Matrix — Indicates whether taxon has non-missing non-gap data in a given matrix.</li><li><span style="line-height: 1.5;">Highlight Apparently Slightly Misaligned — Colors aligned sequences to emphasize sections that seem slightly misaligned.</span></li><li><span style="line-height: 1.5;">Highlight Gaps in Coding Regions — Highlights gaps if site has a codon position, or if both sites to either side have codon positions, and if sites to both sides have states or missing.</span></li><li><span style="line-height: 1.5;">Import and Export Package Introduction — Provides file import/export utilities.</span></li><li><span style="line-height: 1.5;">Import Taxon Names — Reads a tab-delimited text files of voucher codes (column heading &quot;ID&quot;) and taxon names (column heading &quot;name&quot;), matching taxa by voucher code and renaming taxa accordingly</span></li><li><span style="line-height: 1.5;">Information Attached To Matrix — Colors taxon names by whether or not there is information attached to this matrix.</span></li><li><span style="line-height: 1.5;">Initialize DNA Submodels — Initializes default DNA submodels.</span></li><li><span style="line-height: 1.5;">Initialize Genetic Code — Initializes default genetic codes.</span></li><li><span style="line-height: 1.5;">Initialize Parsimony — Initializes default character models for parsimony.</span></li><li><span style="line-height: 1.5;">Initialize predefined categorical parsimony models — Initializes the predefined categorical parsimony models.</span></li><li><span style="line-height: 1.5;">Initialize predefined continuous parsimony models — Initializes the predefined continuous parsimony models.</span></li><li><span style="line-height: 1.5;">Initialize predefined meristic parsimony models — Initializes the predefined meristic parsimony models.</span></li><li><span style="line-height: 1.5;">Insert Node — Provides a tool with which to insert nodes along a branch.</span></li><li><span style="line-height: 1.5;">Installer — Installs packages.</span></li><li><span style="line-height: 1.5;">Interpolate Character Selection — Selects characters that are surrounded by characters selected.</span></li><li><span style="line-height: 1.5;">Inverse of Value — Alters continuous data by inverting values (new value = 1/(original value)).</span></li><li><span style="line-height: 1.5;">iSpecies — Provides a URL to a taxon in iSpecies</span></li><li><span style="line-height: 1.5;">Item values chart — Manages chart of values for items.</span></li><li><span style="line-height: 1.5;">Items bi-plot — Makes a chart comparing two values for each of many items.</span></li><li><span style="line-height: 1.5;">iText — iText is a package for saving PDF files within Mesquite.</span></li><li><span style="line-height: 1.5;">JAMA Introduction — Introduces the JAMA library.</span></li><li><span style="line-height: 1.5;">Jukes Cantor distance — Jukes Cantor distance from a DNA matrix.</span></li><li><span style="line-height: 1.5;">K2P distance — K2P (Kimura 2-parameter) distance from a character matrix.</span></li><li><span style="line-height: 1.5;">Keep Prefix/Suffix of Taxon Name — Keeps only Prefix or Suffix of Taxon Name.</span></li><li><span style="line-height: 1.5;">Kyte &amp; Doolittle Hydrophobicity — Returns hydrophobicity of amino acid using the scale of Kyte, J &amp; R.F. Doolittle (1982) J. Mol. Biol. 157:105-142. Numbers from <a  href="http://www.whatislife.com/reader/protein/aa.html" rel="nofollow">http://www.whatislife.com/reader/protein/aa.html</a></span></li><li><span style="line-height: 1.5;">Label Branch Lengths — An assistant to a tree display that labels branches to show their lengths.</span></li><li><span style="line-height: 1.5;">Label states — A module that displays character states on tree using labels. This is a display-only module, and would be hired by another module that organizes assigning states to the nodes.</span></li><li><span style="line-height: 1.5;">Landmark Drawings — Places drawing of landmarks at each node in tree.</span></li><li><span style="line-height: 1.5;">Likelihood Ancestral States — Coordinates reconstruction of ancestral states by maximum likelihood. Currently each node is estimated independently (i.e., corresponding to PAUP's marginal reconstruction).</span></li><li><span style="line-height: 1.5;">Likelihood in Character — Calculates the negative log likelihood of a tree with respect to a single character.</span></li><li><span style="line-height: 1.5;">Likelihood Surface AsymmMk Model — Shows the likelihood surface for the AsymmMk Model.</span></li><li><span style="line-height: 1.5;">Lineages Through Time — Displays a window showing plot of lineages through time averaged over many trees.</span></li><li><span style="line-height: 1.5;">List of Character Group Labels — Makes windows listing character groups and information about them.</span></li><li><span style="line-height: 1.5;">List of Character Inclusion Sets — Makes windows listing character sets.</span></li><li><span style="line-height: 1.5;">List of Character Partitions — Makes windows listing character partitions.</span></li><li><span style="line-height: 1.5;">List of Character Sets — Makes windows listing character sets.</span></li><li><span style="line-height: 1.5;">List of Character Weight Sets — Makes windows listing character weight sets.</span></li><li><span style="line-height: 1.5;">List of Codon Positions Sets — Makes windows listing codon positions sets.</span></li><li><span style="line-height: 1.5;">List of Genetic Code Sets — Makes windows listing genetic code sets.</span></li><li><span style="line-height: 1.5;">List of Parsimony Model Sets — Makes a window listing parsimony model sets.</span></li><li><span style="line-height: 1.5;">List of Probability Model Sets — Makes windows listing probability model sets.</span></li><li><span style="line-height: 1.5;">List of Taxa Partitions — Makes windows listing taxa partitions.</span></li><li><span style="line-height: 1.5;">List of Taxon Group Labels — Makes windows listing taxon groups and information about them.</span></li><li><span style="line-height: 1.5;">List of Taxon Sets — Makes windows listing taxon sets.</span></li><li><span style="line-height: 1.5;">List Windows Package Introduction — Provides List windows to manage taxa, characters and other information.</span></li><li><span style="line-height: 1.5;">Log transform — Alters continuous data by log transforming values.</span></li><li><span style="line-height: 1.5;">Lower Case — Returns 1 if state symbol is a lower case letter.</span></li><li><span style="line-height: 1.5;">MAFFT Align — Sends the selected sequences to MAFFT to align.</span></li><li><span style="line-height: 1.5;">Maintain Clipboard Match — Finds the first occurrence of the sequence in the clipboard, within a designated taxon, and maintains that match as the clipboard changes.</span></li><li><span style="line-height: 1.5;">Maintain Target Match — Finds the first occurrence of the sequence in the text area below the matrix, within a designated taxon, and maintains that match as the text area changes.</span></li><li><span style="line-height: 1.5;">Majority-Rule Consensus — Calculates the majority-rule consensus tree.</span></li><li><span style="line-height: 1.5;">Make Rerootings of Clade — Presents a tool by which you can touch on a tree; a tree block is made consisting of trees representing all rerootings of the clade of the node touched.</span></li><li><span style="line-height: 1.5;">Manage Alternative Taxon Names — Manages (including NEXUS read/write) alternative taxon names.</span></li><li><span style="line-height: 1.5;">Manage ASSUMPTIONS blocks — Manages ASSUMPTIONS block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage Attached Notes — Manages (including NEXUS read/write) notes attached to taxa, characters and cells of matrices.</span></li><li><span style="line-height: 1.5;">Manage AUTHORS blocks — Manages AUTHORS block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage categorical character matrices — Manages categorical data matrices (including read/write in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage character inclusion sets — Manages (including NEXUS read/write) character inclusion sets (EXSETS).</span></li><li><span style="line-height: 1.5;">Manage Character Models — Manages character models.</span></li><li><span style="line-height: 1.5;">Manage character partititions — Manages (including NEXUS read/write) character partitions.</span></li><li><span style="line-height: 1.5;">Manage character sets — Manages (including NEXUS read/write) character sets (CHARSETS).</span></li><li><span style="line-height: 1.5;">Manage character weight sets — Manages (including NEXUS read/write) character weight sets.</span></li><li><span style="line-height: 1.5;">Manage codon positions — Manages (including NEXUS read/write) codon position sets.</span></li><li><span style="line-height: 1.5;">Manage CODONS blocks — Manages CODONS block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage Continuous char. matrices — Manages continuous data matrices (including read/write in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage DNA/RNA matrices — Manages DNA/RNA data matrices (including read/write in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage Foreign Blocks — Manages unrecognized blocks in a NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage genetic code sets — Manages (including NEXUS read/write) genetic code sets (CODESETs).</span></li><li><span style="line-height: 1.5;">Manage Geographic character matrices — Manages data matrices of geographic data (including read/write in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage hyperlinks — Manages (including NEXUS read/write) hyperlinks.</span></li><li><span style="line-height: 1.5;">Manage LABELS blocks — Manages LABELS block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage Meristic char. matrices — Manages meristic data matrices (including read/write in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage MESQUITE block — Manages MESQUITE block (including compose snapshot of current state of file in MESQUITE block &quot;Auto&quot; in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage MESQUITECHARMODELS blocks — Manages MESQUITECHARMODELS block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage NOTES blocks — Manages footnotes and pictures attached to taxa, characters and data points (including read/write NOTES block in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage parsimony model sets — Manages (including NEXUS read/write) parsimony model sets (TYPESETs).</span></li><li><span style="line-height: 1.5;">Manage pictures — Manages (including NEXUS read/write) pictures.</span></li><li><span style="line-height: 1.5;">Manage probability model sets — Manages (including NEXUS read/write) probability model sets (PROBMODELSETs).</span></li><li><span style="line-height: 1.5;">Manage Protein matrices — Manages Protein data matrices (including read/write in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage SETS blocks — Manages character sets and other sets of various kinds (including read/write SETS block in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage stored character orders — Manages (including NEXUS read/write) character orders.</span></li><li><span style="line-height: 1.5;">Manage TAXA blocks — Manages sets of taxa (including read/write TAXA block in NEXUS file).</span></li><li><span style="line-height: 1.5;">Manage taxa partititions — Manages (including NEXUS read/write) taxa partitions.</span></li><li><span style="line-height: 1.5;">Manage TaxaAssociation blocks — Manages TaxaAssociation blocks in NEXUS file.</span></li><li><span style="line-height: 1.5;">Manage taxon sets — Manages (including NEXUS read/write) taxon sets.</span></li><li><span style="line-height: 1.5;">Manage TREES blocks — Manages tree blocks (including read/write TREES block in NEXUS file).</span></li><li><span style="line-height: 1.5;">Map Continuous — Supplies a reconstruction of continuous-valued ancestral states on a tree. Differs from the basic reconstruction methods in that it supplies simply numbers for nodes, instead of a CharacterHistory</span></li><li><span style="line-height: 1.5;">Mark Taxa — Permits user to mark taxa in the List of Taxa with a specific name.</span></li><li><span style="line-height: 1.5;">Matching Sequence — Finds the next occurrence of a sequence in a matrix of molecular data. Allows a certain number of mismatches.</span></li><li><span style="line-height: 1.5;">Matrices from Ordinations — Supplies character matrices from ordinations of existing matrices.</span></li><li><span style="line-height: 1.5;">Matrix from Clipboard — Makes a character matrix from text in the clipboard.</span></li><li><span style="line-height: 1.5;">Matrix Info — Shows Information about the data in each taxon.</span></li><li><span style="line-height: 1.5;">Matrix Source (indep.) — Coordinates the supply of matrices to other modules. It acts independently, in that it decides what matrix to supply (e.g., by presenting an interface to choose).</span></li><li><span style="line-height: 1.5;">Matrix Source (obed.) — Coordinates the supply of matrices to other modules. It acts obediently, in that it does not present an interface to choose which matrix, but relies on employer to decide.</span></li><li><span style="line-height: 1.5;">Matrix value with current tree — Coordinates the calculation of a number for a character matrix based on a current tree.</span></li><li><span style="line-height: 1.5;">Maximum likelihood reconstruct (Generic categorical) — Assesses likelihood for categorical characters, and r econstructs ancestral states by the maximum marginal probability (&quot;MLE&quot;) criterion</span></li><li><span style="line-height: 1.5;">Mean value of character — Calculates the mean observed trait for continuous character.</span></li><li><span style="line-height: 1.5;">Mean value of character (linked matrix) — Calculates the mean observed trait for continuous character in a linked matrix.</span></li><li><span style="line-height: 1.5;">Merge Matrices from File — Includes a file and concatenates its taxa to an existing taxa block, for instance to add seqences.</span></li><li><span style="line-height: 1.5;">Merge Taxa — Merges selected taxa and their character states.</span></li><li><span style="line-height: 1.5;">Merge Taxa &amp; Matrices from File — Includes a file and concatenates its taxa to an existing taxa block, for instance to add seqences.</span></li><li><span style="line-height: 1.5;">Meristic Data Package Introduction — Provides basic facilities for handling meristic characters.</span></li><li><span style="line-height: 1.5;">Meristic Items Editor — Edits items for meristic data matrix within the editor info panel.</span></li><li><span style="line-height: 1.5;">Mesquite — This is the central module for the Mesquite system. In it is the main method that starts up the Mesquite application. This moduleloads information for all of the other modules, and hires FileCoordinator modules as needed to deal with open files. Thus, in the tree of employees that active modules make, this module is at its root.</span></li><li><span style="line-height: 1.5;">Mesquite Heuristic (Add &amp; rearrange) — Searches for optimal trees by adding taxa then rearranging the tree.</span></li><li><span style="line-height: 1.5;">mesquite.jsci Package Introduction — Introduces the JSci library.</span></li><li><span style="line-height: 1.5;">Mirror Tree Window — Displays a single tree (the same as in a tree window) twice, in mirror image.</span></li><li><span style="line-height: 1.5;">Missing to Gaps — Converts missing data to gaps in selected region.</span></li><li><span style="line-height: 1.5;">Mk1 Estimated Rate — Estimates the rate of a character's evolution under the simple Mk1 model.</span></li><li><span style="line-height: 1.5;">Mk1 Model (Markov 1 parameter) — Defines and maintains simple Markov k-state 1-parameter stochastic models (Lewis, 2001) of character evolution.</span></li><li><span style="line-height: 1.5;">Model Paradigm — Indicates paradigm of model in list window.</span></li><li><span style="line-height: 1.5;">Modify Current Tree — Modifies current tree.</span></li><li><span style="line-height: 1.5;">Molecular Utilities Package Introduction — Provides utilities for molecular data.</span></li><li><span style="line-height: 1.5;">Most Pairs — Chooses taxon pairings regardless of contrast in a character, so as to maximize the number of pairs that are phylogenetically independent.</span></li><li><span style="line-height: 1.5;">Move block — Allows one to manually blocks in a sequence, and split the blocks.</span></li><li><span style="line-height: 1.5;">Move Prefix/Suffix of Taxon Name — Moves the suffix of the taxon name to be to prefix or the prefix to be the suffix</span></li><li><span style="line-height: 1.5;">MrBayes Max. A-Post. Tree — Supplies tree from MrBayes with highest posterior probability (MAP tree).</span></li><li><span style="line-height: 1.5;">MrBayes Package Introduction — Provides links to MrBayes analyses.</span></li><li><span style="line-height: 1.5;">MrBayes Score — Supplies posterior probability score from MrBayes</span></li><li><span style="line-height: 1.5;">MrBayes Trees from .t File — Supplies trees from a &quot;.t&quot; file produced by MrBayes</span></li><li><span style="line-height: 1.5;">MRP Matrices from Trees — Supplies matrices which represent trees for MRP (Matrix Representation with Parsimony) supertree analyses.</span></li><li><span style="line-height: 1.5;">Multi Tree Window — Displays a special tree window with many trees simultaneously.</span></li><li><span style="line-height: 1.5;">Multiply by character — Alters continuous data by multiplying by a continuous character. All items of the matrix are similarly modified.</span></li><li><span style="line-height: 1.5;">Multiply by value — Alters continuous data by multiplying by a value. All items of the matrix are similarly modified.</span></li><li><span style="line-height: 1.5;">Multistate sites in taxon — Reports the number of polymorphic/partially uncertain sites in a taxon for a data matrix.</span></li><li><span style="line-height: 1.5;">Muscle Align — Sends the selected sequence to Muscle to align.</span></li><li><span style="line-height: 1.5;">NBRF/PIR (DNA/RNA) — Imports and exports NBRF files that consist of DNA/RNA sequence data.</span></li><li><span style="line-height: 1.5;">NBRF/PIR (protein) — Imports and exports NBRF files that consist of amino acid sequence data.</span></li><li><span style="line-height: 1.5;">Neutral Coalescence — Performs neutral coalescence within a population by a simple model of a neutral gene with constant population size.Remembers a tree of nodes, with branch lengths assigned according to generation of coalescence. The default population size is 10000.</span></li><li><span style="line-height: 1.5;">NEXUS Blocks List — Makes windows listing NEXUS blocks.</span></li><li><span style="line-height: 1.5;">NEXUS Defaults — Sets whether TITLE and LINK commands are to be suppressed where possible in saving NEXUS files.</span></li><li><span style="line-height: 1.5;">NEXUS file interpreter — Coordinates the reading and writing of NEXUS files.</span></li><li><span style="line-height: 1.5;">NNI Rearranger — Rearranges a tree by nearest neighbor interchange (NNI). Does not handle trees with polytomies.</span></li><li><span style="line-height: 1.5;">No Color — Turns off cell coloring.</span></li><li><span style="line-height: 1.5;">Node Age Constraints — Supplies a tool for tree windows to set node age constraints. A short text string summarizes constraints. E.g. to say minimum age is 2.0, enter &quot;2.0+&quot;. To say range is 2.0 to 4.0, enter &quot;2.0-4.0&quot;. To say maximum is 4.0, enter &quot;0.0-4.0&quot;. To say age is exactly 3.0, enter &quot;3.0&quot;</span></li><li><span style="line-height: 1.5;">Node Count (Speciation Patristic) Distances implied by Tree — Distances among taxa implied by counting nodes along branches of tree from one taxon to another (Speciation Patristic distances).</span></li><li><span style="line-height: 1.5;">Node Depth — Calculates the depth of each node from the highest terminal in its clade.</span></li><li><span style="line-height: 1.5;">Node Locations (2D plot) — Calculates the node locations for a tree plotted in a two dimensional space.</span></li><li><span style="line-height: 1.5;">Node Locations (3D plot) — Calculates the node locations for a tree plotted in a three dimensional space.</span></li><li><span style="line-height: 1.5;">Node Locations (circle) — Calculates the node locations for a tree drawn in circular fashion, with root at center.</span></li><li><span style="line-height: 1.5;">Node Locations (oval) — Calculates the node locations for a tree drawn in oval fashion, with root at center.</span></li><li><span style="line-height: 1.5;">Node Locations (standard) — Calculates the node locations in a tree drawing, for use with vertical or horizontal tree drawers (e.g., the standard diagnonal or square trees).</span></li><li><span style="line-height: 1.5;">Node Namer — Provides a tool to name the nodes of a tree</span></li><li><span style="line-height: 1.5;">Node Position Adjust — Provides a tool to adjust the depth of a node in the tree</span></li><li><span style="line-height: 1.5;">Node-Associated Text — Shows text attached to nodes on the tree.</span></li><li><span style="line-height: 1.5;">Node-Associated Values — Shows values attached to nodes on the tree.</span></li><li><span style="line-height: 1.5;">Nodes Scattergram — Shows values for nodes of tree via a scattergram.</span></li><li><span style="line-height: 1.5;">NONA, Hennig86, PiWe, WinClada — Imports and exports NONA/Hennig86/PiWe/WinClada files.</span></li><li><span style="line-height: 1.5;">Nucleotide Complement — Alters nucleotide data to its complement.</span></li><li><span style="line-height: 1.5;">Number for Character (in List of Characters window) — Supplies numbers for characters for a character list window.</span></li><li><span style="line-height: 1.5;">Number for Matrix (in List of Character Matrices window) — Supplies numbers for character matrcies for a character matrices list window.</span></li><li><span style="line-height: 1.5;">Number for Nodes using Character — Supplies numbers, based on a character, for each node of a tree.</span></li><li><span style="line-height: 1.5;">Number for Nodes using Matrix — Supplies numbers, based on a matrix, for each node of a tree.</span></li><li><span style="line-height: 1.5;">Number for Taxon (in List of Taxa window) — Supplies numbers for taxa for a taxa list window.</span></li><li><span style="line-height: 1.5;">Number for Tree (in List of Trees window) — Supplies numbers for trees for a trees list window.</span></li><li><span style="line-height: 1.5;">Number for Tree Block (in List of Tree Blocks window) — Supplies numbers for tree blocks for a tree blocks list window.</span></li><li><span style="line-height: 1.5;">Number of characters in character set — Indicates number of characters in character set in list window.</span></li><li><span style="line-height: 1.5;">Number of characters in inclusion set — Indicates number of characters in inclusion set in list window.</span></li><li><span style="line-height: 1.5;">Number of Characters Included — Calculates the number of characters in a matrix that are included.</span></li><li><span style="line-height: 1.5;">Number of characters of data matrix — Indicates taxa of data matrix.</span></li><li><span style="line-height: 1.5;">Number of Congruent Trees — For the given tree, returns the number of trees in the tree source that are congruent with it.</span></li><li><span style="line-height: 1.5;">Number of groups in taxa partition — Indicates number of groups in taxa partition in list window.</span></li><li><span style="line-height: 1.5;">Number of Stops — Reports the number of stops coded by nucleotides in a taxon for a data matrix.</span></li><li><span style="line-height: 1.5;">Number of Taxa — Counts the number of taxa in a tree.</span></li><li><span style="line-height: 1.5;">Number of taxa in set — Indicates number of taxa in list window.</span></li><li><span style="line-height: 1.5;">Number of taxa in taxa set — Indicates number of taxa in taxa set in list window.</span></li><li><span style="line-height: 1.5;">Number of Taxa with Non-Gap, Non-Missing — Calculates the number of taxa in the character that contain applicable, non-missing data.</span></li><li><span style="line-height: 1.5;">Number of trees in tree block — Indicates number of trees.</span></li><li><span style="line-height: 1.5;">Number of Trees Matching Criterion — Counts the number of trees in a tree block that match a user-specified criterion. For example, it could count the number of trees in a tree block which are congruent with a particular tree topology (by selecting the &quot;Tree Congruent with Constraint Tree Topology&quot; module in the Criterion for Trees dialog).</span></li><li><span style="line-height: 1.5;">Number of Unambiguous Sites in Sequence — Reports the number of unambiguous sites.</span></li><li><span style="line-height: 1.5;">Numbers for Nodes from Character Reconstructions — Supplies numbers, based on a character reconstruction, for each node of a tree.</span></li><li><span style="line-height: 1.5;">Obtain Branch Lengths from Numbers For Nodes — Assigns a value of branch length for all of a tree's branches based upon the numbers calculated by a Numbers for Nodes.</span></li><li><span style="line-height: 1.5;">Obtain Branch Lengths from Values Attached to Nodes — Assigns branch lengths to the tree taken from values already attached to nodes, for instance support values or divergence times from other programs.</span></li><li><span style="line-height: 1.5;">Occasionally Randomly Modify — With specified probabililty, will ask random tree modifier to modify current tree.</span></li><li><span style="line-height: 1.5;">Open URL — Opens a file on the web as if it were a local data file</span></li><li><span style="line-height: 1.5;">Original File (import source) — Shows the orignal file from which a fuse/imported taxon came.</span></li><li><span style="line-height: 1.5;">Ornamental Trees Package Introduction — Has extra methods for drawing and decorating trees.</span></li><li><span style="line-height: 1.5;">Other Tree Blocks — Supplies blocks of trees from various sources</span></li><li><span style="line-height: 1.5;">Pagel format (ppy) file for Discrete — Imports and exports files formatted for Pagel's Discrete program.</span></li><li><span style="line-height: 1.5;">Pagel format (ppy) file for Multistate — Imports and exports files formatted for Pagel's Multistate program.</span></li><li><span style="line-height: 1.5;">Pagel's 1994 test of correlated (discrete) character evolution — A statistical test, described in Pagel(1994), for nonindependent evolution of two discrete, binary characters</span></li><li><span style="line-height: 1.5;">Pairs Contrasting in State of One Character — Chooses taxon pairings so as to maximize the number of pairs that are phylogenetically independent, subject to the constraint that each pair shows a contrast the states of a binary character.</span></li><li><span style="line-height: 1.5;">Pairs Contrasting in State of Two Characters — Chooses taxon pairings so as to maximize the number of pairs that are phylogenetically independent, subject to the constraint that each pair shows a contrast the states of both of two binary characters.</span></li><li><span style="line-height: 1.5;">Pairwise Comparison — Performs pairwise comparison character correlation tests. Phylogenetically independent pairs are chosen, and the states of two binary characters are compared to see if they are correlated among these pairs.</span></li><li><span style="line-height: 1.5;">Pairwise Comparisons Package Introduction — Performs pairwise comparisons character correlation analyses.</span></li><li><span style="line-height: 1.5;">Pairwise Taxon Difference — Calculates distance between a specific pair of taxa based on a character matrix. Differs from Distance Between Taxa in thatit performs calculation for only a single pair of taxa.</span></li><li><span style="line-height: 1.5;">PAL Introduction — Introduces the PAL library.</span></li><li><span style="line-height: 1.5;">Parameters Explorer — Provides a window to show values returned when parameter settings are varied</span></li><li><span style="line-height: 1.5;">Parsimony Ancestral States — Coordinates the parsimony reconstruction of ancestral states.</span></li><li><span style="line-height: 1.5;">Parsimony Character Steps — Calculates the number of parsimony steps in a character.</span></li><li><span style="line-height: 1.5;">Parsimony Linear — Reconstructs the ancestral states of continuous characters so as to minimize the sum of absolute values of changes (linear, Wagner, Farris or Manhattan parsimony). If the continuous character has multiple items, then length is reported on only the first item.</span></li><li><span style="line-height: 1.5;">Parsimony Meristic Linear — Reconstructs the ancestral states of mersitic characters so as to minimize the sum of absolute values of changes (linear, Wagner, Farris or Manhattan parsimony). If the meristic character has multiple items, then length is reported on only the first item.</span></li><li><span style="line-height: 1.5;">Parsimony Ordered — Reconstructs the ancestral states of categorical characters using parsimony, under the assumption that states are ordered (ordered, Farris or Wagner parsimony; additive). Also counts parsimony steps.</span></li><li><span style="line-height: 1.5;">Parsimony Package Introduction — Preforms parsimony calculations to count steps and treelength of characters on trees.</span></li><li><span style="line-height: 1.5;">Parsimony Squared-change — Reconstructs the ancestral states of continuous characters so as to minimize the sum of squared changes (squared-change or least squares parsimony; Brownian motion likelihood).</span></li><li><span style="line-height: 1.5;">Parsimony Stepmatrix — Reconstructs ancestral states of categorical characters using a stepmatrix (cost matrix).</span></li><li><span style="line-height: 1.5;">Parsimony Unordered — Reconstructs the ancestral states of categorical characters using parsimony, under the assumption that states are unordered (unordered or Fitch parsimony; nonadditive). Also counts parsimony steps.</span></li><li><span style="line-height: 1.5;">Patristic distance correlation — Calculates the product moment correlation coefficient among the off-diagonal elements of the patristic distances matrices of two trees. Ignores unassigned and negative values in the matrices. If employed as a distance, converts scores from 1 to -1 to scores from 0 to 2</span></li><li><span style="line-height: 1.5;">Patristic Distances implied by Tree — Distances among taxa implied by path-length distance along branches of tree from one taxon to another (Patristic distances). Unassigned branches are treated as of length 1.0.</span></li><li><span style="line-height: 1.5;">Phylip (categorical data) — Imports and exports Phylip matrices that consist of basic categorical data with just two states. Exported data will consist of default symbols (0 and 1).</span></li><li><span style="line-height: 1.5;">Phylip (DNA/RNA) — Imports and exports Phylip matrices that consist of DNA/RNA sequence data.</span></li><li><span style="line-height: 1.5;">Phylip (protein) — Imports and exports Phylip matrices that consist of amino acid sequence data.</span></li><li><span style="line-height: 1.5;">Phylip (trees) — Imports and exports Phylip trees.</span></li><li><span style="line-height: 1.5;">Picture Window — Displays a picture in a window.</span></li><li><span style="line-height: 1.5;">Plot Tree — Draws trees plotted in a two dimensional space.</span></li><li><span style="line-height: 1.5;">Plot Tree 3D — Draws trees plotted in a three dimensional space.</span></li><li><span style="line-height: 1.5;">Polytomies Present — Indicates whether polytomies are present.</span></li><li><span style="line-height: 1.5;">Polytomy Assumption — Indicates whether the tree assumes that any polytomies, if present, are hard or soft. If not, the default assumption is used.</span></li><li><span style="line-height: 1.5;">POY (DNA/RNA) — Exports POY files of DNA/RNA sequence data.</span></li><li><span style="line-height: 1.5;">Prefix with Taxon Group Name — Prefixes the taxon name with the taxon group name.</span></li><li><span style="line-height: 1.5;">Principal Components Analysis — Performs principal components analysis on a continous-valued matrix.</span></li><li><span style="line-height: 1.5;">Projector — Supplies a projector tool for tree windows that allows pictures attached to taxa to be displayed.</span></li><li><span style="line-height: 1.5;">Proportion CG in taxon — Reports the proportion of CG in a taxon for a DNA matrix.</span></li><li><span style="line-height: 1.5;">Proportion Gaps In Character — Calculates the proportion of gaps (inapplicable codings) in a character across taxa. Does not include missing (unassigned) data.</span></li><li><span style="line-height: 1.5;">Proportion Gaps in Taxon — Reports the proportion of gaps (inapplicable codings) in a taxon for a data matrix.</span></li><li><span style="line-height: 1.5;">Proportion Internal Gaps — Calculates the proportion of internal gaps (inapplicable codings) in a character across taxa. Does not include missing (unassigned) data.</span></li><li><span style="line-height: 1.5;">Proportion Invariant Model — Defines and maintains models that specify that a certain proportion of characters are invariant.</span></li><li><span style="line-height: 1.5;">Proportion lower case codings in taxon — Reports the percentage of lower case codings in a taxon for a data matrix.</span></li><li><span style="line-height: 1.5;">Proportion Missing — Calculates the proportion of missing data in a character across taxa.</span></li><li><span style="line-height: 1.5;">Proportion missing data in taxon — Reports the proportion of missing data in a taxon for a data matrix.</span></li><li><span style="line-height: 1.5;">Proportion Polymorphic — Calculates the proportion of taxa coded as polymorphic or partially uncertain in a character.</span></li><li><span style="line-height: 1.5;">Proportion Terminal Gaps — Calculates the proportion of Terminal gaps (inapplicable codings) in a character across taxa. Does not include missing (unassigned) data.</span></li><li><span style="line-height: 1.5;">Protein Site Property — Calculates the mean value of amino acid properties at a site across taxa.</span></li><li><span style="line-height: 1.5;">Quick Key Entry — Provides a tool with which to quickly enter data. If this tool is active, then typing a key will cause that value to be entered into all selected cells.</span></li><li><span style="line-height: 1.5;">Random Branch Moves — Rearranges tree by random branch moves.</span></li><li><span style="line-height: 1.5;">Random Fill (Categorical) — Fills cells with a randomly-chosen state. For DNA data, states A, C, G, and T are chosen with equal probability; for other data, states up to and including the maximum state value are chosen with equal probability.</span></li><li><span style="line-height: 1.5;">Random Fill (Gaussian/Normal) — Fills cells with a random state, using a Normal distribution.</span></li><li><span style="line-height: 1.5;">Random Fill (Meristic Uniform) — Fills cells with a random state, uniformly.</span></li><li><span style="line-height: 1.5;">Random Fill (Uniform) — Fills cells with a random state, uniformly.</span></li><li><span style="line-height: 1.5;">Random n characters — Selects n characters randomly.</span></li><li><span style="line-height: 1.5;">Randomize Taxon Order — Randomizes the order of taxa.</span></li><li><span style="line-height: 1.5;">Randomly Modify Current Tree — Modifies current tree by random changes.</span></li><li><span style="line-height: 1.5;">Randomly Modify Matrix — Supplies character matrices that are randomly modified from an existing matrix.</span></li><li><span style="line-height: 1.5;">Randomly Modify Respective Matrices — Supplies character matrices that are randomly modified from a series of existing matrices. The i'th modified matrix is derived from the i'th original matrix.</span></li><li><span style="line-height: 1.5;">Randomly Modify Trees — Modifies each of a series of trees by random changes; the i'th tree from this module comes by modifying the i'th tree from the original source of trees.</span></li><li><span style="line-height: 1.5;">Randomly Resolve Polytomies — Randomly resolves polytomies in tree. All possible resolutions are equiprobable. Thus, if the tree is a polytomous bush, the resulting resolved trees will be distributed equivalently to that from the Equiprobable Trees module.</span></li><li><span style="line-height: 1.5;">Rarefy Characters — Deletes characters randomly to rarefy matrix.</span></li><li><span style="line-height: 1.5;">Rarefy Tree — Rarefies tree by randomly excluding taxa. If some taxa are selected, random exclusion is limited to the selected taxa.</span></li><li><span style="line-height: 1.5;">Raw Nexus block editor — Edits blocks in a NEXUS file.</span></li><li><span style="line-height: 1.5;">Read DATA blocks — Coordinates the reading of a DATA block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Read DISTRIBUTION blocks — Coordinates the reading of a DISTRIBUTION block in NEXUS file.</span></li><li><span style="line-height: 1.5;">Rearranged tree — Supplies trees that are rearrangments of a given tree. The original tree is NOT included among the rearrangements.</span></li><li><span style="line-height: 1.5;">Recode Characters — Recodes categorical data (e.g., state 1 to state 0).</span></li><li><span style="line-height: 1.5;">Reconstruct Ancestral States — Reconstructs ancestral states on the nodes of a tree.</span></li><li><span style="line-height: 1.5;">Reconstruct Deep Coalescence — Reconstructs a contained tree within a containing tree so as to minimize the amount of deep coalescence (failure of lineage sorting). The contained tree is assumed to be rooted.</span></li><li><span style="line-height: 1.5;">Reinterpret Internal Node Labels... — Reinterprets labels from all internal nodes as numbers or text attached to nodes or branches.</span></li><li><span style="line-height: 1.5;">Remove from Taxon Names — Removes a specified number of characters from taxon names.</span></li><li><span style="line-height: 1.5;">Remove Gaps-Only Characters — Removes all characters that are gaps only.</span></li><li><span style="line-height: 1.5;">Remove Internal Node Labels — Deletes labels from all internal nodes.</span></li><li><span style="line-height: 1.5;">Remove Invariant Characters — Removes all characters that have one state or fewer.</span></li><li><span style="line-height: 1.5;">Remove item... — Removes an item in each cell of a continuous matrix.</span></li><li><span style="line-height: 1.5;">Remove Tokens — Removes tokens from start or end of the taxon name.</span></li><li><span style="line-height: 1.5;">Rename item... — Renames an item of a continuous matrix.</span></li><li><span style="line-height: 1.5;">Replaces text in taxon names — Replaces text in taxon names.</span></li><li><span style="line-height: 1.5;">Reshuffle Character — Supplies characters that are reshufflings of an existing character.</span></li><li><span style="line-height: 1.5;">Reshuffle States Within Characters — Shuffles (permutes) character states among taxa within each character.</span></li><li><span style="line-height: 1.5;">Reshuffle States Within Taxa — Shuffles (permutes) character states among characters within each taxon.</span></li><li><span style="line-height: 1.5;">Reshuffle Terminal Taxa — Shuffles (permutes) the taxa among the terminal nodes.</span></li><li><span style="line-height: 1.5;">Reshuffle Within Characters (Taxa Partitioned) — Shuffles (permutes) character states among taxa of each taxa partition, within each character.</span></li><li><span style="line-height: 1.5;">Reshuffle Within Taxa (Char. Partitioned) — Shuffles (permutes) character states among characters of each character partition, within each taxon.</span></li><li><span style="line-height: 1.5;">Resolve Polytomies (to 0-length branches) — Resolves polytomies arbitrarily and assigned the resulting new branches zero length.</span></li><li><span style="line-height: 1.5;">Retention Index for Character — Calculates the Retention Index (RI) for a tree and character.</span></li><li><span style="line-height: 1.5;">Retention Index for Matrix — Calculates the Retention Index (RI) for a tree and matrix.</span></li><li><span style="line-height: 1.5;">Reverse Complement — Reverses the DNA sequence and complements the bases.</span></li><li><span style="line-height: 1.5;">Reverse Sequence — Alters data by reversing sequence of states.</span></li><li><span style="line-height: 1.5;">Rhetenor Package Introduction — Multivariate analyses and morphometrics.</span></li><li><span style="line-height: 1.5;">Right-Ladderize Tree — Sets the tree to its ladderized (right) version.</span></li><li><span style="line-height: 1.5;">Root tree — Sets the tree to rooted.</span></li><li><span style="line-height: 1.5;">Root tree with selected taxa as outgroup — Roots the tree between the selected taxa and the remainder, if possible.</span></li><li><span style="line-height: 1.5;">Rooting status — Indicates whether the tree is rooted or unrooted.</span></li><li><span style="line-height: 1.5;">s of Slatkin &amp; Maddison — Calculates 's' of Slaktin &amp; Maddison 1989 for tree of genes associated with given populations</span></li><li><span style="line-height: 1.5;">Sample Trees from Separate NEXUS File — Supplies a fixed number of randomly-sampled trees directly from a file, without bringing the contained tree block entirely into memory. This allows much larger blocks of trees to be used within constraints of memory, but will make some calculations slower. This module does not know how many trees are in the file, and hence may attempt to read files beyond the number in the file.</span></li><li><span style="line-height: 1.5;">Save matrix copies — Provides for the saving of copies of matrices to separate files. This is available under the Characters menu.</span></li><li><span style="line-height: 1.5;">Scale All Branch Lengths — Adjusts a tree's branch lengths by multiplying them by an amount.</span></li><li><span style="line-height: 1.5;">Scale Selected Branch Lengths — Adjusts lengths of a tree's selected branches by multiplying them by an amount.</span></li><li><span style="line-height: 1.5;">Scattergram — Helps make scattergram charts.</span></li><li><span style="line-height: 1.5;">Scroll To Data — Provides a tool that will scroll the data matrix to the next cell that contains data (i.e., that contains something other than a gap/inapplicable)</span></li><li><span style="line-height: 1.5;">Search Data — Manages data-searching modules.</span></li><li><span style="line-height: 1.5;">Search for Better Tree — Finds better trees (by the chosen optimality criterion) by rearranging the current tree.</span></li><li><span style="line-height: 1.5;">Search within clade — Supplies a tool for tree windows that searches for a better branch arrangement within clade touched.</span></li><li><span style="line-height: 1.5;">Select Branches — Provides a tool and menu with which to select branches in a tree window.</span></li><li><span style="line-height: 1.5;">Select by Matrix Comparison — Selects cells of the matrix that differ compared to another matrix</span></li><li><span style="line-height: 1.5;">Select By Search (Find all) — Selects cells of the matrix according to whether their text contains a given string</span></li><li><span style="line-height: 1.5;">Select Lowercase Ends — Selects ends of a DNA sequence up to the first block of N sites in a row with uppercase (confident) symbols (the user chooses N). Assuming that lowercase letters are used for less certain base calls, this can be used to select and then trim (by painting with gaps) poorly-sequenced terminal regions.</span></li><li><span style="line-height: 1.5;">Select Matching Taxa — Selects taxa according to whether their sequence of states matches the sequence of states in the selected taxon</span></li><li><span style="line-height: 1.5;">Select Same Distribution — Selects cells of the matrix according to whether their distribution of states across taxa in a character is the same as that in the selected character or block of cells</span></li><li><span style="line-height: 1.5;">Select Same Subsequence — Selects cells of the matrix according to whether their sequence of states matches the sequence of states in the selected taxon or block of cells</span></li><li><span style="line-height: 1.5;">Select Taxa — Provides a tool with which to select taxa in a clade in a tree window.</span></li><li><span style="line-height: 1.5;">Select Taxa from Name Search — Select taxa based upon a name search.</span></li><li><span style="line-height: 1.5;">Selected Block Mover — Moves selected blocks of multiple sequences.</span></li><li><span style="line-height: 1.5;">Selected Taxa Convex in Tree — Determines if the selected taxa are convex in a tree. That is, does there exist a rooting of the tree in which the selected taxa form a clade?</span></li><li><span style="line-height: 1.5;">Selected Taxa Form Clade — Determines if the selected taxa form a clade in a tree.</span></li><li><span style="line-height: 1.5;">Selection Summary Strip — Shows the strip at bottom of matrix with character reference.</span></li><li><span style="line-height: 1.5;">Semistrict Consensus — Calculates the semistrict consensus tree.</span></li><li><span style="line-height: 1.5;">Sequence Information Editor — Edits character state names within the editor info panel.</span></li><li><span style="line-height: 1.5;">Sequence Length — Reports the length (total minus gaps) of a molecular sequence in a taxon.</span></li><li><span style="line-height: 1.5;">Set Author — Sets the author for this machine and account.</span></li><li><span style="line-height: 1.5;">Set Polytomy Assumption to Default — Sets assumption that any polytomies in the tree are whatever is current default (which can be set in the submenu File&gt;Defaults&gt;Tree Defaults).</span></li><li><span style="line-height: 1.5;">Set Polytomy Assumption to Hard — Sets assumption that any polytomies in the tree are hard.</span></li><li><span style="line-height: 1.5;">Set Polytomy Assumption to Soft — Sets assumption that any polytomies in the tree are soft.</span></li><li>Set to Arbitrary Symmetrical — Sets the tree to an arbitrary symmetrical tree.</li><li><span style="line-height: 1.5;">Set to Lower Case — Alters nucleotide data to lower case coding (e.g, to indicate less certain).</span></li><li><span style="line-height: 1.5;">Set to Upper Case — Alters nucleotide data to upper case coding (e.g, to indicate more certain).</span></li><li><span style="line-height: 1.5;">Set Tree Defaults — Sets the default state for polytomies &amp; other aspects of tree handling.</span></li><li><span style="line-height: 1.5;">Shade numbers — A module that displays numbers at tree nodes using labels. This is a display-only module, and would be hired by another module that organizes assigning numbers to the nodes.</span></li><li><span style="line-height: 1.5;">Shade states — Shows the states at nodes by shading branches or nodes using colors, black and white, or shades of gray.</span></li><li><span style="line-height: 1.5;">Shared Clades — Calculates the number of shared clades between two trees (excludes the clade consisting of all taxa).</span></li><li><span style="line-height: 1.5;">Shared History Distances (Covariance matrix) from Tree — Distances among taxa implied by length of shared history (distance from root to most recent ancestor). This is the most common Covariance/Variance matrix used in GLS methods. Unassigned branches are treated as of length 1.0.</span></li><li><span style="line-height: 1.5;">Shared Partitions — Calculates the number of shared partitions between two trees. If used as a distance, then converted by subtracting shared partitions from the maximum possible, i.e. the number of partitions in the subtree of shared taxa in the tree with the most partitions.</span></li><li><span style="line-height: 1.5;">Shared Proportion Distances implied by Tree — Distances among taxa implied by proportion of total branchlength (root to tip) that is shared (Shared Proportion distances). Unassigned branches are treated as of length 1.0.</span></li><li><span style="line-height: 1.5;">Shift Other To Match — Shifts other sequences to match the one selected.</span></li><li><span style="line-height: 1.5;">Shift to Minimize Stop Codons — Shifts each sequence by 0, 1, or 2 bases to minimize number of stop codons implied.</span></li><li>Show Columns for All Matrices — Shows the Has Data column for all matrices.</li><li><span style="line-height: 1.5;">Show Distance Matrix — Writes distance matrix to log.</span></li><li><span style="line-height: 1.5;">Show Employee Tree — Shows the window listing the tree of employees of the module of a given window.</span></li><li><span style="line-height: 1.5;">Show List with Genetic Codes — Shows the List of Character window with the genetic codes column on.</span></li><li><span style="line-height: 1.5;">Show Node Numbers — Shows the node numbers on a tree.</span></li><li><span style="line-height: 1.5;">Show Percentile — Calculates and shows percentile boundaries. Finds the upper (right) or lower (left) tail of the distribution that contains the specified percentage of the distribution. If there is not a boundary that exactly matches the specified percentage, then the closest boundary that does not exceed the percentage is shown.</span></li><li><span style="line-height: 1.5;">Show text — Shows text in a window</span></li><li><span style="line-height: 1.5;">Show Tree — Provides tool in the List of Trees window to request that tree be shown in window.</span></li><li><span style="line-height: 1.5;">Show Y = X — Shows the Y=X line on a scattergram</span></li><li><span style="line-height: 1.5;">Shuffle states among taxa — Alters data by shuffling states among taxa within a character.</span></li><li><span style="line-height: 1.5;">Side Pusher — Pushes all sequences on one side.</span></li><li><span style="line-height: 1.5;">Simple (categorical data) — Imports and exports simple matrices that consist of basic categorical data. Exported data will consist of default symbols (0, 1, 2, ...).</span></li><li><span style="line-height: 1.5;">Simple (continuous data) — Imports and exports simple matrices that consist of basic continuous data.</span></li><li><span style="line-height: 1.5;">Simple (DNA/RNA) — Imports and exports simple matrices that consist of DNA/RNA sequence data.</span></li><li><span style="line-height: 1.5;">Simple (Geographic data) — Imports and exports simple matrices that consist of basic geographic data. The first character must be the latitude, and the second character the longitude.</span></li><li><span style="line-height: 1.5;">Simple (protein) — Imports and exports simple matrices that consist of amino acid sequence data.</span></li><li><span style="line-height: 1.5;">Simple Tree Window — Displays a single tree.</span></li><li><span style="line-height: 1.5;">Simplicity Manager — A small module to supervise the interface for simplification management.</span></li><li><span style="line-height: 1.5;">Simplified NEXUS — Exports NEXUS files, for instance of old fashioned style (using DATA block).</span></li><li><span style="line-height: 1.5;">Simplify Taxon Names — Simplifies taxon names so that they will work with MrBayes, RAxML, etc..</span></li><li><span style="line-height: 1.5;">Simulate Ancestral States — Simulates ancestral states on the nodes of a tree.</span></li><li><span style="line-height: 1.5;">Simulated Characters — Supplies simulated characters.</span></li><li><span style="line-height: 1.5;">Simulated Characters on Trees — Supplies simulated characters each from a respective tree.</span></li><li><span style="line-height: 1.5;">Simulated Matrices on Current Tree — Supplies simulated character matrices.</span></li><li><span style="line-height: 1.5;">Simulated Matrices on Trees — Supplies simulated character matrices, each evolved on a different of a series of trees.</span></li><li><span style="line-height: 1.5;">Simulated Tree Blocks — Creates a tree block of simulated trees.</span></li><li><span style="line-height: 1.5;">Simulated Trees — Supplies trees from simulations.</span></li><li><span style="line-height: 1.5;">Single Linkage — Supplies trees obtained from Single Linkage clustering.</span></li><li><span style="line-height: 1.5;">Sister Diversification — Compares sister clades and returns the P value of the one-tailed null hypothesis that uniform clades of state 1 are smaller than uniform clades of state 0 among those with different values (P value calculated by Binomial probability).</span></li><li><span style="line-height: 1.5;">Small State Names Editor — Edits character state names within the editor info panel.</span></li><li><span style="line-height: 1.5;">Sort Taxa (data) — Provides a tool with which to sort taxa automatically.</span></li><li><span style="line-height: 1.5;">Speciation/Extinction Likelihood — Calculates likelihoods using a speciation/extinction model reduced from the BiSSE model (Maddison, Midford &amp; Otto 2007) </span></li><li><span style="line-height: 1.5;">Speciation/extinction Likelihood Calculator — Calculates likelihoods using a speciation/extinction model reduced from the BiSSE model (Maddison, Midford &amp; Otto 2007) </span></li><li><span style="line-height: 1.5;">Splits multi-sequence block — Splits blocks of multiple sequences.</span></li><li><span style="line-height: 1.5;">SPR Rearranger — Rearranges a tree by subtree pruning and regrafting (SPR).</span></li><li><span style="line-height: 1.5;">Sprinkle Missing — Randomly converts entries in the character matrix to missing data, with a certain probability (i.e., it &quot;sprinkles&quot; missing data around the matrix).</span></li><li><span style="line-height: 1.5;">Sprinkle Missing in Cells — Fills cells with a missing data with a specified probability.</span></li><li><span style="line-height: 1.5;">Square Line Tree — Draws trees with square branches made out of lines rather than polygons.</span></li><li><span style="line-height: 1.5;">Square Root transform — Alters continuous data by taking the square root.</span></li><li><span style="line-height: 1.5;">Standardize — Alters continuous data by standardizing to have mean = 0 and unbiased estimate of variance = 1. Modifies only the first item of a multi-item matrix</span></li><li><span style="line-height: 1.5;">State Change Summarizer (over trees) — Summarizes reconstructions of state changes of a character over a series of trees.</span></li><li><span style="line-height: 1.5;">State Consistency Moving Window Strip — Displays overall consistency of states in a moving window as an info strip in a character matrix editor.</span></li><li><span style="line-height: 1.5;">State Consistency Strip — Displays overall consistency of states in each character as an info strip in a character matrix editor.</span></li><li><span style="line-height: 1.5;">State Names Strip — Shows the strip at bottom of matrix with state names.</span></li><li><span style="line-height: 1.5;">States Information (in List of Characters window) — Supplies basic character state information for characters in character list window.</span></li><li><span style="line-height: 1.5;">Stepmatrix — Supplies editor for and manages stepmatrices (cost matrices).</span></li><li><span style="line-height: 1.5;">Steps in Character — Calculates the parsimony steps in a character for a given tree.</span></li><li><span style="line-height: 1.5;">Stochastic Character Evolution Package Introduction — Provides stochastic models and simulations of character evolution.</span></li><li><span style="line-height: 1.5;">Stochastic Character Mapping (Categorical) — Coordinates reconstruction of ancestral states using stochastic character mapping. </span></li><li><span style="line-height: 1.5;">Stored Character Order — Supplies current order applied to characters for character list window.</span></li><li><span style="line-height: 1.5;">Stored Characters — Supplies characters from data files (as opposed to simulated characters, for example).</span></li><li><span style="line-height: 1.5;">Stored Genetic Code — Supplies a user-specified genetic code, stored in the file.</span></li><li><span style="line-height: 1.5;">Stored Matrices — Supplies character matrices from data files (as opposed to simulated characters, for example).</span></li><li><span style="line-height: 1.5;">Stored Parsimony Model — Supplies a user-specified model of character evolution, for parsimony calculations, stored in the file. This can be different from the model assigned to the character as current in the List of Characters window.</span></li><li><span style="line-height: 1.5;">Stored Probability Model — Supplies a user-specified model of character evolution stored in the file.</span></li><li><span style="line-height: 1.5;">Stored Probability Model for Simulation — Supplies a user-specified model of character evolution stored in the file for simulation.</span></li><li><span style="line-height: 1.5;">Stored Taxa — Supplies taxa stored, for instance in a file.</span></li><li><span style="line-height: 1.5;">Stored Taxa Associations — Supplies associations between taxa that are stored, for instance in a file.</span></li><li><span style="line-height: 1.5;">Stored Taxon Pairs — Supplies taxon pairs stored, for instance in a file.</span></li><li><span style="line-height: 1.5;">Stored Tree Blocks — Supplies lists of trees stored, for instance in a file.</span></li><li><span style="line-height: 1.5;">Stored Trees — Supplies trees stored, for instance in a file.</span></li><li><span style="line-height: 1.5;">Strict Consensus — Calculates the strict consensus tree.</span></li><li><span style="line-height: 1.5;">Sum of Branch Lengths — Calculates the sum of branch lengths of the tree, treating unassigned lengths as 0. The length of the root is counted.</span></li><li><span style="line-height: 1.5;">Summarize Changes In Selected Clade Over Trees — Summarizes reconstructions of state changes of a character on descendants of selected branch, over a series of trees. The branches defined are those that are the descendants of the single branch selected on the tree in the tree window.</span></li><li><span style="line-height: 1.5;">Summarize State Changes Over Trees — Summarizes reconstructions of state changes of a character over a series of trees.</span></li><li><span style="line-height: 1.5;">Tab-delimited categorical data file — Imports simple tab-delimited files of categorical data.</span></li><li><span style="line-height: 1.5;">Tab-delimited continuous data file — Imports and exports simple tab-delimited files of continuous data.</span></li><li><span style="line-height: 1.5;">Taxa Association Package Introduction — Provides utilities for managing taxa associations (e.g. for gene/species analyses).</span></li><li><span style="line-height: 1.5;">Taxa blocks list — Makes windows listing blocks of Taxa and information about them.</span></li><li><span style="line-height: 1.5;">Taxa of data matrix — Indicates taxa of data matrix.</span></li><li><span style="line-height: 1.5;">Taxa of tree block — Indicates taxa of tree block.</span></li><li><span style="line-height: 1.5;">Taxa partition from Taxa associations — Constructs a taxa partition from an association.</span></li><li><span style="line-height: 1.5;">Taxa Scattergram — Makes a scatterplot displaying two values (such as proportion of gaps in a sequence, etc.) for a series of taxa.</span></li><li><span style="line-height: 1.5;">Taxon Group Color — Colors by Taxon Group.</span></li><li><span style="line-height: 1.5;">Taxon Group Colors — Shows color assigned to taxon group.</span></li><li><span style="line-height: 1.5;">Taxon Group Symbol — Shows symbol assigned to taxon group.</span></li><li><span style="line-height: 1.5;">Taxon Group Symbol Size — Shows size of symbol assigned to taxon group.</span></li><li><span style="line-height: 1.5;">Taxon Group Visibility in Maps — Shows whether or not a taxon group is shown on a map.</span></li><li><span style="line-height: 1.5;">Taxon Instability Among Trees — Calculates for each taxon the degree to which its implied unweighted patristic distances between that taxon and others differs among trees. For each taxon i this sums over all tree pairs x and y and over all other taxa j: | Dijx - Dijy |/(Dijx + Dijy)^^z where Dijq is the distance between taxa i and j on tree q. Close relationships are emphasized if the exponent z is higher (default is 2).</span></li><li><span style="line-height: 1.5;">Taxon Link — Supplies a tool for tree windows that jumps to hypertext links for taxa.</span></li><li><span style="line-height: 1.5;">Taxon List — Makes windows listing taxa and information about them.</span></li><li><span style="line-height: 1.5;">Taxon Pairs Histogram — Makes a chart showing some value for each of many pairs of taxa.</span></li><li><span style="line-height: 1.5;">Taxon Pairs Scattergram — Makes a chart comparing two values for each of many pairs of taxa.</span></li><li><span style="line-height: 1.5;">Taxon selection coordinator — Coordinates taxon selection.</span></li><li><span style="line-height: 1.5;">Taxon Wand (data) — Provides a tool with which to select taxa automatically.</span></li><li><span style="line-height: 1.5;">Terminal ? to Gaps — Converts terminal missing data codings to gaps.</span></li><li><span style="line-height: 1.5;">Terminal Gaps to ? — Converts terminal gaps to missing data.</span></li><li><span style="line-height: 1.5;">Text Window — Displays Text in a window.</span></li><li><span style="line-height: 1.5;">Ti/Tv Rate Matrix Model — Defines and maintains two-parameter rate matrix models with a transition and transversion rates.</span></li><li><span style="line-height: 1.5;">TNT — Imports and exports TNT files.</span></li><li><span style="line-height: 1.5;">Tokenize Names — Tokenizes names. This can be useful for reading trees produced from exported files.</span></li><li><span style="line-height: 1.5;">Top BLAST Matches — Does a BLAST search against GenBank on selected data and returns the top BLAST matches for each sequence selected.</span></li><li><span style="line-height: 1.5;">Trace All Characters — Summarizes for each node the reconstructions of the states at all characters of the tree.</span></li><li><span style="line-height: 1.5;">Trace Character History — Traces a history of character evolution on the nodes of a drawn tree.</span></li><li><span style="line-height: 1.5;">Trace Character Over Trees — Summarizes at each node reconstructions of the states of a character over a series of trees. The summary is shown on the current tree; if you want to show it on a consensus of the trees, make sure that the current tree is that consensus.</span></li><li><span style="line-height: 1.5;">Transform Trees from Other Source — Transforms trees from another source.</span></li><li><span style="line-height: 1.5;">Translate DNA to Protein — Translates a stored DNA matrix to protein.</span></li><li><span style="line-height: 1.5;">Tree Block Directly From File — Supplies a block of trees directly from a file</span></li><li><span style="line-height: 1.5;">Tree Block Values — Makes a chart showing some value for each of many tree blocks.</span></li><li><span style="line-height: 1.5;">Tree Blocks from Files Listed In File — Supplies tree blocks in files listed in a file.</span></li><li><span style="line-height: 1.5;">Tree Blocks List — Makes windows listing tree blocks and information about them.</span></li><li><span style="line-height: 1.5;">Tree Congruent with Constraint Tree Topology — Determines if tree matches topology of a given constraint tree. This module does not handle backbone constraints; all trees must have the same taxa present. For backbone constraint trees, where the constraint tree need not contain all taxa, use the 'Tree Congruent with Backbone Constraint Tree Topology' module.</span></li><li><span style="line-height: 1.5;">Tree Depth — Calculates the deepest path (in branch length) from terminals to the root, treating unassigned lengths as 0. The length of the root is not counted.</span></li><li><span style="line-height: 1.5;">Tree Farm Package Introduction — Extra utilities for trees, including comparisons, randomizations and manipulations.</span></li><li><span style="line-height: 1.5;">Tree from ToL Web Project... — Gets the tree for the page of the Tree of Life Web Project for the group specified.</span></li><li><span style="line-height: 1.5;">Tree of context — Supplies a single tree from the nearest tree context (e.g., an available tree window).</span></li><li><span style="line-height: 1.5;">Tree of Life Web Project — Provides a URL to a taxon on the Tree of Life Web Project</span></li><li><span style="line-height: 1.5;">Tree of Life Web Project Interface Package Introduction — Provides interfaces to the Tree of Life Web Project.</span></li><li><span style="line-height: 1.5;">Tree Search — Supplies trees resulting from a search to optimize some value.</span></li><li><span style="line-height: 1.5;">Tree Search Package Introduction — Coordinates simple searches for optimal trees.</span></li><li><span style="line-height: 1.5;">Tree Utility Coordinator — Coordinates use of tree utilities in tree window</span></li><li><span style="line-height: 1.5;">Tree Value Satisfies Criterion — Determines if the tree has a value either greater than, less than or equal to one specified.</span></li><li><span style="line-height: 1.5;">Tree value using 2 characters — Coordinates the calculation of a number for a tree based on 2 characters.</span></li><li><span style="line-height: 1.5;">Tree value using character — Coordinates the calculation of a number for a tree based on a character.</span></li><li><span style="line-height: 1.5;">Tree value using character matrix — Calculates a value for the tree using a character data matrix.</span></li><li><span style="line-height: 1.5;">Tree Window — Makes a basic tree window, which contains a tool palette. Hires assistants for the tree window (e.g., Trace Character).</span></li><li><span style="line-height: 1.5;">Tree Window Coordinator — Coordinates the creation of basic tree windows.</span></li><li><span style="line-height: 1.5;">TreeBase — Provides a URL to a taxon in TreeBase</span></li><li><span style="line-height: 1.5;">Treelength — Calculates the parsimony treelength of a given tree and matrix.</span></li><li><span style="line-height: 1.5;">Treelength — Calculates the parsimony length of a tree.</span></li><li><span style="line-height: 1.5;">Trees &amp; Diversification Characters — Evolves a series of trees, each tied to a single character controlling diversification rates.</span></li><li><span style="line-height: 1.5;">Trees List — Makes windows listing trees and information about them.</span></li><li><span style="line-height: 1.5;">Trees Package Introduction — Basic management and utilities for phylogenetic trees.</span></li><li><span style="line-height: 1.5;">Trees Scattergram — Makes a scatterplot displaying two values (such as likelihoods, parsimony scores, imbalance statistics, correlations,etc.) for a series of trees.</span></li><li><span style="line-height: 1.5;">Trim Terminal Gap Characters — Trims characters at edges of matrix that are gaps only.</span></li><li><span style="line-height: 1.5;">Trim Terminal Incomplete Codons — Trims any incomplete codons from the start and end of each sequence.</span></li><li><span style="line-height: 1.5;">Truncate taxon names — Truncates taxon names.</span></li><li><span style="line-height: 1.5;">Type of character data matrix — Indicates type of character data matrix (e.g., continuous, DNA, etc.).</span></li><li><span style="line-height: 1.5;">Type of Model — Indicates type of model (e.g., Mk1, step-matrix) in list window.</span></li><li><span style="line-height: 1.5;">uBio — Provides a URL to a taxon in uBio</span></li><li><span style="line-height: 1.5;">Unambiguous Changes — Counts number of unambiguous changes along branch below based upon the character reconstruction.</span></li><li><span style="line-height: 1.5;">Uncorrected (p) distance (DNA) — Uncorrected (p) distance from a DNA matrix.</span></li><li><span style="line-height: 1.5;">Uncorrected Distance (General) — Uncorrected distance from a character matrix.</span></li><li><span style="line-height: 1.5;">Uniform speciation (Yule) — Generates tree by simple uniform probability speciation (a Yule process) as done by Harding (1971). The chance of speciation is equal for all tips.</span></li><li><span style="line-height: 1.5;">Uniform Speciation with Sampling — Generates tree by simple uniform probability speciation (a Yule process), with subsequent sampling of species.</span></li><li><span style="line-height: 1.5;">Unique ID (characters) — Shows unique id assigned to character.</span></li><li><span style="line-height: 1.5;">Unique ID (taxa) — Shows unique id assigned to taxon.</span></li><li><span style="line-height: 1.5;">Unique IDs default — Sets the preferences for storing unique ids.</span></li><li><span style="line-height: 1.5;">Unroot tree — Sets the tree to unrooted.</span></li><li><span style="line-height: 1.5;">UPGMA — Supplies trees obtained from UPGMA clustering.</span></li><li><span style="line-height: 1.5;">Use Trees from Separate NEXUS File — Supplies trees directly from a file, without bringing the contained tree block entirely into memory. This is a special purpose module designed to allow much larger blocks of trees to be used within constraints of memory, but will make some calculations slower. Except for this special use, we recommend you use Include or Link from the file menu to access external tree files. This module does NOT copy the trees into your main data file, and so if you save your main data file then move it or the tree file, the data file will no longer be able to find the trees. This module does not know how many trees are in the file, and hence may attempt to read files beyond the number in the file.</span></li><li><span style="line-height: 1.5;">User-Specified Nucleotide Frequency Model — Defines and maintains user-specified nucleotide frequency models.</span></li><li><span style="line-height: 1.5;">Utilities — Provides a menu for utilities</span></li><li><span style="line-height: 1.5;">Value for 2 characters with current tree — Coordinates the calculation of a number for two characters based on a current tree.</span></li><li><span style="line-height: 1.5;">Value for Clade of Selected Taxa — Coordinates the calculation of a value for the clade of selected terminal taxa. If no terminal taxa are selected, the value at the root is given. If the selected taxa do not make a clade, no answer is returned.</span></li><li><span style="line-height: 1.5;">Value for Node Selected In Current Tree — Coordinates the calculation of a value for a node selected in a current tree. If no node or more than one node is selected in current tree, no result is returned. If the assessed tree does not share the same clade of the selected node, no answer is returned.</span></li><li><span style="line-height: 1.5;">Value from Tree — Calculates a value for each taxon using a tree.</span></li><li><span style="line-height: 1.5;">Value of item in taxon — Gives the value of an item for in a continuous character, for a specific taxon.</span></li><li><span style="line-height: 1.5;">Values (Tree information panel)... — Makes the Values section of the Tree Information Panel to display values pertaining to the tree.</span></li><li><span style="line-height: 1.5;">Values for Current Tree — Makes the legend in a tree window to display the tree name and values pertaining to the tree.</span></li><li><span style="line-height: 1.5;">Values for Nodes — Shows on a drawn tree various possible numbers at the nodes.</span></li><li><span style="line-height: 1.5;">Variable (Mult. States) — Indicates whether a character has multiple observed states. If taxa are selected, only the selected taxa are examined.</span></li><li><span style="line-height: 1.5;">Variable among taxa — Selects characters that are variable among the selected taxa</span></li><li><span style="line-height: 1.5;">Visibility of Matrix — Indicates whether character data matrix is hidden or not.</span></li><li><span style="line-height: 1.5;">Voucher Database — Lists the voucher database for a taxon.</span></li><li><span style="line-height: 1.5;">Voucher ID Code — Lists the voucher ID code for a taxon.</span></li><li><span style="line-height: 1.5;">Voucher Info from Flexible Table — Supplies voucher information from a flexible tab-delimited table in a text file.</span></li><li><span style="line-height: 1.5;">Web page link — Provides a window to jump to a web page</span></li><li><span style="line-height: 1.5;">Whole or Submodel — Indicates whether model is a complete or partial model of character evolution.</span></li><li><span style="line-height: 1.5;">Window Holder — Helps other modules by holding their windows.</span></li><li><span style="line-height: 1.5;">With probability p — Selects characters randomly with probability p.</span></li><li><span style="line-height: 1.5;">Within-group PCA — Performs within-group principal components analysis on a continous-valued matrix.</span></li></ul>
<!-- InstanceEndEditable -->

    
    
    </div>
	<!-- This clearing element should immediately follow the #mainContent div in order to force the #container div to contain all child floats --><br class="clearfloat" />
  <div id="footer">
    <p> ©2018 David R. Maddison &amp; Wayne P. Maddison</p>
  <!-- end #footer --></div>
<!-- end #container --></div>

<script type="text/javascript">
<!--


var OverviewOfMesquitePanel = null;
var GettingStartedPanel = null;
var BasicMesquiteOperationPanel = null;
var TaxaTreesCharactersPanel = null;
var AnalysesPanel = null;
var WorkflowsPanel = null;
var GettingHelpPanel = null;
var TechnicalDetailsPanel = null;
var HistoryPanel = null;
var PublishingResultsPanel = null;

if (qsParm['BasicMesquiteOperationPanel']=='open')
 	BasicMesquiteOperationPanel = new Spry.Widget.CollapsiblePanel("BasicMesquiteOperationPanel", {contentIsOpen: true});
else
 	BasicMesquiteOperationPanel = new Spry.Widget.CollapsiblePanel("BasicMesquiteOperationPanel", {contentIsOpen: false});
	
if (qsParm['OverviewOfMesquitePanel']=='open')
 	OverviewOfMesquitePanel = new Spry.Widget.CollapsiblePanel("OverviewOfMesquitePanel", {contentIsOpen: true});
else
 	OverviewOfMesquitePanel = new Spry.Widget.CollapsiblePanel("OverviewOfMesquitePanel", {contentIsOpen: false});
	
if (qsParm['GettingStartedPanel']=='open')
 	GettingStartedPanel = new Spry.Widget.CollapsiblePanel("GettingStartedPanel", {contentIsOpen: true});
else
 	GettingStartedPanel = new Spry.Widget.CollapsiblePanel("GettingStartedPanel", {contentIsOpen: false});

if (qsParm['TaxaTreesCharactersPanel']=='open')
 	TaxaTreesCharactersPanel = new Spry.Widget.CollapsiblePanel("TaxaTreesCharactersPanel", {contentIsOpen: true});
else
 	TaxaTreesCharactersPanel = new Spry.Widget.CollapsiblePanel("TaxaTreesCharactersPanel", {contentIsOpen: false});

if (qsParm['AnalysesPanel']=='open')
 	AnalysesPanel = new Spry.Widget.CollapsiblePanel("AnalysesPanel", {contentIsOpen: true});
else
 	AnalysesPanel = new Spry.Widget.CollapsiblePanel("AnalysesPanel", {contentIsOpen: false});

if (qsParm['WorkflowsPanel']=='open')
 	WorkflowsPanel = new Spry.Widget.CollapsiblePanel("WorkflowsPanel", {contentIsOpen: true});
else
 	WorkflowsPanel = new Spry.Widget.CollapsiblePanel("WorkflowsPanel", {contentIsOpen: false});

if (qsParm['GettingHelpPanel']=='open')
 	GettingHelpPanel = new Spry.Widget.CollapsiblePanel("GettingHelpPanel", {contentIsOpen: true});
else
 	GettingHelpPanel = new Spry.Widget.CollapsiblePanel("GettingHelpPanel", {contentIsOpen: false});
    
if (qsParm['TechnicalDetailsPanel']=='open')
 	TechnicalDetailsPanel = new Spry.Widget.CollapsiblePanel("TechnicalDetailsPanel", {contentIsOpen: true});
else
 	TechnicalDetailsPanel = new Spry.Widget.CollapsiblePanel("TechnicalDetailsPanel", {contentIsOpen: false});

if (qsParm['HistoryPanel']=='open')
 	HistoryPanel = new Spry.Widget.CollapsiblePanel("HistoryPanel", {contentIsOpen: true});
else
 	HistoryPanel = new Spry.Widget.CollapsiblePanel("HistoryPanel", {contentIsOpen: false});
if (qsParm['PublishingResultsPanel']=='open')
 	PublishingResultsPanel = new Spry.Widget.CollapsiblePanel("PublishingResultsPanel", {contentIsOpen: true});
else
 	PublishingResultsPanel = new Spry.Widget.CollapsiblePanel("PublishingResultsPanel", {contentIsOpen: false});


//-->
</script>
</body>
<!-- InstanceEnd --></html>
